<!DOCTYPE HTML>
<html lang="en" class="sidebar-visible no-js light">
    <head>
        <!-- Book generated using mdBook -->
        <meta charset="UTF-8">
        <title>gget</title>
        <meta name="robots" content="noindex" />


        <!-- Custom HTML head -->
        <!-- Google tag (gtag.js) -->
        <script async src="https://www.googletagmanager.com/gtag/js?id=G-ZM4N3XBEW7"></script>
        <script>
          window.dataLayer = window.dataLayer || [];
          function gtag(){dataLayer.push(arguments);}
          gtag('js', new Date());
        
          gtag('config', 'G-ZM4N3XBEW7');
        </script>

        <meta name="description" content="gget enables efficient querying of genomic reference databases">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="theme-color" content="#ffffff" />

        <link rel="shortcut icon" href="favicon.png">
        <link rel="stylesheet" href="css/variables.css">
        <link rel="stylesheet" href="css/general.css">
        <link rel="stylesheet" href="css/chrome.css">
        <link rel="stylesheet" href="css/print.css" media="print">

        <!-- Fonts -->
        <link rel="stylesheet" href="FontAwesome/css/font-awesome.css">
        <link rel="stylesheet" href="fonts/fonts.css">

        <!-- Highlight.js Stylesheets -->
        <link rel="stylesheet" href="highlight.css">
        <link rel="stylesheet" href="tomorrow-night.css">
        <link rel="stylesheet" href="ayu-highlight.css">

        <!-- Custom theme stylesheets -->

    </head>
    <body>
    <div id="body-container">
        <!-- Provide site root to javascript -->
        <script>
            var path_to_root = "";
            var default_theme = window.matchMedia("(prefers-color-scheme: dark)").matches ? "navy" : "light";
        </script>

        <!-- Work around some values being stored in localStorage wrapped in quotes -->
        <script>
            try {
                var theme = localStorage.getItem('mdbook-theme');
                var sidebar = localStorage.getItem('mdbook-sidebar');

                if (theme.startsWith('"') && theme.endsWith('"')) {
                    localStorage.setItem('mdbook-theme', theme.slice(1, theme.length - 1));
                }

                if (sidebar.startsWith('"') && sidebar.endsWith('"')) {
                    localStorage.setItem('mdbook-sidebar', sidebar.slice(1, sidebar.length - 1));
                }
            } catch (e) { }
        </script>

        <!-- Set the theme before any content is loaded, prevents flash -->
        <script>
            var theme;
            try { theme = localStorage.getItem('mdbook-theme'); } catch(e) { }
            if (theme === null || theme === undefined) { theme = default_theme; }
            var html = document.querySelector('html');
            html.classList.remove('no-js')
            html.classList.remove('light')
            html.classList.add(theme);
            html.classList.add('js');
        </script>

        <!-- Hide / unhide sidebar before it is displayed -->
        <script>
            var html = document.querySelector('html');
            var sidebar = null;
            if (document.body.clientWidth >= 1080) {
                try { sidebar = localStorage.getItem('mdbook-sidebar'); } catch(e) { }
                sidebar = sidebar || 'visible';
            } else {
                sidebar = 'hidden';
            }
            html.classList.remove('sidebar-visible');
            html.classList.add("sidebar-" + sidebar);
        </script>

        <nav id="sidebar" class="sidebar" aria-label="Table of contents">
            <div class="sidebar-scrollbox">
                <ol class="chapter"><li class="chapter-item expanded affix "><a href="en/introduction.html">Introduction</a></li><li class="chapter-item expanded affix "><a href="en/updates.html">What's New?</a></li><li class="chapter-item expanded affix "><li class="part-title">User Guide</li><li class="chapter-item expanded "><a href="en/installation.html">Installation</a></li><li class="chapter-item expanded "><a href="en/quick_start_guide.html">Quick Start Guide</a></li><li class="chapter-item expanded affix "><li class="part-title">Manual</li><li class="chapter-item expanded "><a href="en/alphafold.html">gget alphafold</a></li><li class="chapter-item expanded "><a href="en/archs4.html">gget archs4</a></li><li class="chapter-item expanded "><a href="en/blast.html">gget blast</a></li><li class="chapter-item expanded "><a href="en/blat.html">gget blat</a></li><li class="chapter-item expanded "><a href="en/cellxgene.html">gget cellxgene</a></li><li class="chapter-item expanded "><a href="en/cosmic.html">gget cosmic</a></li><li class="chapter-item expanded "><a href="en/diamond.html">gget diamond</a></li><li class="chapter-item expanded "><a href="en/elm.html">gget elm</a></li><li class="chapter-item expanded "><a href="en/enrichr.html">gget enrichr</a></li><li class="chapter-item expanded "><a href="en/gpt.html">gget gpt</a></li><li class="chapter-item expanded "><a href="en/info.html">gget info</a></li><li class="chapter-item expanded "><a href="en/muscle.html">gget muscle</a></li><li class="chapter-item expanded "><a href="en/pdb.html">gget pdb</a></li><li class="chapter-item expanded "><a href="en/ref.html">gget ref</a></li><li class="chapter-item expanded "><a href="en/search.html">gget search</a></li><li class="chapter-item expanded "><a href="en/setup.html">gget setup</a></li><li class="chapter-item expanded "><a href="en/seq.html">gget seq</a></li><li class="spacer"></li><li class="chapter-item expanded "><a href="en/contributing.html">Contributing Guide</a></li><li class="chapter-item expanded "><a href="en/code_of_conduct.html">Code of Conduct</a></li><li class="chapter-item expanded "><a href="en/cite.html">Citation Guidelines</a></li><li class="spacer"></li><li class="chapter-item expanded affix "><li class="part-title">Espa√±ol</li><li class="chapter-item expanded "><a href="es/introduction.html">Introduci√≥n</a></li><li class="chapter-item expanded "><a href="es/updates.html">¬°Lo m√°s reciente!</a></li><li class="chapter-item expanded affix "><li class="part-title">Gu√≠a del usario</li><li class="chapter-item expanded "><a href="es/installation.html">Instalaci√≥n</a></li><li class="chapter-item expanded "><a href="es/quick_start_guide.html">Gu√≠a de inicio r√°pido</a></li><li class="chapter-item expanded affix "><li class="part-title">Manu√°l</li><li class="chapter-item expanded "><a href="es/alphafold.html">gget alphafold</a></li><li class="chapter-item expanded "><a href="es/archs4.html">gget archs4</a></li><li class="chapter-item expanded "><a href="es/blast.html">gget blast</a></li><li class="chapter-item expanded "><a href="es/blat.html">gget blat</a></li><li class="chapter-item expanded "><a href="es/cellxgene.html">gget cellxgene</a></li><li class="chapter-item expanded "><a href="es/cosmic.html">gget cosmic</a></li><li class="chapter-item expanded "><a href="es/diamond.html">gget diamond</a></li><li class="chapter-item expanded "><a href="es/elm.html">gget elm</a></li><li class="chapter-item expanded "><a href="es/enrichr.html">gget enrichr</a></li><li class="chapter-item expanded "><a href="es/gpt.html">gget gpt</a></li><li class="chapter-item expanded "><a href="es/info.html">gget info</a></li><li class="chapter-item expanded "><a href="es/muscle.html">gget muscle</a></li><li class="chapter-item expanded "><a href="es/pdb.html">gget pdb</a></li><li class="chapter-item expanded "><a href="es/ref.html">gget ref</a></li><li class="chapter-item expanded "><a href="es/search.html">gget search</a></li><li class="chapter-item expanded "><a href="es/setup.html">gget setup</a></li><li class="chapter-item expanded "><a href="es/seq.html">gget seq</a></li><li class="spacer"></li><li class="chapter-item expanded "><a href="es/contributing.html">Gu√≠a de contribuci√≥n</a></li><li class="chapter-item expanded "><a href="es/code_of_conduct.html">Codigo de conducto</a></li><li class="chapter-item expanded "><a href="es/cite.html">C√≥mo citar</a></li></ol>
            </div>
            <div id="sidebar-resize-handle" class="sidebar-resize-handle"></div>
        </nav>

        <!-- Track and set sidebar scroll position -->
        <script>
            var sidebarScrollbox = document.querySelector('#sidebar .sidebar-scrollbox');
            sidebarScrollbox.addEventListener('click', function(e) {
                if (e.target.tagName === 'A') {
                    sessionStorage.setItem('sidebar-scroll', sidebarScrollbox.scrollTop);
                }
            }, { passive: true });
            var sidebarScrollTop = sessionStorage.getItem('sidebar-scroll');
            sessionStorage.removeItem('sidebar-scroll');
            if (sidebarScrollTop) {
                // preserve sidebar scroll position when navigating via links within sidebar
                sidebarScrollbox.scrollTop = sidebarScrollTop;
            } else {
                // scroll sidebar to current active section when navigating via "next/previous chapter" buttons
                var activeSection = document.querySelector('#sidebar .active');
                if (activeSection) {
                    activeSection.scrollIntoView({ block: 'center' });
                }
            }
        </script>

        <div id="page-wrapper" class="page-wrapper">

            <div class="page">
                                <div id="menu-bar-hover-placeholder"></div>
                <div id="menu-bar" class="menu-bar sticky">
                    <div class="left-buttons">
                        <button id="sidebar-toggle" class="icon-button" type="button" title="Toggle Table of Contents" aria-label="Toggle Table of Contents" aria-controls="sidebar">
                            <i class="fa fa-bars"></i>
                        </button>
                        <button id="theme-toggle" class="icon-button" type="button" title="Change theme" aria-label="Change theme" aria-haspopup="true" aria-expanded="false" aria-controls="theme-list">
                            <i class="fa fa-paint-brush"></i>
                        </button>
                        <ul id="theme-list" class="theme-popup" aria-label="Themes" role="menu">
                            <li role="none"><button role="menuitem" class="theme" id="light">Light</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="rust">Rust</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="coal">Coal</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="navy">Navy</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="ayu">Ayu</button></li>
                        </ul>
                        <button id="search-toggle" class="icon-button" type="button" title="Search. (Shortkey: s)" aria-label="Toggle Searchbar" aria-expanded="false" aria-keyshortcuts="S" aria-controls="searchbar">
                            <i class="fa fa-search"></i>
                        </button>
                    </div>

                    <h1 class="menu-title">gget</h1>

                    <div class="right-buttons">
                        <a href="print.html" title="Print this book" aria-label="Print this book">
                            <i id="print-button" class="fa fa-print"></i>
                        </a>

                    </div>
                </div>

                <div id="search-wrapper" class="hidden">
                    <form id="searchbar-outer" class="searchbar-outer">
                        <input type="search" id="searchbar" name="searchbar" placeholder="Search this book ..." aria-controls="searchresults-outer" aria-describedby="searchresults-header">
                    </form>
                    <div id="searchresults-outer" class="searchresults-outer hidden">
                        <div id="searchresults-header" class="searchresults-header"></div>
                        <ul id="searchresults">
                        </ul>
                    </div>
                </div>

                <!-- Apply ARIA attributes after the sidebar and the sidebar toggle button are added to the DOM -->
                <script>
                    document.getElementById('sidebar-toggle').setAttribute('aria-expanded', sidebar === 'visible');
                    document.getElementById('sidebar').setAttribute('aria-hidden', sidebar !== 'visible');
                    Array.from(document.querySelectorAll('#sidebar a')).forEach(function(link) {
                        link.setAttribute('tabIndex', sidebar === 'visible' ? 0 : -1);
                    });
                </script>

                <div id="content" class="content">
                    <main>
                        <p><a href="https://pypi.org/project/gget"><img src="https://img.shields.io/pypi/v/gget" alt="pypi version" /></a>
<a href="https://anaconda.org/bioconda/gget"><img src="https://anaconda.org/bioconda/gget/badges/version.svg" alt="image" /></a>
<a href="https://pepy.tech/project/gget"><img src="https://static.pepy.tech/personalized-badge/gget?period=total&amp;units=international_system&amp;left_color=grey&amp;right_color=brightgreen&amp;left_text=Downloads" alt="Downloads" /></a>
<a href="https://anaconda.org/bioconda/gget"><img src="https://img.shields.io/conda/dn/bioconda/gget?logo=Anaconda" alt="Conda" /></a>
<a href="en/LICENSE"><img src="https://img.shields.io/pypi/l/gget" alt="license" /></a>
<img src="https://github.com/pachterlab/gget/actions/workflows/ci.yml/badge.svg" alt="status" />
<img src="https://github.com/lauraluebbert/test_gget_alphafold/actions/workflows/CI_alphafold.yml/badge.svg" alt="status" />
<a href="https://github.com/pachterlab/gget/"><img src="https://img.shields.io/github/stars/pachterlab/gget.svg?style=social" alt="Star on GitHub" /></a></p>
<p><a href="https://raw.githubusercontent.com/pachterlab/gget/main/figures/gget_overview.png"><img align="right" width="50%" height="50%" src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_overview.png?raw=true" /></a></p>
<h1 id="welcome"><a class="header" href="#welcome">Welcome!</a></h1>
<p><code>gget</code> is a free, open-source command-line tool and Python package that enables efficient querying of genomic databases.<br />
<br>
<code>gget</code> consists of a collection of separate but interoperable modules, each designed to facilitate one type of database querying in a single line of code.<br />
<br></p>
<p><a href="en//gget/en/alphafold.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_alphafold.png?raw=true" width="32%" height="32%" /></a>
<a href="en//gget/en/archs4.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_archs4.png?raw=true" width="32%" height="32%" /></a>
<a href="en//gget/en/blast.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_blast.png?raw=true" width="32%" height="32%" /></a></p>
<p><a href="en//gget/en/blat.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_blat.png?raw=true" width="32%" height="32%" /></a>
<a href="en//gget/en/cellxgene.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_cellxgene.png?raw=true" width="32%" height="32%" /></a>
<a href="en//gget/en/cosmic.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_cosmic.png?raw=true" width="32%" height="32%" /></a></p>
<p><a href="en//gget/en/diamond.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_diamond.png?raw=true" width="32%" height="32%" /></a>
<a href="en//gget/en/elm.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_elm.png?raw=true" width="32%" height="32%" /></a>
<a href="en//gget/en/enrichr.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_enrichr.png?raw=true" width="32%" height="32%" /></a></p>
<p><a href="en//gget/en/info.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_info.png?raw=true" width="32%" height="32%" /></a>
<a href="en//gget/en/muscle.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_muscle.png?raw=true" width="32%" height="32%" /></a>
<a href="en//gget/en/pdb.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_pdb.png?raw=true" width="32%" height="32%" /></a></p>
<p><a href="en//gget/en/ref.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_ref.png?raw=true" width="32%" height="32%" /></a>
<a href="en//gget/en/search.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_search.png?raw=true" width="32%" height="32%" /></a>
<a href="en//gget/en/seq.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_seq.png?raw=true" width="32%" height="32%" /></a> </p>
<h3 id="more-tutorials"><a class="header" href="#more-tutorials"><a href="https://github.com/pachterlab/gget_examples">More tutorials</a></a></h3>
<br>  
<p>If you use <code>gget</code> in a publication, please <a href="en//gget/en/cite.html">cite*</a>:</p>
<pre><code>Luebbert, L., &amp; Pachter, L. (2023). Efficient querying of genomic reference databases with gget. Bioinformatics. https://doi.org/10.1093/bioinformatics/btac836
</code></pre>
<p>Read the article here: <a href="https://doi.org/10.1093/bioinformatics/btac836">https://doi.org/10.1093/bioinformatics/btac836</a></p>
<div style="break-before: page; page-break-before: always;"></div><h2 id="-whats-new"><a class="header" href="#-whats-new">‚ú® What's new</a></h2>
<p><strong>Version ‚â• 0.28.3</strong> (January 22, 2024):</p>
<ul>
<li><strong><a href="en/./search.html"><code>gget search</code></a> and <a href="en/./ref.html"><code>gget ref</code></a> now also support fungi üçÑ, protists üåù, and invertebrate metazoa üêù üêú üêå üêô (in addition to vertebrates and plants)</strong></li>
<li><strong>New module: <a href="en/./cosmic.html"><code>gget cosmic</code></a></strong></li>
<li><a href="en/./enrichr.html"><code>gget enrichr</code></a>: Fix duplicate scatter dots in plot when pathway names are duplicated</li>
<li><a href="en/./elm.html"><code>gget elm</code></a>:
<ul>
<li>Changed ortho results column name 'Ortholog_UniProt_ID' to 'Ortholog_UniProt_Acc' to correctly reflect the column contents, which are UniProt Accessions. 'UniProt ID' was changed to 'UniProt Acc' in the documentation for all <code>gget</code> modules.</li>
<li>Changed ortho results column name 'motif_in_query' to 'motif_inside_subject_query_overlap'.</li>
<li>Added interaction domain information to results (new columns: &quot;InteractionDomainId&quot;, &quot;InteractionDomainDescription&quot;, &quot;InteractionDomainName&quot;).</li>
<li>The regex string for regular expression matches was encapsulated as follows: &quot;(?=(regex))&quot; (instead of directly passing the regex string &quot;regex&quot;) to enable capturing all occurrences of a motif when the motif length is variable and there are repeats in the sequence (<a href="https://regex101.com/r/HUWLlZ/1">https://regex101.com/r/HUWLlZ/1</a>).</li>
</ul>
</li>
<li><a href="en/./setup.html"><code>gget setup</code></a>: Use the <code>out</code> argument to specify a directory the ELM database will be downloaded into. Completes <a href="https://github.com/pachterlab/gget/issues/119">this feature request</a>.</li>
<li><a href="en/./diamond.html"><code>gget diamond</code></a>: The DIAMOND command is now run with <code>--ignore-warnings</code> flag, allowing niche sequences such as amino acid sequences that only contain nucleotide characters and repeated sequences. This is also true for DIAMOND alignments performed within <a href="en/./elm.html"><code>gget elm</code></a>.</li>
<li><a href="en/./ref.html"><code>gget ref</code></a> and <a href="en/./search.html"><code>gget search</code></a> back-end change: the current Ensembl release is fetched from the new <a href="https://ftp.ensembl.org/pub/VERSION">release file</a> on the Ensembl FTP site to avoid errors during uploads of new releases.</li>
<li><a href="en/./search.html"><code>gget search</code></a>: 
<ul>
<li>FTP link results (<code>--ftp</code>) are saved in txt file format instead of json.</li>
<li>Fix URL links to Ensembl gene summary for species with a subspecies name and invertebrates.</li>
</ul>
</li>
<li><a href="en/./ref.html"><code>gget ref</code></a>:
<ul>
<li>Back-end changes to increase speed</li>
<li>New argument: <code>list_iv_species</code> to list all available invertebrate species (can be combined with the <code>release</code> argument to fetch all species available from a specific Ensembl release)</li>
</ul>
</li>
</ul>
<p><strong>Version ‚â• 0.28.2</strong> (November 15, 2023):</p>
<ul>
<li><a href="en/./info.html"><code>gget info</code></a>: Return a logging error message when the NCBI server fails for a reason other than a fetch fail (this is an error on the server side rather than an error with <code>gget</code>)</li>
<li>Replace deprecated 'text' argument to find()-type methods whenever used with dependency <code>BeautifulSoup</code></li>
<li><a href="en/elm.html"><code>gget elm</code></a>: Remove false positive and true negative instances from returned results</li>
<li><a href="en/elm.html"><code>gget elm</code></a>: Add <code>expand</code> argument</li>
</ul>
<p><strong>Version ‚â• 0.28.0</strong> (November 5, 2023):</p>
<ul>
<li>Updated documentation of <a href="en/./muscle.html"><code>gget muscle</code></a> to add a tutorial on how to visualize sequences with sequence name lengths + slight change to returned visualization so it's a bit more robust to varying sequence names</li>
<li><a href="en/./muscle.html"><code>gget muscle</code></a> now also allows a list of sequences as input (as an alternative to providing the path to a FASTA file)</li>
<li>Allow missing gene filter for <a href="en/cellxgene.html"><code>gget cellxgene</code></a>  (fixes <a href="https://github.com/pachterlab/gget/issues/110">bug</a>)</li>
<li><a href="en/./seq.html"><code>gget seq</code></a>: Allow missing gene names (fixes <a href="https://github.com/pachterlab/gget/issues/107">https://github.com/pachterlab/gget/issues/107</a>)</li>
<li>New arguments for <a href="en/enrichr.html"><code>gget enrichr</code></a>: Use arguments <code>kegg_out</code> and <code>kegg_rank</code> to create an image of the KEGG pathway with the genes from the enrichment analysis highlighted (thanks to <a href="https://github.com/pachterlab/gget/pull/106">this PR</a> by <a href="https://github.com/noriakis">Noriaki Sato</a>)</li>
<li>New modules: <a href="en/elm.html"><code>gget elm</code></a> and <a href="en/diamond.html"><code>gget diamond</code></a></li>
</ul>
<p><strong>Version ‚â• 0.27.9</strong> (August 7, 2023):</p>
<ul>
<li>New arguments for <a href="en/enrichr.html"><code>gget enrichr</code></a>: Use argument <code>background_list</code> to provide a list of background genes</li>
<li><a href="en/search.html"><code>gget search</code></a> now also searches <a href="https://ensembl.org/">Ensembl</a> synonyms (in addition to gene descriptions and names) to return more comprehensive search results (thanks to <a href="https://github.com/KleinSamuel">Samuel Klein</a> for the <a href="https://github.com/pachterlab/gget/issues/90">suggestion</a>)</li>
</ul>
<p><strong>Version ‚â• 0.27.8</strong> (July 12, 2023):</p>
<ul>
<li>New argument for <a href="en/search.html"><code>gget search</code></a>: Specify the Ensembl release from which information is fetched with <code>-r</code> <code>--release</code></li>
<li>Fixed <a href="https://github.com/pachterlab/gget/issues/91">bug</a> in <a href="en/pdb.html"><code>gget pdb</code></a> (this bug was introduced in version 0.27.5)</li>
</ul>
<p><strong>Version ‚â• 0.27.7</strong> (May 15, 2023):</p>
<ul>
<li>Moved dependencies for modules <a href="en/gpt.html"><code>gget gpt</code></a> and <a href="en/cellxgene.html"><code>gget cellxgene</code></a> from automatically installed requirements to <a href="en/setup.html"><code>gget setup</code></a>.</li>
<li>Updated <a href="en/alphafold.html"><code>gget alphafold</code></a> dependencies for compatibility with Python &gt;= 3.10.</li>
<li>Added <code>census_version</code> argument to <a href="en/cellxgene.html"><code>gget cellxgene</code></a>.</li>
</ul>
<p><strong>Version ‚â• 0.27.6</strong> (May 1, 2023) (YANKED due to problems with dependencies -&gt; replaced with version 0.27.7):</p>
<ul>
<li>Thanks to PR by <a href="https://github.com/tdido">Tom√°s Di Domenico</a>: <a href="en/search.html"><code>gget search</code></a> can now also query plant üå± Ensembl IDs.</li>
<li>New module: <a href="en/cellxgene.html"><code>gget cellxgene</code></a></li>
</ul>
<p><strong>Version ‚â• 0.27.5</strong> (April 6, 2023):</p>
<ul>
<li>Updated <a href="en/search.html"><code>gget search</code></a> to function correctly with new <a href="https://pypi.org/project/pandas/2.0.0/">Pandas</a> version 2.0.0 (released on April 3rd, 2023) as well as older versions of Pandas</li>
<li>Updated <a href="en/info.html"><code>gget info</code></a> with new flags <code>uniprot</code> and <code>ncbi</code> which allow turning off results from these databases independently to save runtime (note: flag <code>ensembl_only</code> was deprecated)</li>
<li>All gget modules now feature a <code>-q / --quiet</code> (Python: <code>verbose=False</code>) flag to turn off progress information</li>
</ul>
<p><strong>Version ‚â• 0.27.4</strong> (March 19, 2023):</p>
<ul>
<li>New module: <a href="en/gpt.html"><code>gget gpt</code></a></li>
</ul>
<p><strong>Version ‚â• 0.27.3</strong> (March 11, 2023):</p>
<ul>
<li><a href="en/info.html"><code>gget info</code></a> excludes PDB IDs by default to increase speed (PDB results can be included using flag <code>--pdb</code> / <code>pdb=True</code>).</li>
</ul>
<p><strong>Version ‚â• 0.27.2</strong> (January 1, 2023):</p>
<ul>
<li>Updated <a href="en/alphafold.html"><code>gget alphafold</code></a> to <a href="https://github.com/deepmind/alphafold/releases/tag/v2.3.0">DeepMind's AlphaFold v2.3.0</a> (including new arguments <code>multimer_for_monomer</code> and <code>multimer_recycles</code>)</li>
</ul>
<p><strong>Version ‚â• 0.27.0</strong> (December 10, 2022):</p>
<ul>
<li>Updated <a href="en/alphafold.html"><code>gget alphafold</code></a> to match recent changes by DeepMind</li>
<li>Updated version number to match <a href="https://github.com/lauraluebbert">gget's creator</a>'s age following a long-standing Pachter lab tradition</li>
</ul>
<p><strong>Version ‚â• 0.3.13</strong> (November 11, 2022):</p>
<ul>
<li>Reduced runtime for <a href="en/enrichr.html"><code>gget enrichr</code></a> and <a href="en/archs4.html"><code>gget archs4</code></a> when used with Ensembl IDs</li>
</ul>
<p><strong>Version ‚â• 0.3.12</strong> (November 10, 2022):</p>
<ul>
<li><a href="en/info.html"><code>gget info</code></a> now also returns subcellular localisation data from UniProt</li>
<li>New <a href="en/info.html"><code>gget info</code></a> flag <code>ensembl_only</code> returns only Ensembl results</li>
<li>Reduced runtime for <a href="en/info.html"><code>gget info</code></a> and <a href="en/seq.html"><code>gget seq</code></a></li>
</ul>
<p><strong>Version ‚â• 0.3.11</strong> (September 7, 2022):</p>
<ul>
<li>New module: <a href="en/pdb.html"><code>gget pdb</code></a></li>
</ul>
<p><strong>Version ‚â• 0.3.10</strong> (September 2, 2022):</p>
<ul>
<li><a href="en/alphafold.html"><code>gget alphafold</code></a> now also returns pLDDT values for generating plots from output without rerunning the program (also see the <a href="https://github.com/pachterlab/gget/discussions/39">gget alphafold FAQ</a>)</li>
</ul>
<p><strong>Version ‚â• 0.3.9</strong> (August 25, 2022):</p>
<ul>
<li>Updated openmm installation instructions for <a href="en/alphafold.html"><code>gget alphafold</code></a></li>
</ul>
<p><strong>Version ‚â• 0.3.8</strong> (August 12, 2022):</p>
<ul>
<li>Fixed mysql-connector-python version requirements</li>
</ul>
<p><strong>Version ‚â• 0.3.7</strong> (August 9, 2022):</p>
<ul>
<li><strong>NOTE:</strong> The <a href="http://ftp.ensembl.org/pub/">Ensembl FTP site</a> changed its structure on August 8, 2022. Please upgrade to <code>gget</code> version ‚â• 0.3.7 if you use <a href="en/ref.html"><code>gget ref</code></a></li>
</ul>
<p><strong>Version ‚â• 0.3.5</strong> (August 6, 2022):</p>
<ul>
<li>New module: <a href="en/alphafold.html"><code>gget alphafold</code></a></li>
</ul>
<p><strong>Version ‚â• 0.2.6</strong> (July 7, 2022):</p>
<ul>
<li><a href="en/ref.html"><code>gget ref</code></a> now supports plant genomes! üå±</li>
</ul>
<p><strong>Version ‚â• 0.2.5</strong> (June 30, 2022):</p>
<ul>
<li><strong>NOTE:</strong> <a href="https://www.uniprot.org/">UniProt</a> changed the structure of their API on June 28, 2022. Please upgrade to <code>gget</code> version ‚â• 0.2.5 if you use any of the modules querying data from UniProt (<a href="en/info.html"><code>gget info</code></a> and <a href="en/seq.html"><code>gget seq</code></a>).</li>
</ul>
<p><strong>Version ‚â• 0.2.3:</strong> (June 26, 2022):</p>
<ul>
<li>JSON is now the default output format for the command-line interface for modules that previously returned data frame (CSV) format by default (the output can be converted to data frame/CSV using flag <code>[-csv][--csv]</code>). Data frame/CSV remains the default output for Jupyter Lab / Google Colab (and can be converted to JSON with <code>json=True</code>).</li>
<li>For all modules, the first required argument was converted to a positional argument and should not be named anymore in the command-line, e.g. <code>gget ref -s human</code> ‚Üí <code>gget ref human</code>.</li>
<li><code>gget info</code>: <code>[--expand]</code> is deprecated. The module will now always return all of the available information.</li>
<li>Slight changes to the output returned by <code>gget info</code>, including the return of versioned Ensembl IDs.</li>
<li><code>gget info</code> and <code>gget seq</code> now support ü™± WormBase and ü™∞ FlyBase IDs.</li>
<li><code>gget archs4</code> and <code>gget enrichr</code> now also take Ensembl IDs as input with added flag <code>[-e][--ensembl]</code> (<code>ensembl=True</code> in Jupyter Lab / Google Colab).</li>
<li><code>gget seq</code> argument <code>seqtype</code> was replaced by flag <code>[-t][--translate]</code> (<code>translate=True/False</code> in Jupyter Lab / Google Colab) which will return either nucleotide (<code>False</code>) or amino acid (<code>True</code>) sequences.</li>
<li><code>gget search</code> argument <code>seqtype</code> was renamed to <code>id_type</code> for clarity (still taking the same arguments 'gene' or 'transcript').</li>
</ul>
<div style="break-before: page; page-break-before: always;"></div><p><a href="https://pypi.org/project/gget"><img src="https://img.shields.io/pypi/v/gget" alt="pypi version" /></a>
<a href="https://anaconda.org/bioconda/gget"><img src="https://anaconda.org/bioconda/gget/badges/version.svg" alt="image" /></a></p>
<h2 id="installation"><a class="header" href="#installation">Installation</a></h2>
<pre><code class="language-bash">pip install --upgrade gget
</code></pre>
<p>Alternative:</p>
<pre><code class="language-bash">conda install -c bioconda gget
</code></pre>
<p>For use in Jupyter Lab / Google Colab:</p>
<pre><code class="language-python">import gget
</code></pre>
<div style="break-before: page; page-break-before: always;"></div><h2 id="-quick-start-guide"><a class="header" href="#-quick-start-guide">ü™Ñ Quick start guide</a></h2>
<p>Command line:</p>
<pre><code class="language-bash"># Fetch all Homo sapiens reference and annotation FTPs from the latest Ensembl release
$ gget ref homo_sapiens

# Get Ensembl IDs of human genes with &quot;ace2&quot; or &quot;angiotensin converting enzyme 2&quot; in their name/description
$ gget search -s homo_sapiens 'ace2' 'angiotensin converting enzyme 2'

# Look up gene ENSG00000130234 (ACE2) and its transcript ENST00000252519
$ gget info ENSG00000130234 ENST00000252519

# Fetch the amino acid sequence of the canonical transcript of gene ENSG00000130234
$ gget seq --translate ENSG00000130234

# Quickly find the genomic location of (the start of) that amino acid sequence
$ gget blat MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS

# BLAST (the start of) that amino acid sequence
$ gget blast MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS

# Align multiple nucleotide or amino acid sequences against each other (also accepts path to FASTA file)  
$ gget muscle MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS

# Align one or more amino acid sequences against a reference (containing one or more sequences) (local BLAST) (also accepts paths to FASTA files)  
$ gget diamond MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS -ref MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS  

# Use Enrichr for an ontology analysis of a list of genes
$ gget enrichr -db ontology ACE2 AGT AGTR1 ACE AGTRAP AGTR2 ACE3P

# Get the human tissue expression of gene ACE2
$ gget archs4 -w tissue ACE2

# Get the protein structure (in PDB format) of ACE2 as stored in the Protein Data Bank (PDB ID returned by gget info)
$ gget pdb 1R42 -o 1R42.pdb

# Find Eukaryotic Linear Motifs (ELMs) in a protein sequence
$ gget setup elm # setup only needs to be run once
$ gget elm -o results MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS

# Fetch a scRNAseq count matrix (AnnData format) based on specified gene(s), tissue(s), and cell type(s) (default species: human)
$ gget setup cellxgene # setup only needs to be run once
$ gget cellxgene --gene ACE2 SLC5A1 --tissue lung --cell_type 'mucus secreting cell' -o example_adata.h5ad

# Predict the protein structure of GFP from its amino acid sequence
$ gget setup alphafold # setup only needs to be run once
$ gget alphafold MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK
</code></pre>
<p>Python (Jupyter Lab / Google Colab):</p>
<pre><code class="language-python">import gget
gget.ref(&quot;homo_sapiens&quot;)
gget.search([&quot;ace2&quot;, &quot;angiotensin converting enzyme 2&quot;], &quot;homo_sapiens&quot;)
gget.info([&quot;ENSG00000130234&quot;, &quot;ENST00000252519&quot;])
gget.seq(&quot;ENSG00000130234&quot;, translate=True)
gget.blat(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)
gget.blast(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)
gget.muscle([&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;, &quot;MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS&quot;])
gget.diamond(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;, reference=&quot;MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS&quot;)
gget.enrichr([&quot;ACE2&quot;, &quot;AGT&quot;, &quot;AGTR1&quot;, &quot;ACE&quot;, &quot;AGTRAP&quot;, &quot;AGTR2&quot;, &quot;ACE3P&quot;], database=&quot;ontology&quot;, plot=True)
gget.archs4(&quot;ACE2&quot;, which=&quot;tissue&quot;)
gget.pdb(&quot;1R42&quot;, save=True)

gget.setup(&quot;elm&quot;) # setup only needs to be run once
ortho_df, regex_df = gget.elm(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)

gget.setup(&quot;cellxgene&quot;) # setup only needs to be run once
gget.cellxgene(gene = [&quot;ACE2&quot;, &quot;SLC5A1&quot;], tissue = &quot;lung&quot;, cell_type = &quot;mucus secreting cell&quot;)

gget.setup(&quot;alphafold&quot;) # setup only needs to be run once
gget.alphafold(&quot;MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK&quot;)
</code></pre>
<p>Call <code>gget</code> from R using <a href="https://rstudio.github.io/reticulate/">reticulate</a>:</p>
<pre><code class="language-r">system(&quot;pip install gget&quot;)
install.packages(&quot;reticulate&quot;)
library(reticulate)
gget &lt;- import(&quot;gget&quot;)

gget$ref(&quot;homo_sapiens&quot;)
gget$search(list(&quot;ace2&quot;, &quot;angiotensin converting enzyme 2&quot;), &quot;homo_sapiens&quot;)
gget$info(list(&quot;ENSG00000130234&quot;, &quot;ENST00000252519&quot;))
gget$seq(&quot;ENSG00000130234&quot;, translate=TRUE)
gget$blat(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)
gget$blast(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)
gget$muscle(list(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;, &quot;MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS&quot;), out=&quot;out.afa&quot;)
gget$diamond(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;, reference=&quot;MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS&quot;)
gget$enrichr(list(&quot;ACE2&quot;, &quot;AGT&quot;, &quot;AGTR1&quot;, &quot;ACE&quot;, &quot;AGTRAP&quot;, &quot;AGTR2&quot;, &quot;ACE3P&quot;), database=&quot;ontology&quot;)
gget$archs4(&quot;ACE2&quot;, which=&quot;tissue&quot;)
gget$pdb(&quot;1R42&quot;, save=TRUE)
</code></pre>
<h4 id="more-examples"><a class="header" href="#more-examples"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-alphafold-"><a class="header" href="#gget-alphafold-">gget alphafold ü™¢</a></h2>
<p>Predict the 3D structure of a protein from its amino acid sequence using a simplified version of <a href="https://www.deepmind.com/">DeepMind</a>‚Äôs <a href="https://github.com/deepmind/alphafold">AlphaFold2</a> originally released and benchmarked for <a href="https://colab.research.google.com/github/deepmind/alphafold/blob/main/notebooks/AlphaFold.ipynb">AlphaFold Colab</a>.<br />
Returns: Predicted structure (PDB) and alignment error (json).</p>
<p>Before using <code>gget alphafold</code> for the first time, run <code>gget setup alphafold</code> / <code>gget.setup(&quot;alphafold&quot;)</code> once (also see <a href="en/setup.html"><code>gget setup</code></a>).</p>
<p><strong>Positional argument</strong><br />
<code>sequence</code><br />
Amino acid sequence (str), or list of sequences (<em>gget alphafold will automatically use the multimer model if multiple sequences are passed</em>), or path to FASTA file.</p>
<p><strong>Optional arguments</strong><br />
<code>-mr</code> <code>--multimer_recycles</code><br />
The multimer model will continue recycling until the predictions stop changing, up to the limit set here. Default: 3.<br />
For higher accuracy, at the potential cost of longer inference times, set this to 20.</p>
<p><code>-o</code> <code>--out</code><br />
Path to folder to save prediction results in (str). Default: &quot;./[date_time]_gget_alphafold_prediction&quot;.</p>
<p><strong>Flags</strong><br />
<code>-mfm</code> <code>--multimer_for_monomer</code><br />
Use multimer model for a monomer.</p>
<p><code>-r</code> <code>--relax</code><br />
AMBER relax the best model. </p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed. </p>
<p><code>plot</code><br />
Python only. <code>plot=True</code> provides an interactive, 3D graphical overview of the predicted structure and alignment quality using <a href="https://pypi.org/project/py3Dmol/">py3Dmol</a> and <a href="https://matplotlib.org/">matplotlib</a> (default: True).</p>
<p><code>show_sidechains</code><br />
Python only. <code>show_sidechains=True</code> includes side chains in the plot (default: True).</p>
<h3 id="example"><a class="header" href="#example">Example</a></h3>
<pre><code class="language-bash"># Generate new prediction from amino acid sequence
gget alphafold MAAHKGAEHHHKAAEHHEQAAKHHHAAAEHHEKGEHEQAAHHADTAYAHHKHAEEHAAQAAKHDAEHHAPKPH

# Find similar sequences deposited on the PDB for comparative analysis
gget blast --database pdbaa MAAHKGAEHHHKAAEHHEQAAKHHHAAAEHHEKGEHEQAAHHADTAYAHHKHAEEHAAQAAKHDAEHHAPKPH

# Fetch the PDB files of similar structures returned by gget blast for comparison, to get a measure for model quality
gget pdb 3UQ3 -o 3UQ3.pdb
gget pdb 2K42 -o 2K42.pdb
</code></pre>
<pre><code class="language-python"># Python
gget.alphafold(&quot;MAAHKGAEHHHKAAEHHEQAAKHHHAAAEHHEKGEHEQAAHHADTAYAHHKHAEEHAAQAAKHDAEHHAPKPH&quot;)

# Find similar sequences deposited on the PDB for comparative analysis
gget.blast(&quot;MAAHKGAEHHHKAAEHHEQAAKHHHAAAEHHEKGEHEQAAHHADTAYAHHKHAEEHAAQAAKHDAEHHAPKPH&quot;, database=&quot;pdbaa&quot;)

# Fetch the PDB files of similar structures returned by gget blast for comparison, to get a measure for model quality
gget.pdb(&quot;3UQ3&quot;, save=True)
gget.pdb(&quot;2K42&quot;, save=True)
</code></pre>
<p>‚Üí <code>gget alphafold</code> returns the predicted structure (PDB) and predicted alignment error (.json) in a new folder (&quot;./[date_time]_gget_alphafold_prediction&quot;). The use case above exemplifies how to use <a href="en/blast.html"><code>gget blast</code></a> and <a href="en/pdb.html"><code>gget pdb</code></a> for a comparative analysis of the new prediction. PDB files can be viewed interactively in 3D <a href="https://rcsb.org/3d-view">online</a>, or using programs like <a href="https://pymol.org/">PyMOL</a> or <a href="https://www.blender.org/">Blender</a>. To compare two PDB files, you can use <a href="https://rcsb.org/alignment">this website</a>. The Python interface also returns <a href="https://twitter.com/NeuroLuebbert/status/1555968042948915200">interactive plots</a>, which can be generated from the PDB and JSON as described in the <a href="https://github.com/pachterlab/gget/discussions/39">gget alphafold FAQ</a> Q4.</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/4qxGF1tbZ3I?si=mEqQ5oSnDYtg2OP7" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
<h3 id="example-in-google-colab"><a class="header" href="#example-in-google-colab"><a href="https://github.com/pachterlab/gget_examples/blob/main/gget_alphafold.ipynb">Example in Google Colab</a></a></h3>
<h3 id="gget-alphafold-faq"><a class="header" href="#gget-alphafold-faq"><a href="https://github.com/pachterlab/gget/discussions/39">gget alphafold FAQ</a></a></h3>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-archs4-"><a class="header" href="#gget-archs4-">gget archs4 üêÅ</a></h2>
<p>Find the most correlated genes to a gene of interest or find the gene's tissue expression atlas using <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong><br />
<code>gene</code><br />
Short name (gene symbol) of gene of interest, e.g. STAT4.<br />
Alternatively: use flag <code>--ensembl</code> to input an Ensembl gene IDs, e.g. ENSG00000138378.</p>
<p><strong>Optional arguments</strong><br />
<code>-w</code> <code>--which</code><br />
'correlation' (default) or 'tissue'.<br />
'correlation' returns a gene correlation table that contains the 100 most correlated genes to the gene of interest. The Pearson correlation is calculated over all samples and tissues in <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.<br />
'tissue' returns a tissue expression atlas calculated from human or mouse samples (as defined by 'species') in <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.</p>
<p><code>-s</code> <code>--species</code><br />
'human' (default) or 'mouse'.<br />
Defines whether to use human or mouse samples from <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.<br />
(Only for tissue expression atlas.)</p>
<p><code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.csv (or .json). Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-e</code> <code>--ensembl</code><br />
Add this flag if <code>gene</code> is given as an Ensembl gene ID.</p>
<p><code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed. </p>
<h3 id="examples"><a class="header" href="#examples">Examples</a></h3>
<pre><code class="language-bash">gget archs4 ACE2
</code></pre>
<pre><code class="language-python"># Python
gget.archs4(&quot;ACE2&quot;)
</code></pre>
<p>‚Üí Returns the 100 most correlated genes to ACE2:</p>
<div class="table-wrapper"><table><thead><tr><th>gene_symbol</th><th>pearson_correlation</th></tr></thead><tbody>
<tr><td>SLC5A1</td><td>0.579634</td></tr>
<tr><td>CYP2C18</td><td>0.576577</td></tr>
<tr><td>. . .</td><td>. . .</td></tr>
</tbody></table>
</div>
<p><br/><br/></p>
<pre><code class="language-bash">gget archs4 -w tissue ACE2
</code></pre>
<pre><code class="language-python"># Python
gget.archs4(&quot;ACE2&quot;, which=&quot;tissue&quot;)
</code></pre>
<p>‚Üí Returns the tissue expression of ACE2 (by default, human data is used):</p>
<div class="table-wrapper"><table><thead><tr><th>id</th><th>min</th><th>q1</th><th>median</th><th>q3</th><th>max</th></tr></thead><tbody>
<tr><td>System.Urogenital/Reproductive System.Kidney.RENAL CORTEX</td><td>0.113644</td><td>8.274060</td><td>9.695840</td><td>10.51670</td><td>11.21970</td></tr>
<tr><td>System.Digestive System.Intestine.INTESTINAL EPITHELIAL CELL</td><td>0.113644</td><td>5.905560</td><td>9.570450</td><td>13.26470</td><td>13.83590</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td></tr>
</tbody></table>
</div>
<p><br/><br/>
Check out <a href="https://davetang.org/muse/2023/05/16/check-where-a-gene-is-expressed-from-the-command-line/">this tutorial</a> by Dave Tang who wrote an R script to create this figure from the <code>gget archs4</code> JSON output:</p>
<p><img src="https://github.com/pachterlab/gget/assets/56094636/f2a34a9e-beaa-45a5-a678-d38399dd3017" alt="image" /></p>
<h4 id="more-examples-1"><a class="header" href="#more-examples-1"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python.  The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-blast-"><a class="header" href="#gget-blast-">gget blast üí•</a></h2>
<p>BLAST a nucleotide or amino acid sequence to any <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> database.<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong><br />
<code>sequence</code><br />
Nucleotide or amino acid sequence, or path to FASTA or .txt file.</p>
<p><strong>Optional arguments</strong><br />
<code>-p</code> <code>--program</code><br />
'blastn', 'blastp', 'blastx', 'tblastn', or 'tblastx'.<br />
Default: 'blastn' for nucleotide sequences; 'blastp' for amino acid sequences.</p>
<p><code>-db</code> <code>--database</code><br />
'nt', 'nr', 'refseq_rna', 'refseq_protein', 'swissprot', 'pdbaa', or 'pdbnt'.<br />
Default: 'nt' for nucleotide sequences; 'nr' for amino acid sequences.<br />
<a href="https://ncbi.github.io/blast-cloud/blastdb/available-blastdbs.html">More info on BLAST databases</a></p>
<p><code>-l</code> <code>--limit</code><br />
Limits number of hits to return. Default: 50.</p>
<p><code>-e</code> <code>--expect</code><br />
Defines the <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=FAQ#expect">expect value</a> cutoff. Default: 10.0.</p>
<p><code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.csv (or .json). Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-lcf</code> <code>--low_comp_filt</code><br />
Turns on <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=FAQ#LCR">low complexity filter</a>.</p>
<p><code>-mbo</code> <code>--megablast_off</code><br />
Turns off MegaBLAST algorithm. Default: MegaBLAST on (blastn only).</p>
<p><code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed.</p>
<p><code>wrap_text</code><br />
Python only. <code>wrap_text=True</code> displays data frame with wrapped text for easy reading (default: False).</p>
<h3 id="example-1"><a class="header" href="#example-1">Example</a></h3>
<pre><code class="language-bash">gget blast MKWMFKEDHSLEHRCVESAKIRAKYPDRVPVIVEKVSGSQIVDIDKRKYLVPSDITVAQFMWIIRKRIQLPSEKAIFLFVDKTVPQSR
</code></pre>
<pre><code class="language-python"># Python
gget.blast(&quot;MKWMFKEDHSLEHRCVESAKIRAKYPDRVPVIVEKVSGSQIVDIDKRKYLVPSDITVAQFMWIIRKRIQLPSEKAIFLFVDKTVPQSR&quot;)
</code></pre>
<p>‚Üí Returns the BLAST result of the sequence of interest. <code>gget blast</code> automatically detects this sequence as an amino acid sequence and therefore sets the BLAST program to <em>blastp</em> with database <em>nr</em>.</p>
<div class="table-wrapper"><table><thead><tr><th>Description</th><th>Scientific Name</th><th>Common Name</th><th>Taxid</th><th>Max Score</th><th>Total Score</th><th>Query Cover</th><th>...</th></tr></thead><tbody>
<tr><td>PREDICTED: gamma-aminobutyric acid receptor-as...</td><td>Colobus angolensis palliatus</td><td>NaN</td><td>336983</td><td>180</td><td>180</td><td>100%</td><td>...</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>...</td></tr>
</tbody></table>
</div>
<p><strong>BLAST from .fa or .txt file:</strong></p>
<pre><code class="language-bash">gget blast fasta.fa
</code></pre>
<pre><code class="language-python"># Python
gget.blast(&quot;fasta.fa&quot;)
</code></pre>
<p>‚Üí Returns the BLAST results of the first sequence contained in the fasta.fa file. </p>
<h4 id="more-examples-2"><a class="header" href="#more-examples-2"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-blat-"><a class="header" href="#gget-blat-">gget blat üéØ</a></h2>
<p>Find the genomic location of a nucleotide or amino acid sequence using <a href="https://genome.ucsc.edu/cgi-bin/hgBlat">BLAT</a>.<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong><br />
<code>sequence</code><br />
Nucleotide or amino acid sequence, or path to FASTA or .txt file.</p>
<p><strong>Optional arguments</strong><br />
<code>-st</code> <code>--seqtype</code><br />
'DNA', 'protein', 'translated%20RNA', or 'translated%20DNA'.<br />
Default: 'DNA' for nucleotide sequences; 'protein' for amino acid sequences.</p>
<p><code>-a</code> <code>--assembly</code><br />
'human' (hg38) (default), 'mouse' (mm39), 'zebrafinch' (taeGut2),<br />
or any of the species assemblies available <a href="https://genome.ucsc.edu/cgi-bin/hgBlat">here</a> (use short assembly name).</p>
<p><code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.csv (or .json). Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed. </p>
<h3 id="example-2"><a class="header" href="#example-2">Example</a></h3>
<pre><code class="language-bash">gget blat -a taeGut2 MKWMFKEDHSLEHRCVESAKIRAKYPDRVPVIVEKVSGSQIVDIDKRKYLVPSDITVAQFMWIIRKRIQLPSEKAIFLFVDKTVPQSR
</code></pre>
<pre><code class="language-python"># Python
gget.blat(&quot;MKWMFKEDHSLEHRCVESAKIRAKYPDRVPVIVEKVSGSQIVDIDKRKYLVPSDITVAQFMWIIRKRIQLPSEKAIFLFVDKTVPQSR&quot;, assembly=&quot;taeGut2&quot;)
</code></pre>
<p>‚Üí Returns BLAT results for assembly taeGut2 (zebra finch). In the above example, <code>gget blat</code> automatically detects this sequence as an amino acid sequence and therefore sets the BLAT seqtype to <em>protein</em>.</p>
<div class="table-wrapper"><table><thead><tr><th>genome</th><th>query_size</th><th>aligned_start</th><th>aligned_end</th><th>matches</th><th>mismatches</th><th>%_aligned</th><th>...</th></tr></thead><tbody>
<tr><td>taeGut2</td><td>88</td><td>12</td><td>88</td><td>77</td><td>0</td><td>87.5</td><td>...</td></tr>
</tbody></table>
</div>
<h4 id="more-examples-3"><a class="header" href="#more-examples-3"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python.  The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-cellxgene-"><a class="header" href="#gget-cellxgene-">gget cellxgene üç±</a></h2>
<p>Query data from <a href="https://cellxgene.cziscience.com/">CZ CELLxGENE Discover</a> using the <a href="https://github.com/chanzuckerberg/cellxgene-census">CZ CELLxGENE Discover Census</a>.</p>
<p>Returns: An AnnData object containing the count matrix and metadata of single-cell RNA sequencing data from the defined tissues/genes/etc.</p>
<p>Before using <code>gget cellxgene</code> for the first time, run <code>gget setup cellxgene</code> / <code>gget.setup(&quot;cellxgene&quot;)</code> once (also see <a href="en/setup.html"><code>gget setup</code></a>).</p>
<p><strong>Optional arguments</strong><br />
<code>-s</code> <code>--species</code><br />
Choice of 'homo_sapiens' or 'mus_musculus'. Default: 'homo_sapiens'.</p>
<p><code>-g</code> <code>--gene</code><br />
Str or list of gene name(s) or Ensembl ID(s). Default: None.<br />
NOTE: Use <code>-e / --ensembl</code> (Python: <code>ensembl=True</code>) when providing Ensembl ID(s) instead of gene name(s).<br />
See https://cellxgene.cziscience.com/gene-expression for examples of available genes.</p>
<p><code>-cv</code> <code>--census_version</code><br />
Str defining version of Census, e.g. &quot;2023-05-15&quot;, or &quot;latest&quot; or &quot;stable&quot;. Default: &quot;stable&quot;.</p>
<p><code>-cn</code> <code>--column_names</code><br />
List of metadata columns to return (stored in AnnData.obs).<br />
Default: ['dataset_id', 'assay', 'suspension_type', 'sex', 'tissue_general', 'tissue', 'cell_type']<br />
For more options, see: https://api.cellxgene.cziscience.com/curation/ui/#/ -&gt; Schemas -&gt; dataset</p>
<p><code>-o</code> <code>--out</code><br />
Path to file to save generated AnnData .h5ad file (or .csv with <code>-mo / --meta_only</code>).<br />
Required when using from command line!</p>
<p><strong>Flags</strong><br />
<code>-e</code> <code>--ensembl</code><br />
Use when genes are provided as Ensembl IDs instead of gene names.</p>
<p><code>-mo</code> <code>--meta_only</code><br />
Only returns metadata data frame (corresponds to AnnData.obs).</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed.</p>
<p><strong>Optional arguments corresponding to CZ CELLxGENE Discover metadata attributes</strong><br />
<code>--tissue</code><br />
Str or list of tissue(s), e.g. ['lung', 'blood']. Default: None.<br />
See https://cellxgene.cziscience.com/gene-expression for examples of available tissues.</p>
<p><code>--cell_type</code><br />
Str or list of cell type (s), e.g. ['mucus secreting cell', 'neuroendocrine cell']. Default: None.<br />
See https://cellxgene.cziscience.com/gene-expression and select a tissue to see examples of available cell types.</p>
<p><code>--development_stage</code><br />
Str or list of development stage(s). Default: None.</p>
<p><code>--disease</code><br />
Str or list of disease(s). Default: None.</p>
<p><code>--sex</code><br />
Str or list of sex(es), e.g. 'female'. Default: None.</p>
<p><code>--dataset_id</code><br />
Str or list of CELLxGENE dataset ID(s). Default: None.</p>
<p><code>--tissue_general_ontology_term_id</code><br />
Str or list of high-level tissue UBERON ID(s). Default: None.<br />
Tissue labels and their corresponding UBERON IDs are listed <a href="https://github.com/chanzuckerberg/single-cell-data-portal/blob/9b94ccb0a2e0a8f6182b213aa4852c491f6f6aff/backend/wmg/data/tissue_mapper.py">here</a>.</p>
<p><code>--tissue_general</code><br />
Str or list of high-level tissue label(s). Default: None.<br />
Tissue labels and their corresponding UBERON IDs are listed <a href="https://github.com/chanzuckerberg/single-cell-data-portal/blob/9b94ccb0a2e0a8f6182b213aa4852c491f6f6aff/backend/wmg/data/tissue_mapper.py">here</a>.</p>
<p><code>--tissue_ontology_term_id</code><br />
Str or list of tissue ontology term ID(s) as defined in the <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">CELLxGENE dataset schema</a>. Default: None.</p>
<p><code>--assay_ontology_term_id</code><br />
Str or list of assay ontology term ID(s) as defined in the <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">CELLxGENE dataset schema</a>. Default: None.</p>
<p><code>--assay</code><br />
Str or list of assay(s) as defined in the <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">CELLxGENE dataset schema</a>. Default: None.</p>
<p><code>--cell_type_ontology_term_id</code><br />
Str or list of cell type ontology term ID(s) as defined in the <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">CELLxGENE dataset schema</a>. Default: None.</p>
<p><code>--development_stage_ontology_term_id</code><br />
Str or list of development stage ontology term ID(s) as defined in the <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">CELLxGENE dataset schema</a>. Default: None.</p>
<p><code>--disease_ontology_term_id</code><br />
Str or list of disease ontology term ID(s) as defined in the <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">CELLxGENE dataset schema</a>. Default: None.</p>
<p><code>--donor_id</code><br />
Str or list of donor ID(s) as defined in the <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">CELLxGENE dataset schema</a>. Default: None.</p>
<p><code>--self_reported_ethnicity_ontology_term_id</code><br />
Str or list of self-reported ethnicity ontology ID(s) as defined in the <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">CELLxGENE dataset schema</a>. Default: None.</p>
<p><code>--self_reported_ethnicity</code><br />
Str or list of self-reported ethnicity as defined in the <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">CELLxGENE dataset schema</a>. Default: None.</p>
<p><code>--sex_ontology_term_id</code><br />
Str or list of sex ontology ID(s) as defined in the <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">CELLxGENE dataset schema</a>. Default: None.</p>
<p><code>--suspension_type</code><br />
Str or list of suspension type(s) as defined in the <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">CELLxGENE dataset schema</a>. Default: None.</p>
<h3 id="examples-1"><a class="header" href="#examples-1">Examples</a></h3>
<pre><code class="language-bash">gget cellxgene --gene ACE2 ABCA1 SLC5A1 --tissue lung --cell_type 'mucus secreting cell' 'neuroendocrine cell' -o example_adata.h5ad
</code></pre>
<pre><code class="language-python"># Python
adata = gget.cellxgene(
    gene = [&quot;ACE2&quot;, &quot;ABCA1&quot;, &quot;SLC5A1&quot;],
    tissue = &quot;lung&quot;,
    cell_type = [&quot;mucus secreting cell&quot;, &quot;neuroendocrine cell&quot;]
)
adata
</code></pre>
<p>‚Üí Returns an AnnData object containing the scRNAseq ACE2, ABCA1, and SLC5A1 count matrix of 3322 human lung mucus secreting and neuroendocrine cells from CZ CELLxGENE Discover and their corresponding metadata.</p>
<p><br/><br/></p>
<p>Fetch metadata (corresponds to AnnData.obs) only:</p>
<pre><code class="language-bash">gget cellxgene --meta_only --gene ENSMUSG00000015405 --ensembl --tissue lung --species mus_musculus -o example_meta.csv
</code></pre>
<pre><code class="language-python"># Python
df = gget.cellxgene(
    meta_only = True,
    gene = &quot;ENSMUSG00000015405&quot;,
    ensembl = True,
    tissue = &quot;lung&quot;,  
    species = &quot;mus_musculus&quot;
)
df
</code></pre>
<p>‚Üí Returns only the metadata from ENSMUSG00000015405 (ACE2) expression datasets corresponding to mouse lung cells.</p>
<p>Also see: <a href="https://chanzuckerberg.github.io/cellxgene-census/notebooks/api_demo/census_gget_demo.html">https://chanzuckerberg.github.io/cellxgene-census/notebooks/api_demo/census_gget_demo.html</a></p>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python.  The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-cosmic-"><a class="header" href="#gget-cosmic-">gget cosmic ü™ê</a></h2>
<p>Search for genes, mutations, etc associated with cancers using the <a href="https://cancer.sanger.ac.uk/cosmic">COSMIC</a> (Catalogue Of Somatic Mutations In Cancer) database.<br />
Return format: JSON (command-line) or data frame/CSV (Python).<br />
This module was written by <a href="https://github.com/AubakirovArman">@AubakirovArman</a>.</p>
<p><strong>Positional argument</strong><br />
<code>searchterm</code><br />
Search term, which can be a mutation, or gene (or Ensembl ID), or sample, etc. as defined using the <code>entity</code> argument. Example: 'EGFR'</p>
<p><strong>Optional arguments</strong><br />
<code>-e</code> <code>--entity</code><br />
'mutations' (default), 'genes', 'cancer', 'tumour site', 'studies', 'pubmed', or 'samples'.<br />
Defines the type of the supplied search term.</p>
<p><code>-l</code> <code>--limit</code><br />
Limits number of hits to return. Default: 100.</p>
<p><code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.csv (or .json). Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed.</p>
<h3 id="example-3"><a class="header" href="#example-3">Example</a></h3>
<pre><code class="language-bash">gget cosmic -e genes EGFR
</code></pre>
<pre><code class="language-python"># Python
gget.cosmic(&quot;EGFR&quot;, entity=&quot;genes&quot;)
</code></pre>
<p>‚Üí Returns the COSMIC hits for gene 'EGFR' in the format:</p>
<div class="table-wrapper"><table><thead><tr><th>Gene</th><th>Alternate IDs</th><th>Tested samples</th><th>Simple Mutations</th><th>Fusions</th><th>Coding Mutations</th><th>...</th></tr></thead><tbody>
<tr><td>EGFR</td><td>EGFR,ENST00000275493.6,...</td><td>210280</td><td>31900</td><td>0</td><td>31900</td><td>...</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td></tr>
</tbody></table>
</div><div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-diamond-"><a class="header" href="#gget-diamond-">gget diamond üíé</a></h2>
<p>Align multiple protein or translated DNA sequences using <a href="https://www.nature.com/articles/nmeth.3176">DIAMOND</a> (DIAMOND is similar to BLAST, but this is a local computation).<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong><br />
<code>query</code><br />
Sequences (str or list) or path to FASTA file containing sequences to be aligned against the reference.</p>
<p><strong>Required arguments</strong><br />
<code>-ref</code> <code>--reference</code><br />
Reference sequences (str or list) or path to FASTA file containing reference sequences.</p>
<p><strong>Optional arguments</strong><br />
<code>-db</code> <code>--diamond_db</code><br />
Path to save DIAMOND database created from <code>reference</code> (str).<br />
Default: None -&gt; Temporary db file will be deleted after alignment or saved in <code>out</code> if <code>out</code> is provided.</p>
<p><code>-s</code> <code>--sensitivity</code><br />
Sensitivity of alignment (str). Default: &quot;very-sensitive&quot;.<br />
One of the following: fast, mid-sensitive, sensitive, more-sensitive, very-sensitive, or ultra-sensitive.</p>
<p><code>-t</code> <code>--threads</code><br />
Number of threads used (int). Default: 1.</p>
<p><code>-db</code> <code>--diamond_binary</code><br />
Path to DIAMOND binary (str). Default: None -&gt; Uses DIAMOND binary installed with <code>gget</code>.</p>
<p><code>-o</code> <code>--out</code><br />
Path to the folder to save results in (str), e.g. &quot;path/to/directory&quot;. Default: Standard out; temporary files are deleted.</p>
<p><strong>Flags</strong><br />
<code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed.</p>
<h3 id="example-4"><a class="header" href="#example-4">Example</a></h3>
<pre><code class="language-bash"># !!! Make sure to list the positional argument first here so it is not added as a reference sequence
gget diamond GGETISAWESQME ELVISISALIVE LQVEFRANKLIN PACHTERLABRQCKS -ref GGETISAWESQMEELVISISALIVELQVEFRANKLIN PACHTERLABRQCKS
</code></pre>
<pre><code class="language-python"># Python
gget.diamond([&quot;GGETISAWESQME&quot;, &quot;ELVISISALIVE&quot;, &quot;LQVEFRANKLIN&quot;, &quot;PACHTERLABRQCKS&quot;], reference=[&quot;GGETISAWESQMEELVISISALIVELQVEFRANKLIN&quot;, &quot;PACHTERLABRQCKS&quot;])
</code></pre>
<p>‚Üí Returns results in JSON (command-line) or data frame/CSV (Python) format:</p>
<div class="table-wrapper"><table><thead><tr><th>query_accession</th><th>subject_accession</th><th>identity_percentage</th><th>query_seq_length</th><th>subject_seq_length</th><th>length</th><th>mismatches</th><th>gap_openings</th><th>query_start</th><th>query_end</th><th>subject_start</th><th>subject_end</th><th>e-value</th><th>bit_score</th></tr></thead><tbody>
<tr><td>Seq0</td><td>Seq0</td><td>100</td><td>13</td><td>37</td><td>13</td><td>0</td><td>0</td><td>1</td><td>13</td><td>1</td><td>13</td><td>2.82e-09</td><td>30.8</td></tr>
<tr><td>Seq2</td><td>Seq0</td><td>100</td><td>12</td><td>37</td><td>12</td><td>0</td><td>0</td><td>1</td><td>12</td><td>26</td><td>37</td><td>4.35e-08</td><td>27.7</td></tr>
<tr><td>Seq3</td><td>Seq1</td><td>100</td><td>15</td><td>15</td><td>15</td><td>0</td><td>0</td><td>1</td><td>15</td><td>1</td><td>15</td><td>2.01e-11</td><td>36.2</td></tr>
</tbody></table>
</div>
<h4 id="more-examples-4"><a class="header" href="#more-examples-4"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-elm-"><a class="header" href="#gget-elm-">gget elm üé≠</a></h2>
<p>Locally predict Eukaryotic Linear Motifs from an amino acid sequence or UniProt Acc using data from the <a href="http://elm.eu.org/">ELM database</a>. ELM data can be downloaded &amp; distributed for non-commercial use according to the <a href="http://elm.eu.org/media/Elm_academic_license.pdf">ELM Software License Agreement</a>.<br />
Return format: JSON (command-line) or data frame/CSV (Python). This module returns two data frames (or JSON formatted files) (see examples).</p>
<p>Before using <code>gget elm</code> for the first time, run <code>gget setup elm</code> / <code>gget.setup(&quot;elm&quot;)</code> once (also see <a href="en/setup.html"><code>gget setup</code></a>).</p>
<p><strong>Positional argument</strong><br />
<code>sequence</code><br />
Amino acid sequence or Uniprot Acc (str).<br />
When providing a Uniprot Acc, use flag <code>--uniprot</code> (Python: <code>uniprot=True</code>).</p>
<p><strong>Optional arguments</strong><br />
<code>-s</code> <code>--sensitivity</code><br />
Sensitivity of DIAMOND alignment (str). Default: &quot;very-sensitive&quot;.<br />
One of the following: fast, mid-sensitive, sensitive, more-sensitive, very-sensitive, or ultra-sensitive.</p>
<p><code>-t</code> <code>--threads</code><br />
Number of threads used in DIAMOND alignment (int). Default: 1.</p>
<p><code>-bin</code> <code>--diamond_binary</code><br />
Path to DIAMOND binary (str). Default: None -&gt; Uses DIAMOND binary installed with <code>gget</code>.</p>
<p><code>-o</code> <code>--out</code><br />
Path to the folder to save results in (str), e.g. &quot;path/to/directory&quot;. Default: Standard out; temporary files are deleted.</p>
<p><strong>Flags</strong><br />
<code>-u</code> <code>--uniprot</code><br />
Set to True if <code>sequence</code> is a Uniprot Acc instead of an amino acid sequence.</p>
<p><code>-e</code> <code>--expand</code><br />
Expand the information returned in the regex data frame to include the protein names, organisms, and references that the motif was orignally validated on. </p>
<p><code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed.</p>
<h3 id="examples-2"><a class="header" href="#examples-2">Examples</a></h3>
<p>Find ELMs in an amino acid sequence:</p>
<pre><code class="language-bash">gget setup elm          # Downloads/updates local ELM database
gget elm -o gget_elm_results LIAQSIGQASFV
</code></pre>
<pre><code class="language-python"># Python
gget.setup(‚Äúelm‚Äù)      # Downloads/updates local ELM database
ortholog_df, regex_df = gget.elm(&quot;LIAQSIGQASFV&quot;)
</code></pre>
<p>Find ELMs giving a UniProt Acc as input:</p>
<pre><code class="language-bash">gget setup elm          # Downloads/updates local ELM database
gget elm -o gget_elm_results --uniprot Q02410 -e
</code></pre>
<pre><code class="language-python"># Python
gget.setup(‚Äúelm‚Äù)      # Downloads/updates local ELM database
ortholog_df, regex_df = gget.elm(&quot;Q02410&quot;, uniprot=True, expand=True)
</code></pre>
<p>‚Üí Returns two data frames (or JSON formatted dictionaries for command line) containing extensive information about linear motifs associated with orthologous proteins and motifs found in the input sequence directly based on their regex expressions:</p>
<p>ortholog_df:</p>
<div class="table-wrapper"><table><thead><tr><th style="text-align: center">Ortholog_UniProt_Acc</th><th style="text-align: center">ProteinName</th><th style="text-align: center">class_accession</th><th style="text-align: center">ELMIdentifier</th><th style="text-align: center">FunctionalSiteName</th><th style="text-align: center">Description</th><th style="text-align: center">Organism</th><th style="text-align: center">‚Ä¶</th></tr></thead><tbody>
<tr><td style="text-align: center">Q02410</td><td style="text-align: center">APBA1_HUMAN</td><td style="text-align: center">ELME000357</td><td style="text-align: center">LIG_CaMK_CASK_1</td><td style="text-align: center">CASK CaMK domain binding ligand motif</td><td style="text-align: center">Motif that mediates binding to the calmodulin-dependent protein kinase (CaMK) domain of the peripheral plasma membrane protein CASK/Lin2.</td><td style="text-align: center">Homo sapiens</td><td style="text-align: center">‚Ä¶</td></tr>
<tr><td style="text-align: center">Q02410</td><td style="text-align: center">APBA1_HUMAN</td><td style="text-align: center">ELME000091</td><td style="text-align: center">LIG_PDZ_Class_2</td><td style="text-align: center">PDZ domain ligands</td><td style="text-align: center">The C-terminal class 2 PDZ-binding motif is classically represented by a pattern such as</td><td style="text-align: center">Homo sapiens</td><td style="text-align: center">‚Ä¶</td></tr>
</tbody></table>
</div>
<p>regex_df:</p>
<div class="table-wrapper"><table><thead><tr><th style="text-align: center">Instance_accession</th><th style="text-align: center">ELMIdentifier</th><th style="text-align: center">FunctionalSiteName</th><th style="text-align: center">ELMType</th><th style="text-align: center">Description</th><th style="text-align: center">Instances (Matched Sequence)</th><th style="text-align: center">Organism</th><th style="text-align: center">‚Ä¶</th></tr></thead><tbody>
<tr><td style="text-align: center">ELME000321</td><td style="text-align: center">CLV_C14_Caspase3-7</td><td style="text-align: center">Caspase cleavage motif</td><td style="text-align: center">CLV</td><td style="text-align: center">Caspase-3 and Caspase-7 cleavage site.</td><td style="text-align: center">ERSDG</td><td style="text-align: center">Mus musculus</td><td style="text-align: center">‚Ä¶</td></tr>
<tr><td style="text-align: center">ELME000102</td><td style="text-align: center">CLV_NRD_NRD_1</td><td style="text-align: center">NRD cleavage site</td><td style="text-align: center">CLV</td><td style="text-align: center">N-Arg dibasic convertase (NRD/Nardilysin) cleavage site.</td><td style="text-align: center">RRA</td><td style="text-align: center">Rattus norvegicus</td><td style="text-align: center">‚Ä¶</td></tr>
<tr><td style="text-align: center">ELME000100</td><td style="text-align: center">CLV_PCSK_PC1ET2_1</td><td style="text-align: center">PCSK cleavage site</td><td style="text-align: center">CLV</td><td style="text-align: center">NEC1/NEC2 cleavage site.</td><td style="text-align: center">KRD</td><td style="text-align: center">Mus musculus</td><td style="text-align: center">‚Ä¶</td></tr>
<tr><td style="text-align: center">ELME000146</td><td style="text-align: center">CLV_PCSK_SKI1_1</td><td style="text-align: center">PCSK cleavage site</td><td style="text-align: center">CLV</td><td style="text-align: center">Subtilisin/kexin isozyme-1 (SKI1) cleavage site.</td><td style="text-align: center">RLLTA</td><td style="text-align: center">Homo sapiens</td><td style="text-align: center">‚Ä¶</td></tr>
<tr><td style="text-align: center">ELME000231</td><td style="text-align: center">DEG_APCC_DBOX_1</td><td style="text-align: center">APCC-binding Destruction motifs</td><td style="text-align: center">DEG</td><td style="text-align: center">An RxxL-based motif that binds to the Cdh1 and Cdc20 components of APC/C thereby targeting the protein for destruction in a cell cycle dependent manner</td><td style="text-align: center">SRVKLNIVR</td><td style="text-align: center">Saccharomyces cerevisiae S288c</td><td style="text-align: center">‚Ä¶</td></tr>
<tr><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td></tr>
</tbody></table>
</div>
<h4 id="more-examples-5"><a class="header" href="#more-examples-5"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-enrichr-"><a class="header" href="#gget-enrichr-">gget enrichr üí∞</a></h2>
<p>Perform an enrichment analysis on a list of genes using <a href="https://maayanlab.cloud/Enrichr/">Enrichr</a>.<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong><br />
<code>genes</code><br />
Short names (gene symbols) of genes to perform enrichment analysis on, e.g. PHF14 RBM3 MSL1 PHF21A.<br />
Alternatively: use flag <code>--ensembl</code> to input a list of Ensembl gene IDs, e.g. ENSG00000106443 ENSG00000102317 ENSG00000188895.</p>
<p><strong>Other required arguments</strong><br />
<code>-db</code> <code>--database</code><br />
Database to use as reference for the enrichment analysis.<br />
Supports any database listed <a href="https://maayanlab.cloud/Enrichr/#libraries">here</a> under 'Gene-set Library' or one of the following shortcuts:<br />
'pathway'‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ (KEGG_2021_Human)<br />
'transcription'‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ (ChEA_2016)<br />
'ontology'‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ(GO_Biological_Process_2021)<br />
'diseases_drugs'‚ÄÉ‚ÄÉ‚ÄÉ(GWAS_Catalog_2019)<br />
'celltypes'‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ(PanglaoDB_Augmented_2021)<br />
'kinase_interactions'‚ÄÉ  (KEA_2015)</p>
<p><strong>Optional arguments</strong><br />
<code>-bkg_l</code> <code>--background_list</code><br />
Short names (gene symbols) of background genes to perform enrichment analysis on, e.g. NSUN3 POLRMT NLRX1.<br />
Alternatively: use flag <code>--ensembl_background</code> to input a list of Ensembl gene IDs.<br />
See <a href="https://x.com/ChiHoangCaltech/status/1689679611335155712?s=20">this Tweetorial</a> to learn why you should use a background gene list when performing an enrichment analysis.</p>
<p><code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.csv (or .json). (Default: Standard out.)<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><code>-ko</code> <code>--kegg_out</code><br />
Path to the png file the marked KEGG pathway images will be saved in, e.g. path/to/directory/pathway.png. (Default: None)</p>
<p><code>-kr</code> <code>--kegg_rank</code><br />
Rank of the KEGG pathway to be plotted. (Default: 1)</p>
<p><code>figsize</code><br />
Python only. (width, height) of plot in inches. (Default: (10,10))</p>
<p><code>ax</code><br />
Python only. Pass a matplotlib axes object for plot customization. (Default: None)</p>
<p><strong>Flags</strong><br />
<code>-e</code> <code>--ensembl</code><br />
Add this flag if <code>genes</code> are given as Ensembl gene IDs.</p>
<p><code>-e_b</code> <code>--ensembl_background</code><br />
Add this flag if <code>background_list</code> are given as Ensembl gene IDs.</p>
<p><code>-bkg</code> <code>--background</code><br />
If True, use set of &gt; 20,000 default background genes listed <a href="https://github.com/pachterlab/gget/blob/main/gget/constants/enrichr_bkg_genes.txt">here</a>.</p>
<p><code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed. </p>
<p><code>plot</code><br />
Python only. <code>plot=True</code> provides a graphical overview of the first 15 results (default: False).</p>
<h3 id="examples-3"><a class="header" href="#examples-3">Examples</a></h3>
<pre><code class="language-bash">gget enrichr -db ontology ACE2 AGT AGTR1
</code></pre>
<pre><code class="language-python"># Python
gget.enrichr([&quot;ACE2&quot;, &quot;AGT&quot;, &quot;AGTR1&quot;], database=&quot;ontology&quot;, plot=True)
</code></pre>
<p>‚Üí Returns pathways/functions involving genes ACE2, AGT, and AGTR1 from the <em>GO Biological Process 2021</em> database. In Python, <code>plot=True</code> returns a graphical overview of the results:</p>
<p><img src="https://github.com/pachterlab/gget/blob/main/figures/gget_enrichr_results.png?raw=true" alt="alt text" /></p>
<p><br/><br/></p>
<p><strong>Use <code>gget enrichr</code> with a background gene list:</strong><br />
See <a href="https://x.com/ChiHoangCaltech/status/1689679611335155712?s=20">this Tweetorial</a> to learn why you should use a background gene list when performing an enrichment analysis.</p>
<pre><code class="language-bash"># Here, we are passing the input genes first (positional argument 'genes'), so they are not added to the background gene list behind the '-bkgr_l' argument
gget enrichr \
	PHF14 RBM3 MSL1 PHF21A ARL10 INSR JADE2 P2RX7 LINC00662 CCDC101 PPM1B KANSL1L CRYZL1 ANAPC16 TMCC1 CDH8 RBM11 CNPY2 HSPA1L CUL2 PLBD2 LARP7 TECPR2 ZNF302 CUX1 MOB2 CYTH2 SEC22C EIF4E3 ROBO2 ADAMTS9-AS2 CXXC1 LINC01314 ATF7 ATP5F1 \
	-db ChEA_2022 \
	-bkg_l NSUN3 POLRMT NLRX1 SFXN5 ZC3H12C SLC25A39 ARSG DEFB29 PCMTD2 ACAA1A LRRC1 2810432D09RIK SEPHS2 SAC3D1 TMLHE LOC623451 TSR2 PLEKHA7 GYS2 ARHGEF12 HIBCH LYRM2 ZBTB44 ENTPD5 RAB11FIP2 LIPT1 INTU ANXA13 KLF12 SAT2 GAL3ST2 VAMP8 FKBPL AQP11 TRAP1 PMPCB TM7SF3 RBM39 BRI3 KDR ZFP748 NAP1L1 DHRS1 LRRC56 WDR20A STXBP2 KLF1 UFC1 CCDC16 9230114K14RIK RWDD3 2610528K11RIK ACO1 CABLES1 LOC100047214 YARS2 LYPLA1 KALRN GYK ZFP787 ZFP655 RABEPK ZFP650 4732466D17RIK EXOSC4 WDR42A GPHN 2610528J11RIK 1110003E01RIK MDH1 1200014M14RIK AW209491 MUT 1700123L14RIK 2610036D13RIK PHF14 RBM3 MSL1 PHF21A ARL10 INSR JADE2 P2RX7 LINC00662 CCDC101 PPM1B KANSL1L CRYZL1 ANAPC16 TMCC1 CDH8 RBM11 CNPY2 HSPA1L CUL2 PLBD2 LARP7 TECPR2 ZNF302 CUX1 MOB2 CYTH2 SEC22C EIF4E3 ROBO2 ADAMTS9-AS2 CXXC1 LINC01314 ATF7 ATP5F1COX15 TMEM30A NSMCE4A TM2D2 RHBDD3 ATXN2 NFS1 3110001I20RIK BC038156 C330002I19RIK ZFYVE20 POLI TOMM70A LOC100047782 2410012H22RIK RILP A230062G08RIK PTTG1IP RAB1 AFAP1L1 LYRM5 2310026E23RIK SLC7A6OS MAT2B 4932438A13RIK LRRC8A SMO NUPL2
</code></pre>
<pre><code class="language-python"># Python
gget.enrichr(
	genes = [
		&quot;PHF14&quot;, &quot;RBM3&quot;, &quot;MSL1&quot;, &quot;PHF21A&quot;, &quot;ARL10&quot;, &quot;INSR&quot;, &quot;JADE2&quot;, &quot;P2RX7&quot;,
		&quot;LINC00662&quot;, &quot;CCDC101&quot;, &quot;PPM1B&quot;, &quot;KANSL1L&quot;, &quot;CRYZL1&quot;, &quot;ANAPC16&quot;, &quot;TMCC1&quot;,
		&quot;CDH8&quot;, &quot;RBM11&quot;, &quot;CNPY2&quot;, &quot;HSPA1L&quot;, &quot;CUL2&quot;, &quot;PLBD2&quot;, &quot;LARP7&quot;, &quot;TECPR2&quot;, 
		&quot;ZNF302&quot;, &quot;CUX1&quot;, &quot;MOB2&quot;, &quot;CYTH2&quot;, &quot;SEC22C&quot;, &quot;EIF4E3&quot;, &quot;ROBO2&quot;,
		&quot;ADAMTS9-AS2&quot;, &quot;CXXC1&quot;, &quot;LINC01314&quot;, &quot;ATF7&quot;, &quot;ATP5F1&quot;
	], 
	database = &quot;ChEA_2022&quot;,
	background_list = [
		&quot;NSUN3&quot;,&quot;POLRMT&quot;,&quot;NLRX1&quot;,&quot;SFXN5&quot;,&quot;ZC3H12C&quot;,&quot;SLC25A39&quot;,&quot;ARSG&quot;,
		&quot;DEFB29&quot;,&quot;PCMTD2&quot;,&quot;ACAA1A&quot;,&quot;LRRC1&quot;,&quot;2810432D09RIK&quot;,&quot;SEPHS2&quot;,
		&quot;SAC3D1&quot;,&quot;TMLHE&quot;,&quot;LOC623451&quot;,&quot;TSR2&quot;,&quot;PLEKHA7&quot;,&quot;GYS2&quot;,&quot;ARHGEF12&quot;,
		&quot;HIBCH&quot;,&quot;LYRM2&quot;,&quot;ZBTB44&quot;,&quot;ENTPD5&quot;,&quot;RAB11FIP2&quot;,&quot;LIPT1&quot;,
		&quot;INTU&quot;,&quot;ANXA13&quot;,&quot;KLF12&quot;,&quot;SAT2&quot;,&quot;GAL3ST2&quot;,&quot;VAMP8&quot;,&quot;FKBPL&quot;,
		&quot;AQP11&quot;,&quot;TRAP1&quot;,&quot;PMPCB&quot;,&quot;TM7SF3&quot;,&quot;RBM39&quot;,&quot;BRI3&quot;,&quot;KDR&quot;,&quot;ZFP748&quot;,
		&quot;NAP1L1&quot;,&quot;DHRS1&quot;,&quot;LRRC56&quot;,&quot;WDR20A&quot;,&quot;STXBP2&quot;,&quot;KLF1&quot;,&quot;UFC1&quot;,
		&quot;CCDC16&quot;,&quot;9230114K14RIK&quot;,&quot;RWDD3&quot;,&quot;2610528K11RIK&quot;,&quot;ACO1&quot;,
		&quot;CABLES1&quot;, &quot;LOC100047214&quot;,&quot;YARS2&quot;,&quot;LYPLA1&quot;,&quot;KALRN&quot;,&quot;GYK&quot;,
		&quot;ZFP787&quot;,&quot;ZFP655&quot;,&quot;RABEPK&quot;,&quot;ZFP650&quot;,&quot;4732466D17RIK&quot;,&quot;EXOSC4&quot;,
		&quot;WDR42A&quot;,&quot;GPHN&quot;,&quot;2610528J11RIK&quot;,&quot;1110003E01RIK&quot;,&quot;MDH1&quot;,&quot;1200014M14RIK&quot;,
		&quot;AW209491&quot;,&quot;MUT&quot;,&quot;1700123L14RIK&quot;,&quot;2610036D13RIK&quot;,
		&quot;PHF14&quot;, &quot;RBM3&quot;, &quot;MSL1&quot;, &quot;PHF21A&quot;, &quot;ARL10&quot;, &quot;INSR&quot;, &quot;JADE2&quot;, 
		&quot;P2RX7&quot;, &quot;LINC00662&quot;, &quot;CCDC101&quot;, &quot;PPM1B&quot;, &quot;KANSL1L&quot;, &quot;CRYZL1&quot;, 
		&quot;ANAPC16&quot;, &quot;TMCC1&quot;,&quot;CDH8&quot;, &quot;RBM11&quot;, &quot;CNPY2&quot;, &quot;HSPA1L&quot;, &quot;CUL2&quot;, 
		&quot;PLBD2&quot;, &quot;LARP7&quot;, &quot;TECPR2&quot;, &quot;ZNF302&quot;, &quot;CUX1&quot;, &quot;MOB2&quot;, &quot;CYTH2&quot;, 
		&quot;SEC22C&quot;, &quot;EIF4E3&quot;, &quot;ROBO2&quot;, &quot;ADAMTS9-AS2&quot;, &quot;CXXC1&quot;, &quot;LINC01314&quot;, &quot;ATF7&quot;, 
		&quot;ATP5F1&quot;&quot;COX15&quot;,&quot;TMEM30A&quot;,&quot;NSMCE4A&quot;,&quot;TM2D2&quot;,&quot;RHBDD3&quot;,&quot;ATXN2&quot;,&quot;NFS1&quot;,
		&quot;3110001I20RIK&quot;,&quot;BC038156&quot;,&quot;C330002I19RIK&quot;,&quot;ZFYVE20&quot;,&quot;POLI&quot;,&quot;TOMM70A&quot;,
		&quot;LOC100047782&quot;,&quot;2410012H22RIK&quot;,&quot;RILP&quot;,&quot;A230062G08RIK&quot;,
		&quot;PTTG1IP&quot;,&quot;RAB1&quot;,&quot;AFAP1L1&quot;, &quot;LYRM5&quot;,&quot;2310026E23RIK&quot;,
		&quot;SLC7A6OS&quot;,&quot;MAT2B&quot;,&quot;4932438A13RIK&quot;,&quot;LRRC8A&quot;,&quot;SMO&quot;,&quot;NUPL2&quot;
	],
	plot=True
)
</code></pre>
<p>‚Üí Returns hits of the input gene list given the background gene list from the transcription factor/target library <em>ChEA 2022</em>. In Python, <code>plot=True</code> returns a graphical overview of the results:</p>
<p><img src="https://github.com/pachterlab/gget/blob/main/figures/gget_enrichr_results_2.png?raw=true" alt="alt text" /></p>
<p><br/><br/></p>
<p><strong>Generate a KEGG pathway image with the genes from the enrichment analysis highlighted:</strong><br />
This feature is available thanks to a <a href="https://github.com/pachterlab/gget/pull/106">PR</a> by <a href="https://github.com/noriakis">Noriaki Sato</a>.</p>
<pre><code class="language-bash">gget enrichr -db pathway --kegg_out kegg.png --kegg_rank 1 ZBP1 IRF3 RIPK1
</code></pre>
<pre><code class="language-python"># Python
gget.enrichr([&quot;ZBP1&quot;, &quot;IRF3&quot;, &quot;RIPK1&quot;], database=&quot;pathway&quot;, kegg_out=&quot;kegg.png&quot;, kegg_rank=1)
</code></pre>
<p>‚Üí In addition to the standard <code>gget enrichr</code> output, the <code>kegg_out</code> argument saves an image with the genes from the enrichment analysis highlighted in the KEGG pathway:</p>
<p><img src="https://github.com/pachterlab/gget/assets/56094636/b0aa5a64-69d0-4a6a-85db-3e7baf9cb2a4" alt="kegg" /></p>
<p><br/><br/></p>
<p>The following example was submitted by <a href="https://github.com/DylanLawless">Dylan Lawless</a> via <a href="https://github.com/pachterlab/gget/pull/54">PR</a>:<br />
<strong>Use <code>gget enrichr</code> in R and create a similar plot using <a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a>.</strong><br />
NOTE the switch of axes compared to the Python plot.</p>
<pre><code class="language-r">system(&quot;pip install gget&quot;)
install.packages(&quot;reticulate&quot;)
library(reticulate)
gget &lt;- import(&quot;gget&quot;)

# Perform enrichment analysis on a list of genes
df &lt;- gget$enrichr(list(&quot;ACE2&quot;, &quot;AGT&quot;, &quot;AGTR1&quot;), database = &quot;ontology&quot;)

# Count number of overlapping genes
df$overlapping_genes_count &lt;- lapply(df$overlapping_genes, length) |&gt; as.numeric()

# Only keep the top 15 results
df &lt;- df[1:15, ]

# Plot
library(ggplot2)

df |&gt;
	ggplot() +
	geom_bar(aes(
		x = -log10(adj_p_val),
		y = reorder(path_name, -adj_p_val)
	),
	stat = &quot;identity&quot;,
  	fill = &quot;lightgrey&quot;,
  	width = 0.5,
	color = &quot;black&quot;) +
	geom_text(
		aes(
			y = path_name,
			x = (-log10(adj_p_val)),
			label = overlapping_genes_count
		),
		nudge_x = 0.75,
		show.legend = NA,
		color = &quot;red&quot;
	) +
  	geom_text(
		aes(
			y = Inf,
			x = Inf,
      			hjust = 1,
      			vjust = 1,
			label = &quot;# of overlapping genes&quot;
		),
		show.legend = NA,
		color = &quot;red&quot;
	) +
	geom_vline(linetype = &quot;dotted&quot;, linewidth = 1, xintercept = -log10(0.05)) +
	ylab(&quot;Pathway name&quot;) +
	xlab(&quot;-log10(adjusted P value)&quot;)
</code></pre>
<h4 id="more-examples-6"><a class="header" href="#more-examples-6"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-gpt-"><a class="header" href="#gget-gpt-">gget gpt üí¨</a></h2>
<p>Generates natural language text based on a given prompt using the <a href="https://openai.com/">OpenAI</a> API's 'openai.ChatCompletion.create' endpoint. 
This module, including its source code, documentation and unit tests, were partly written by OpenAI's Chat-GTP3.</p>
<p>NOTE:<br />
OpenAI API calls are only 'free' for the first three months after generating your OpenAI Account (OpenAI provides a $5 credit that expires).<br />
You can define a hard monthly billing limit (e.g. $1) <a href="https://platform.openai.com/account/billing/limits">here</a>.<br />
See their pricing and FAQ <a href="https://openai.com/pricing">here</a>.<br />
Get your OpenAI API key <a href="https://platform.openai.com/account/api-keys">here</a>.</p>
<p>Returns: A string containing the generated text.</p>
<p>Before using <code>gget gpt</code> for the first time, run <code>gget setup gpt</code> / <code>gget.setup(&quot;gpt&quot;)</code> once (also see <a href="en/setup.html"><code>gget setup</code></a>).</p>
<p><strong>Positional argument</strong><br />
<code>prompt</code><br />
The input prompt to generate text from (str).</p>
<p><code>api_key</code><br />
Your OpenAI API key (str) (<a href="https://platform.openai.com/account/api-keys">get your API key</a>).</p>
<p><strong>Optional arguments</strong><br />
<code>-m</code> <code>--model</code><br />
The name of the GPT model to use for generating the text (str). Default is &quot;gpt-3.5-turbo&quot;.<br />
See https://platform.openai.com/docs/models/gpt-4 for more information on the available models.</p>
<p><code>-temp</code> <code>--temperature</code><br />
Value between 0 and 2 that controls the level of randomness and creativity in the generated text (float).<br />
Higher values result in more creative and varied text. Default is 1.</p>
<p><code>-tp</code> <code>--top_p</code><br />
Controls the diversity of the generated text as an alternative to sampling with temperature (float).<br />
Higher values result in more diverse and unexpected text. Default is 1.<br />
Note: OpenAI recommends altering this or temperature but not both.</p>
<p><code>-s</code> <code>--stop</code><br />
A sequence of tokens to mark the end of the generated text (str). Default is None.</p>
<p><code>-mt</code> <code>--max_tokens</code><br />
Controls the maximum length of the generated text, in tokens (int). Default is 200.</p>
<p><code>-pp</code> <code>--presence_penalty</code><br />
Number between -2.0 and 2.0. Higher values result increase the model's likelihood to talk about new topics (float). Default is 0.</p>
<p><code>-fp</code> <code>--frequency_penalty</code><br />
Number between -2.0 and 2.0. Higher values decrease the model's likelihood to repeat the same line verbatim (float). Default is 0.</p>
<p><code>-lb</code> <code>--logit_bias</code><br />
A dictionary that specifies a bias towards certain tokens in the generated text (dict). Default is None.</p>
<p><code>-o</code> <code>--out</code><br />
If provided, saves the generated text to a file with the specified path (str). Default: Standard out.</p>
<h3 id="example-5"><a class="header" href="#example-5">Example</a></h3>
<pre><code class="language-bash">gget gpt &quot;How are you today GPT?&quot; your_api_token
</code></pre>
<pre><code class="language-python"># Python
print(gget.gpt(&quot;How are you today GPT?&quot;, &quot;your_api_token&quot;))
</code></pre>
<br>
<img width="725" alt="Screen Shot 2023-03-18 at 3 42 32 PM" src="https://user-images.githubusercontent.com/56094636/226143902-6fa2d0c7-7eea-4382-b1d2-df6c3f0d5fd5.png">
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-info-"><a class="header" href="#gget-info-">gget info üí°</a></h2>
<p>Fetch extensive gene and transcript metadata from <a href="https://www.ensembl.org/">Ensembl</a>, <a href="https://www.uniprot.org/">UniProt</a>, and <a href="https://www.ncbi.nlm.nih.gov/">NCBI</a> using Ensembl IDs.<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong><br />
<code>ens_ids</code><br />
One or more Ensembl IDs.</p>
<p><strong>Optional arguments</strong><br />
<code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.csv (or .json). Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-n</code> <code>--ncbi</code><br />
TURN OFF results from <a href="https://www.ncbi.nlm.nih.gov/">NCBI</a>.<br />
Python: <code>ncbi=False</code> prevents data retrieval from NCBI (default: True).</p>
<p><code>-u</code> <code>--uniprot</code><br />
TURN OFF results from <a href="https://www.uniprot.org/">UniProt</a>.<br />
Python: <code>uniprot=False</code> prevents data retrieval from UniProt (default: True).</p>
<p><code>-pdb</code> <code>--pdb</code><br />
INCLUDE <a href="https://www.ebi.ac.uk/pdbe/">PDB</a> IDs in output (might increase runtime).<br />
Python: <code>pdb=True</code> includes PDB IDs in the results (default: False).</p>
<p><code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed.</p>
<p><code>wrap_text</code><br />
Python only. <code>wrap_text=True</code> displays data frame with wrapped text for easy reading (default: False).</p>
<h3 id="example-6"><a class="header" href="#example-6">Example</a></h3>
<pre><code class="language-bash">gget info ENSG00000034713 ENSG00000104853 ENSG00000170296
</code></pre>
<pre><code class="language-python"># Python
gget.info([&quot;ENSG00000034713&quot;, &quot;ENSG00000104853&quot;, &quot;ENSG00000170296&quot;])
</code></pre>
<p>‚Üí Returns extensive information about each requested Ensembl ID:</p>
<div class="table-wrapper"><table><thead><tr><th></th><th>uniprot_id</th><th>ncbi_gene_id</th><th>primary_gene_name</th><th>synonyms</th><th>protein_names</th><th>ensembl_description</th><th>uniprot_description</th><th>ncbi_description</th><th>biotype</th><th>canonical_transcript</th><th>...</th></tr></thead><tbody>
<tr><td>ENSG00000034713</td><td>P60520</td><td>11345</td><td>GABARAPL2</td><td>[ATG8, ATG8C, FLC3A, GABARAPL2, GATE-16, GATE16, GEF-2, GEF2]</td><td>Gamma-aminobutyric acid receptor-associated protein like 2 (GABA(A) receptor-associated protein-like 2)...</td><td>GABA type A receptor associated protein like 2 [Source:HGNC Symbol;Acc:HGNC:13291]</td><td>FUNCTION: Ubiquitin-like modifier involved in intra- Golgi traffic (By similarity). Modulates intra-Golgi transport through coupling between NSF activity and ...</td><td>Enables ubiquitin protein ligase binding activity. Involved in negative regulation of proteasomal protein catabolic process and protein...</td><td>protein_coding</td><td>ENST00000037243.7</td><td>...</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>...</td></tr>
</tbody></table>
</div>
<h4 id="more-examples-7"><a class="header" href="#more-examples-7"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-muscle-"><a class="header" href="#gget-muscle-">gget muscle ü¶æ</a></h2>
<p>Align multiple nucleotide or amino acid sequences to each other using <a href="https://www.drive5.com/muscle/">Muscle5</a>.<br />
Return format: ClustalW formatted standard out or aligned FASTA (.afa).</p>
<p><strong>Positional argument</strong><br />
<code>fasta</code><br />
List of sequences or path to FASTA or .txt file containing the nucleotide or amino acid sequences to be aligned.</p>
<p><strong>Optional arguments</strong><br />
<code>-o</code> <code>--out</code><br />
Path to the aligned FASTA file the results will be saved in, e.g. path/to/directory/results.afa. Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-s5</code> <code>--super5</code><br />
Aligns input using the <a href="https://drive5.com/muscle5/Muscle5_SuppMat.pdf">Super5 algorithm</a> instead of the <a href="https://drive5.com/muscle5/Muscle5_SuppMat.pdf">Parallel Perturbed Probcons (PPP) algorithm</a> to decrease time and memory.<br />
Use for large inputs (a few hundred sequences).</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed. </p>
<h3 id="example-7"><a class="header" href="#example-7">Example</a></h3>
<pre><code class="language-bash">gget muscle MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS
</code></pre>
<pre><code class="language-python"># Python
gget.muscle([&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;, &quot;MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS&quot;])
</code></pre>
<pre><code class="language-bash">gget muscle fasta.fa
</code></pre>
<pre><code class="language-python"># Python
gget.muscle(&quot;fasta.fa&quot;)
</code></pre>
<p>‚Üí Returns an overview of the aligned sequences with ClustalW coloring. (To return an aligned FASTA (.afa) file, use <code>--out</code> argument (or <code>save=True</code> in Jupyter Lab/Google Colab).) In the above example, the 'fasta.fa' includes several sequences to be aligned (e.g. isoforms returned from <code>gget seq</code>). </p>
<p><img src="https://github.com/pachterlab/gget/blob/main/figures/example_muscle_return.png?raw=true" alt="alt text" /></p>
<p>You can also view aligned fasta files returned by <code>gget.muscle</code> using programs like <a href="https://github.com/arvestad/alv"><code>alv</code></a>, as shown below:</p>
<pre><code class="language-python"># Python
!pip install biopython
!pip install alv
from Bio import AlignIO
import alv

gget.muscle(&quot;fasta.fa&quot;, out=&quot;fasta_aligned.afa&quot;)
msa = AlignIO.read(&quot;fasta_aligned.afa&quot;, &quot;fasta&quot;)
alv.view(msa)
</code></pre>
<h4 id="more-examples-8"><a class="header" href="#more-examples-8"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-pdb-"><a class="header" href="#gget-pdb-">gget pdb üîÆ</a></h2>
<p>Query <a href="https://www.rcsb.org/">RCSB Protein Data Bank (PDB)</a> for the protein structure/metadata of a given PDB ID.<br />
Return format: Resource 'pdb' is returned in PDB format. All other resources are returned in JSON format.</p>
<p><strong>Positional argument</strong><br />
<code>pdb_id</code><br />
PDB ID to be queried, e.g. '7S7U'.</p>
<p><strong>Optional arguments</strong><br />
<code>-r</code> <code>--resource</code><br />
Defines type of information to be returned. One of the following:<br />
'pdb': Returns the protein structure in PDB format (default).<br />
'entry': Information about PDB structures at the top level of PDB structure hierarchical data organization.<br />
'pubmed': Get PubMed annotations (data integrated from PubMed) for a given entry's primary citation.<br />
'assembly': Information about PDB structures at the quaternary structure level.<br />
'branched_entity': Get branched entity description (define entity ID as 'identifier').<br />
'nonpolymer_entity': Get non-polymer entity data (define entity ID as 'identifier').<br />
'polymer_entity': Get polymer entity data (define entity ID as 'identifier').<br />
'uniprot': Get UniProt annotations for a given macromolecular entity (define entity ID as 'identifier').<br />
'branched_entity_instance': Get branched entity instance description (define chain ID as 'identifier').<br />
'polymer_entity_instance': Get polymer entity instance (a.k.a chain) data (define chain ID as 'identifier').<br />
'nonpolymer_entity_instance': Get non-polymer entity instance description (define chain ID as 'identifier').</p>
<p><code>-i</code> <code>--identifier</code><br />
Can be used to define assembly, entity or chain ID (default: None). Assembly/entity IDs are numbers (e.g. 1), and chain IDs are letters (e.g. 'A').</p>
<p><code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/7S7U.pdb or path/to/directory/7S7U_entry.json. Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<h3 id="examples-4"><a class="header" href="#examples-4">Examples</a></h3>
<pre><code class="language-bash">gget pdb 7S7U -o 7S7U.pdb
</code></pre>
<pre><code class="language-python"># Python
gget.pdb(&quot;7S7U&quot;, save=True)
</code></pre>
<p>‚Üí Saves the structure of 7S7U in PDB format as '7S7U.pdb' in the current working directory.</p>
<p><strong>Find PDB crystal structures for a comparative analysis of protein structure:</strong></p>
<pre><code class="language-bash"># Find PDB IDs associated with an Ensembl ID 
gget info ENSG00000130234

# Alternatively: Since many entries in the PDB do not have linked Ensembl IDs, 
# you will likely find more PDB entries by BLASTing the sequence agains the PDB.

# Get the amino acid sequence of a transcript from an Ensembl ID
gget seq --translate ENSG00000130234 -o gget_seq_results.fa

# BLAST an amino acid sequence to find similar structures in the PDB
gget blast --database pdbaa gget_seq_results.fa

# Get PDB files from the PDB IDs returned by gget blast for comparative analysis
gget pdb 7DQA -o 7DQA.pdb
gget pdb 7CT5 -o 7CT5.pdb
</code></pre>
<pre><code class="language-python"># Find PDB IDs associated with an Ensembl ID 
gget.info(&quot;ENSG00000130234&quot;)

# Alternatively: Since many entries in the PDB do not have linked Ensembl IDs, 
# you will likely find more PDB entries by BLASTing the sequence agains the PDB.

# Get the amino acid sequence of a transcript from an Ensembl ID
gget.seq(&quot;ENSG00000130234&quot;, translate=True, save=True)

# BLAST an amino acid sequence to find similar structures in the PDB
gget.blast(&quot;gget_seq_results.fa&quot;, database=&quot;pdbaa&quot;)

# Get PDB files from the PDB IDs returned by gget blast for comparative analysis
gget.pdb(&quot;7DQA&quot;, save=True)
gget.pdb(&quot;7CT5&quot;, save=True)
</code></pre>
<p>‚Üí The use case above exemplifies how to find PDB files for comparative analysis of protein structure starting with Ensembl IDs or amino acid sequences. The fetched PDB files can also be compared to predicted structures generated by <a href="en/alphafold.html"><code>gget alphafold</code></a>. PDB files can be viewed interactively in 3D <a href="https://rcsb.org/3d-view">online</a>, or using programs like <a href="https://pymol.org/">PyMOL</a> or <a href="https://www.blender.org/">Blender</a>. To compare two PDB files, you can use <a href="https://rcsb.org/alignment">this website</a>.</p>
<h4 id="more-examples-9"><a class="header" href="#more-examples-9"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-ref-"><a class="header" href="#gget-ref-">gget ref üìñ</a></h2>
<p>Fetch FTPs and their respective metadata (or use flag <code>ftp</code> to only return the links) for reference genomes and annotations from <a href="https://www.ensembl.org/">Ensembl</a> by species.<br />
Return format: dictionary/JSON.</p>
<p><strong>Positional argument</strong><br />
<code>species</code><br />
Species for which the FTPs will be fetched in the format genus_species, e.g. homo_sapiens.<br />
Supports all available vertebrate and invertebrate (plants, fungi, protists, and invertebrate metazoa) genomes from Ensembl, except bacteria.<br />
Note: Not required when using flags <code>--list_species</code> or <code>--list_iv_species</code>.<br />
Supported shortcuts: 'human', 'mouse'</p>
<p><strong>Optional arguments</strong><br />
<code>-w</code> <code>--which</code><br />
Defines which results to return. Default: 'all' -&gt; Returns all available results.<br />
Possible entries are one or a combination (as comma-separated list) of the following:<br />
'gtf'‚ÄÉ- Returns the annotation (GTF).<br />
'cdna'‚ÄÉ- Returns the trancriptome (cDNA).<br />
'dna'‚ÄÉ- Returns the genome (DNA).<br />
'cds'‚ÄÉ- Returns the coding sequences corresponding to Ensembl genes. (Does not contain UTR or intronic sequence.)<br />
'cdrna'‚ÄÉ- Returns transcript sequences corresponding to non-coding RNA genes (ncRNA).<br />
'pep'‚ÄÉ- Returns the protein translations of Ensembl genes.</p>
<p><code>-r</code> <code>--release</code><br />
Defines the Ensembl release number from which the files are fetched, e.g. 104. Default: latest Ensembl release.</p>
<p><code>-o</code> <code>--out</code><br />
Path to the JSON file the results will be saved in, e.g. path/to/directory/results.json. Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-l</code> <code>--list_species</code><br />
Lists all available vertebrate species. (Python: combine with <code>species=None</code>.)</p>
<p><code>-liv</code> <code>--list_iv_species</code><br />
Lists all available invertebrate species. (Python: combine with <code>species=None</code>.)</p>
<p><code>-ftp</code> <code>--ftp</code><br />
Returns only the requested FTP links.</p>
<p><code>-d</code> <code>--download</code><br />
Command-line only. Downloads the requested FTPs to the current directory (requires <a href="https://curl.se/docs/">curl</a> to be installed).</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed. </p>
<h3 id="examples-5"><a class="header" href="#examples-5">Examples</a></h3>
<p><strong>Use <code>gget ref</code> in combination with <a href="https://www.kallistobus.tools/kb_usage/kb_ref/">kallisto | bustools</a> to build a reference index:</strong></p>
<pre><code class="language-bash">kb ref -i INDEX -g T2G -f1 FASTA $(gget ref --ftp -w dna,gtf homo_sapiens)
</code></pre>
<p>‚Üí kb ref builds a reference index using the latest DNA and GTF files of species <strong>Homo sapiens</strong> passed to it by <code>gget ref</code>.</p>
<p><br/><br/></p>
<p><strong>List all available genomes from Ensembl release 103:</strong></p>
<pre><code class="language-bash">gget ref --list_species -r 103
</code></pre>
<pre><code class="language-python"># Python
gget.ref(species=None, list_species=True, release=103)
</code></pre>
<p>‚Üí Returns a list with all available genomes (checks if GTF and FASTAs are available) from Ensembl release 103.<br />
(If no release is specified, <code>gget ref</code> will always return information from the latest Ensembl release.)</p>
<p><br/><br/></p>
<p><strong>Get the genome reference for a specific species:</strong></p>
<pre><code class="language-bash">gget ref -w gtf,dna homo_sapiens
</code></pre>
<pre><code class="language-python"># Python
gget.ref(&quot;homo_sapiens&quot;, which=[&quot;gtf&quot;, &quot;dna&quot;])
</code></pre>
<p>‚Üí Returns a JSON with the latest human GTF and FASTA FTPs, and their respective metadata, in the format:</p>
<pre><code>{
    &quot;homo_sapiens&quot;: {
        &quot;annotation_gtf&quot;: {
            &quot;ftp&quot;: &quot;http://ftp.ensembl.org/pub/release-106/gtf/homo_sapiens/Homo_sapiens.GRCh38.106.gtf.gz&quot;,
            &quot;ensembl_release&quot;: 106,
            &quot;release_date&quot;: &quot;28-Feb-2022&quot;,
            &quot;release_time&quot;: &quot;23:27&quot;,
            &quot;bytes&quot;: &quot;51379459&quot;
        },
        &quot;genome_dna&quot;: {
            &quot;ftp&quot;: &quot;http://ftp.ensembl.org/pub/release-106/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz&quot;,
            &quot;ensembl_release&quot;: 106,
            &quot;release_date&quot;: &quot;21-Feb-2022&quot;,
            &quot;release_time&quot;: &quot;09:35&quot;,
            &quot;bytes&quot;: &quot;881211416&quot;
        }
    }
}
</code></pre>
<h4 id="more-examples-10"><a class="header" href="#more-examples-10"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-search-"><a class="header" href="#gget-search-">gget search üîé</a></h2>
<p>Fetch genes and transcripts from <a href="https://www.ensembl.org/">Ensembl</a> using free-form search terms.<br />
Results are matched based on the &quot;gene name&quot; and &quot;description&quot; sections in the Ensembl database. <code>gget</code> version &gt;= 0.27.9 also includes results that match the Ensembl &quot;synonym&quot; section.<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong><br />
<code>searchwords</code><br />
One or more free form search words, e.g. gaba nmda. (Note: Search is not case-sensitive.)</p>
<p><strong>Other required arguments</strong><br />
<code>-s</code> <code>--species</code><br />
Species or database to be searched.<br />
A species can be passed in the format 'genus_species', e.g. 'homo_sapiens' or 'arabidopsis_thaliana'.<br />
To pass a specific database, pass the name of the CORE database, e.g. 'mus_musculus_dba2j_core_105_1'.<br />
All available core databases can be found here:<br />
Vertebrates: <a href="http://ftp.ensembl.org/pub/current/mysql/">http://ftp.ensembl.org/pub/current/mysql/</a><br />
Invertebrates: <a href="http://ftp.ensemblgenomes.org/pub/current/">http://ftp.ensemblgenomes.org/pub/current/</a> + select kingdom + go to mysql/<br />
Supported shortcuts: 'human', 'mouse'.</p>
<p><strong>Optional arguments</strong><br />
<code>-r</code> <code>--release</code><br />
Defines the Ensembl release number from which the files are fetched, e.g. 104. Default: None -&gt; latest Ensembl release is used.<br />
Note: <em>Does not apply to invertebrate species</em> (you can pass a specific core database (which includes a release number) to the <code>species</code> argument instead).<br />
This argument is overwritten if a specific database (which includes a release number) is passed to the species argument.</p>
<p><code>-t</code> <code>--id_type</code><br />
'gene' (default) or 'transcript'<br />
Returns genes or transcripts, respectively.</p>
<p><code>-ao</code> <code>--andor</code><br />
'or' (default) or 'and'<br />
'or': Returns all genes that INCLUDE AT LEAST ONE of the searchwords in their name/description.<br />
'and': Returns only genes that INCLUDE ALL of the searchwords in their name/description.</p>
<p><code>-l</code> <code>--limit</code><br />
Limits the number of search results, e.g. 10. Default: None.</p>
<p><code>-o</code> <code>--out</code><br />
Path to the csv the results will be saved in, e.g. path/to/directory/results.csv (or .json). Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed. </p>
<p><code>wrap_text</code><br />
Python only. <code>wrap_text=True</code> displays data frame with wrapped text for easy reading (default: False).</p>
<h3 id="example-8"><a class="header" href="#example-8">Example</a></h3>
<pre><code class="language-bash">gget search -s human gaba gamma-aminobutyric
</code></pre>
<pre><code class="language-python"># Python
gget.search([&quot;gaba&quot;, &quot;gamma-aminobutyric&quot;], &quot;homo_sapiens&quot;)
</code></pre>
<p>‚Üí Returns all genes that contain at least one of the search words in their name or Ensembl/external reference description:</p>
<div class="table-wrapper"><table><thead><tr><th>ensembl_id</th><th>gene_name</th><th>ensembl_description</th><th>ext_ref_description</th><th>biotype</th><th>url</th></tr></thead><tbody>
<tr><td>ENSG00000034713</td><td>GABARAPL2</td><td>GABA type A receptor associated protein like 2 [Source:HGNC Symbol;Acc:HGNC:13291]</td><td>GABA type A receptor associated protein like 2</td><td>protein_coding</td><td>https://uswest.ensembl.org/homo_sapiens/Gene/Summary?g=ENSG00000034713</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td></tr>
</tbody></table>
</div>
<h4 id="more-examples-11"><a class="header" href="#more-examples-11"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-setup-"><a class="header" href="#gget-setup-">gget setup üîß</a></h2>
<p>Function to install/download third-party dependencies for a specified gget module.</p>
<p><strong>Positional argument</strong><br />
<code>module</code><br />
gget module for which dependencies should be installed.</p>
<p><strong>Optional arguments</strong><br />
<code>-o</code> <code>--out</code><br />
Path to the folder downloaded files will be saved in (currently only applies to module = 'elm').<br />
NOTE: Do NOT use this argument when downloading the files for use with <code>gget.elm</code>.<br />
Default: None (downloaded files are saved inside the <code>gget</code> package installation folder).</p>
<p><strong>Flags</strong><br />
<code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed. </p>
<h3 id="example-9"><a class="header" href="#example-9">Example</a></h3>
<pre><code class="language-bash">gget setup alphafold
</code></pre>
<pre><code class="language-python"># Python
gget.setup(&quot;alphafold&quot;)
</code></pre>
<p>‚Üí Installs all (modified) third-party dependencies and downloads model parameters (~4GB) required to run <a href="en/alphafold.html"><code>gget alphafold</code></a>. </p>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-seq-"><a class="header" href="#gget-seq-">gget seq üß¨</a></h2>
<p>Fetch nucleotide or amino acid sequence(s) of a gene (and all its isoforms) or a transcript by Ensembl ID.<br />
Return format: FASTA.</p>
<p><strong>Positional argument</strong><br />
<code>ens_ids</code><br />
One or more Ensembl IDs.</p>
<p><strong>Optional arguments</strong><br />
<code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.fa. Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-t</code> <code>--translate</code><br />
Returns amino acid (instead of nucleotide) sequences.<br />
Nucleotide sequences are fetched from <a href="https://www.ensembl.org/">Ensembl</a>.<br />
Amino acid sequences are fetched from <a href="https://www.uniprot.org/">UniProt</a>.</p>
<p><code>-iso</code> <code>--isoforms</code><br />
Returns the sequences of all known transcripts.<br />
(Only for gene IDs.)</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed.</p>
<h3 id="examples-6"><a class="header" href="#examples-6">Examples</a></h3>
<pre><code class="language-bash">gget seq ENSG00000034713 ENSG00000104853 ENSG00000170296
</code></pre>
<pre><code class="language-python"># Python
gget.seq([&quot;ENSG00000034713&quot;, &quot;ENSG00000104853&quot;, &quot;ENSG00000170296&quot;])
</code></pre>
<p>‚Üí Returns the nucleotide sequences of ENSG00000034713, ENSG00000104853, and ENSG00000170296 in FASTA format.</p>
<pre><code class="language-bash">gget seq -t -iso ENSG00000034713
</code></pre>
<pre><code class="language-python"># Python
gget.seq(&quot;ENSG00000034713&quot;, translate=True, isoforms=True)
</code></pre>
<p>‚Üí Returns the amino acid sequences of all known transcripts of ENSG00000034713 in FASTA format.</p>
<h4 id="more-examples-12"><a class="header" href="#more-examples-12"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><h1 id="welcome-to-ggets-contributing-guide"><a class="header" href="#welcome-to-ggets-contributing-guide">Welcome to gget's contributing guide</a></h1>
<p>Thank you for investing your time in contributing to our project! Any contribution you make will be reflected on the <a href="https://github.com/pachterlab/gget">gget repo</a>. ‚ú®</p>
<p>Read our <a href="en/./code_of_conduct.html">Code of Conduct</a> to keep our community approachable and respectable.</p>
<p>In this guide you will get an overview of the contribution workflow from opening an issue or creating a pull request (PR) to reviewing and merging a PR.</p>
<h2 id="issues"><a class="header" href="#issues">Issues</a></h2>
<h3 id="create-a-new-issue"><a class="header" href="#create-a-new-issue">Create a new issue</a></h3>
<p>If you spot a problem with gget or you have an idea for a new feature, <a href="https://github.com/pachterlab/gget/issues">check if an issue already exists</a>. If a related issue doesn't exist, you can open a new issue using the relevant <a href="https://github.com/pachterlab/gget/issues/new/choose">issue form</a>.</p>
<h3 id="solve-an-issue"><a class="header" href="#solve-an-issue">Solve an issue</a></h3>
<p>Scan through our <a href="https://github.com/pachterlab/gget/issues">existing issues</a> to find one that interests you. You can narrow down the search using <code>labels</code> as filters. If you find an issue to work on, you are welcome to open a PR with a fix.</p>
<h2 id="contribute-through-pull-requests"><a class="header" href="#contribute-through-pull-requests">Contribute through pull requests</a></h2>
<h3 id="getting-started"><a class="header" href="#getting-started">Getting started</a></h3>
<ol>
<li>Fork the repository.</li>
</ol>
<ul>
<li>
<p>Using GitHub Desktop:</p>
<ul>
<li><a href="https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/getting-started-with-github-desktop">Getting started with GitHub Desktop</a> will guide you through setting up Desktop.</li>
<li>Once Desktop is set up, you can use it to <a href="https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/cloning-and-forking-repositories-from-github-desktop">fork the repo</a>!</li>
</ul>
</li>
<li>
<p>Using the command line:</p>
<ul>
<li><a href="https://docs.github.com/en/github/getting-started-with-github/fork-a-repo#fork-an-example-repository">Fork the repo</a> so that you can make your changes without affecting the original project until you're ready to merge them.</li>
</ul>
</li>
</ul>
<ol start="2">
<li>Create a working branch and start with your changes!</li>
</ol>
<h3 id="commit-your-update"><a class="header" href="#commit-your-update">Commit your update</a></h3>
<p>Commit the changes once you are happy with them.</p>
<h3 id="-self-review-the-following-before-creating-a-pull-request-"><a class="header" href="#-self-review-the-following-before-creating-a-pull-request-">‚ÄºÔ∏è Self-review the following before creating a Pull Request ‚ÄºÔ∏è</a></h3>
<ol>
<li>Review the content for technical accuracy.</li>
<li>Copy-edit the changes/comments for grammar, spelling, and adherence to the general style of existing gget code.</li>
<li>Format your code using <a href="https://black.readthedocs.io/en/stable/getting_started.html">black</a>.</li>
<li>Make sure the unit tests pass:
<ul>
<li>Developer dependencies can be installed with <code>pip install -r dev-requirements.txt</code></li>
<li>Run existing unit tests from the gget repository root with <code>coverage run -m pytest -ra -v tests &amp;&amp; coverage report --omit=main.py,tests*</code></li>
</ul>
</li>
<li>Add new unit tests if applicable:
<ul>
<li>Arguments and expected results are stored in json files in ./tests/fixtures/</li>
<li>Unit tests can be added to ./tests/test_*.py and will be automatically detected</li>
</ul>
</li>
<li>Make sure the edits are compatible with both the Python and the command line interface
<ul>
<li>The command line interface and arguments are defined in ./gget/main.py</li>
</ul>
</li>
<li>Add new modules/arguments to the documentation if applicable:
<ul>
<li>The manual for each module can be edited/added as ./docs/src/*.md</li>
</ul>
</li>
</ol>
<p>If you have any questions, feel free to start a <a href="https://github.com/pachterlab/gget/discussions">discussion</a> or create an issue as described above.</p>
<h3 id="pull-request"><a class="header" href="#pull-request">Pull Request</a></h3>
<p>When you're finished with the changes, <a href="https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request">create a pull request</a>, also known as a PR.</p>
<p>‚ÄºÔ∏è Please make all PRs against the <code>dev</code> branch of the gget repository. </p>
<ul>
<li>Don't forget to <a href="https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue">link PR to issue</a> if you are solving one.</li>
<li>Enable the checkbox to <a href="https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork">allow maintainer edits</a> so the branch can be updated for a merge.</li>
<li>If you run into any merge issues, checkout this <a href="https://github.com/skills/resolve-merge-conflicts">git tutorial</a> to help you resolve merge conflicts and other issues.</li>
</ul>
<p>Once you submit your PR, a gget team member will review your proposal. We may ask questions or request additional information.</p>
<h3 id="your-pr-is-merged"><a class="header" href="#your-pr-is-merged">Your PR is merged!</a></h3>
<p>Congratulations! üéâ	 The gget team thanks you. ‚ú®</p>
<p>Once your PR is merged, your contributions will be publicly visible on the <a href="https://github.com/pachterlab/gget">gget repo</a>.</p>
<div style="break-before: page; page-break-before: always;"></div><h1 id="contributor-covenant-code-of-conduct"><a class="header" href="#contributor-covenant-code-of-conduct">Contributor Covenant Code of Conduct</a></h1>
<h2 id="our-pledge"><a class="header" href="#our-pledge">Our Pledge</a></h2>
<p>We as members, contributors, and leaders pledge to make participation in our
community a harassment-free experience for everyone, regardless of age, body
size, visible or invisible disability, ethnicity, sex characteristics, gender
identity and expression, level of experience, education, socio-economic status,
nationality, personal appearance, race, religion, or sexual identity
and orientation.</p>
<p>We pledge to act and interact in ways that contribute to an open, welcoming,
diverse, inclusive, and healthy community.</p>
<h2 id="our-standards"><a class="header" href="#our-standards">Our Standards</a></h2>
<p>Examples of behavior that contributes to a positive environment for our
community include:</p>
<ul>
<li>Demonstrating empathy and kindness toward other people</li>
<li>Being respectful of differing opinions, viewpoints, and experiences</li>
<li>Giving and gracefully accepting constructive feedback</li>
<li>Accepting responsibility and apologizing to those affected by our mistakes,
and learning from the experience</li>
<li>Focusing on what is best not just for us as individuals, but for the
overall community</li>
</ul>
<p>Examples of unacceptable behavior include:</p>
<ul>
<li>The use of sexualized language or imagery, and sexual attention or
advances of any kind</li>
<li>Trolling, insulting or derogatory comments, and personal or political attacks</li>
<li>Public or private harassment</li>
<li>Publishing others' private information, such as a physical or email
address, without their explicit permission</li>
<li>Other conduct which could reasonably be considered inappropriate in a
professional setting</li>
</ul>
<h2 id="enforcement-responsibilities"><a class="header" href="#enforcement-responsibilities">Enforcement Responsibilities</a></h2>
<p>Community leaders are responsible for clarifying and enforcing our standards of
acceptable behavior and will take appropriate and fair corrective action in
response to any behavior that they deem inappropriate, threatening, offensive,
or harmful.</p>
<p>Community leaders have the right and responsibility to remove, edit, or reject
comments, commits, code, wiki edits, issues, and other contributions that are
not aligned to this Code of Conduct, and will communicate reasons for moderation
decisions when appropriate.</p>
<h2 id="scope"><a class="header" href="#scope">Scope</a></h2>
<p>This Code of Conduct applies within all community spaces, and also applies when
an individual is officially representing the community in public spaces.
Examples of representing our community include using an official e-mail address,
posting via an official social media account, or acting as an appointed
representative at an online or offline event.</p>
<h2 id="enforcement"><a class="header" href="#enforcement">Enforcement</a></h2>
<p>Instances of abusive, harassing, or otherwise unacceptable behavior may be
reported to the community leaders responsible for enforcement at
lpachter@caltech.edu.
All complaints will be reviewed and investigated promptly and fairly.</p>
<p>All community leaders are obligated to respect the privacy and security of the
reporter of any incident.</p>
<h2 id="enforcement-guidelines"><a class="header" href="#enforcement-guidelines">Enforcement Guidelines</a></h2>
<p>Community leaders will follow these Community Impact Guidelines in determining
the consequences for any action they deem in violation of this Code of Conduct:</p>
<h3 id="1-correction"><a class="header" href="#1-correction">1. Correction</a></h3>
<p><strong>Community Impact</strong>: Use of inappropriate language or other behavior deemed
unprofessional or unwelcome in the community.</p>
<p><strong>Consequence</strong>: A private, written warning from community leaders, providing
clarity around the nature of the violation and an explanation of why the
behavior was inappropriate. A public apology may be requested.</p>
<h3 id="2-warning"><a class="header" href="#2-warning">2. Warning</a></h3>
<p><strong>Community Impact</strong>: A violation through a single incident or series
of actions.</p>
<p><strong>Consequence</strong>: A warning with consequences for continued behavior. No
interaction with the people involved, including unsolicited interaction with
those enforcing the Code of Conduct, for a specified period of time. This
includes avoiding interactions in community spaces as well as external channels
like social media. Violating these terms may lead to a temporary or
permanent ban.</p>
<h3 id="3-temporary-ban"><a class="header" href="#3-temporary-ban">3. Temporary Ban</a></h3>
<p><strong>Community Impact</strong>: A serious violation of community standards, including
sustained inappropriate behavior.</p>
<p><strong>Consequence</strong>: A temporary ban from any sort of interaction or public
communication with the community for a specified period of time. No public or
private interaction with the people involved, including unsolicited interaction
with those enforcing the Code of Conduct, is allowed during this period.
Violating these terms may lead to a permanent ban.</p>
<h3 id="4-permanent-ban"><a class="header" href="#4-permanent-ban">4. Permanent Ban</a></h3>
<p><strong>Community Impact</strong>: Demonstrating a pattern of violation of community
standards, including sustained inappropriate behavior,  harassment of an
individual, or aggression toward or disparagement of classes of individuals.</p>
<p><strong>Consequence</strong>: A permanent ban from any sort of public interaction within
the community.</p>
<h2 id="attribution"><a class="header" href="#attribution">Attribution</a></h2>
<p>This Code of Conduct is adapted from the <a href="https://www.contributor-covenant.org">Contributor Covenant</a>,
version 2.0, available at <a href="https://www.contributor-covenant.org/version/2/0/code_of_conduct.html">https://www.contributor-covenant.org/version/2/0/code_of_conduct.html</a>.</p>
<p>Community Impact Guidelines were inspired by <a href="https://github.com/mozilla/diversity">Mozilla's code of conduct
enforcement ladder</a>.</p>
<p>For answers to common questions about this code of conduct, see the FAQ at
<a href="https://www.contributor-covenant.org/faq">https://www.contributor-covenant.org/faq</a>. Translations are available at
<a href="https://www.contributor-covenant.org/translations">https://www.contributor-covenant.org/translations</a>.</p>
<div style="break-before: page; page-break-before: always;"></div><p><a href="en/LICENSE"><img src="https://img.shields.io/pypi/l/gget" alt="license" /></a></p>
<h1 id="citation"><a class="header" href="#citation">Citation</a></h1>
<p>If you use <code>gget</code> in a publication, please cite:<br />
Luebbert, L., &amp; Pachter, L. (2023). Efficient querying of genomic reference databases with gget. Bioinformatics. https://doi.org/10.1093/bioinformatics/btac836</p>
<ul>
<li>
<p>If using <code>gget alphafold</code>, please also cite:</p>
<ul>
<li>Jumper, J., Evans, R., Pritzel, A. et al. Highly accurate protein structure prediction with AlphaFold. Nature 596, 583‚Äì589 (2021). https://doi.org/10.1038/s41586-021-03819-2</li>
</ul>
<p>And, if applicable:</p>
<ul>
<li>Evans, R. et al. Protein complex prediction with AlphaFold-Multimer. bioRxiv 2021.10.04.463034; https://doi.org/10.1101/2021.10.04.463034</li>
</ul>
</li>
<li>
<p>If using <code>gget archs4</code>, please also cite:</p>
<ul>
<li>
<p>Lachmann A, Torre D, Keenan AB, Jagodnik KM, Lee HJ, Wang L, Silverstein MC, Ma‚Äôayan A. Massive mining of publicly available RNA-seq data from human and mouse. Nature Communications 9. Article number: 1366 (2018), doi:10.1038/s41467-018-03751-6</p>
</li>
<li>
<p>Bray NL, Pimentel H, Melsted P and Pachter L, Near optimal probabilistic RNA-seq quantification, Nature Biotechnology 34, p 525--527 (2016). https://doi.org/10.1038/nbt.3519</p>
</li>
</ul>
</li>
<li>
<p>If using <code>gget blast</code>, please also cite:</p>
<ul>
<li>Altschul SF, Gish W, Miller W, Myers EW, Lipman DJ. Basic local alignment search tool. J Mol Biol. 1990 Oct 5;215(3):403-10. doi: 10.1016/S0022-2836(05)80360-2. PMID: 2231712.</li>
</ul>
</li>
<li>
<p>If using <code>gget blat</code>, please also cite:</p>
<ul>
<li>Kent WJ. BLAT--the BLAST-like alignment tool. Genome Res. 2002 Apr;12(4):656-64. doi: 10.1101/gr.229202. PMID: 11932250; PMCID: PMC187518.</li>
</ul>
</li>
<li>
<p>If using <code>gget cellxgene</code>, please also cite:</p>
<ul>
<li>Chanzuckerberg Initiative. (n.d.). CZ CELLxGENE Discover. Retrieved [insert date here], from https://cellxgene.cziscience.com/</li>
</ul>
</li>
<li>
<p>If using <code>gget diamond</code>, please also cite:</p>
<ul>
<li>Buchfink, B., Xie, C. &amp; Huson, D. Fast and sensitive protein alignment using DIAMOND. Nat Methods 12, 59‚Äì60 (2015). https://doi.org/10.1038/nmeth.3176</li>
</ul>
</li>
<li>
<p>If using <code>gget elm</code>, please also cite:</p>
<ul>
<li>Manjeet Kumar, Sushama Michael, Jes√∫s Alvarado-Valverde, B√°lint M√©sz√°ros, Hugo S√°mano‚ÄêS√°nchez, Andr√°s Zeke, Laszlo Dobson, Tamas Lazar, Mihkel √ñrd, Anurag Nagpal, Nazanin Farahi, Melanie K√§ser, Ramya Kraleti, Norman E Davey, Rita Pancsa, Luc√≠a B Chemes, Toby J Gibson, The Eukaryotic Linear Motif resource: 2022 release, Nucleic Acids Research, Volume 50, Issue D1, 7 January 2022, Pages D497‚ÄìD508, https://doi.org/10.1093/nar/gkab975</li>
</ul>
</li>
<li>
<p>If using <code>gget enrichr</code>, please also cite:</p>
<ul>
<li>
<p>Chen EY, Tan CM, Kou Y, Duan Q, Wang Z, Meirelles GV, Clark NR, Ma'ayan A.
Enrichr: interactive and collaborative HTML5 gene list enrichment analysis tool. BMC Bioinformatics. 2013; 128(14).</p>
</li>
<li>
<p>Kuleshov MV, Jones MR, Rouillard AD, Fernandez NF, Duan Q, Wang Z, Koplev S, Jenkins SL, Jagodnik KM, Lachmann A, McDermott MG, Monteiro CD, Gundersen GW, Ma'ayan A.
Enrichr: a comprehensive gene set enrichment analysis web server 2016 update. Nucleic Acids Research. 2016; gkw377.</p>
</li>
<li>
<p>Xie Z, Bailey A, Kuleshov MV, Clarke DJB., Evangelista JE, Jenkins SL, Lachmann A, Wojciechowicz ML, Kropiwnicki E, Jagodnik KM, Jeon M, &amp; Ma‚Äôayan A.
Gene set knowledge discovery with Enrichr. Current Protocols, 1, e90. 2021. doi: 10.1002/cpz1.90. </p>
</li>
</ul>
</li>
<li>
<p>If using <code>gget muscle</code>, please also cite:</p>
<ul>
<li>Edgar RC (2021), MUSCLE v5 enables improved estimates of phylogenetic tree confidence by ensemble bootstrapping, bioRxiv 2021.06.20.449169. https://doi.org/10.1101/2021.06.20.449169.</li>
</ul>
</li>
</ul>
<hr />
<h1 id="disclaimer"><a class="header" href="#disclaimer">Disclaimer</a></h1>
<p><code>gget</code> is only as accurate as the databases/servers/APIs it queries from. The accuracy or reliability of the data is not guaranteed or warranted in any way and the providers disclaim liability of any kind whatsoever, including, without limitation, liability for quality, performance, merchantability and fitness for a particular purpose arising out of the use, or inability to use the data.</p>
<div style="break-before: page; page-break-before: always;"></div><p><a href="https://pypi.org/project/gget"><img src="https://img.shields.io/pypi/v/gget" alt="pypi version" /></a>
<a href="https://anaconda.org/bioconda/gget"><img src="https://anaconda.org/bioconda/gget/badges/version.svg" alt="image" /></a>
<a href="https://pepy.tech/project/gget"><img src="https://static.pepy.tech/personalized-badge/gget?period=total&amp;units=international_system&amp;left_color=grey&amp;right_color=brightgreen&amp;left_text=Downloads" alt="Downloads" /></a>
<a href="https://anaconda.org/bioconda/gget"><img src="https://img.shields.io/conda/dn/bioconda/gget?logo=Anaconda" alt="Conda" /></a>
<a href="es/LICENSE"><img src="https://img.shields.io/pypi/l/gget" alt="license" /></a>
<img src="https://github.com/pachterlab/gget/actions/workflows/ci.yml/badge.svg" alt="status" />
<img src="https://github.com/lauraluebbert/test_gget_alphafold/actions/workflows/CI_alphafold.yml/badge.svg" alt="status" />
<a href="https://github.com/pachterlab/gget/"><img src="https://img.shields.io/github/stars/pachterlab/gget.svg?style=social" alt="Star on GitHub" /></a></p>
<p><a href="https://raw.githubusercontent.com/pachterlab/gget/main/figures/gget_overview.png"><img align="right" width="50%" height="50%" src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_overview.png?raw=true" /></a></p>
<h1 id="bienvenidos"><a class="header" href="#bienvenidos">¬°Bienvenidos!</a></h1>
<p><code>gget</code> es un programa gratuito de c√≥digo fuente abierta de Terminal y Python que permite la consulta eficiente de bases de datos gen√≥micas.<br />
<br>
<code>gget</code> consiste en un conjunto de m√≥dulos separados pero interoperables, cada uno dise√±ado para facilitar un tipo de consulta de base de datos en una sola l√≠nea de c√≥digo.
<br>
<br>
<br>
<br></p>
<p><a href="es/alphafold.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_alphafold_es.png?raw=true" width="32%" height="32%" /></a>
<a href="es/archs4.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_archs4_es.png?raw=true" width="32%" height="32%" /></a>
<a href="es/blast.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_blast_es.png?raw=true" width="32%" height="32%" /></a></p>
<p><a href="es/blat.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_blat_es.png?raw=true" width="32%" height="32%" /></a>
<a href="es/cellxgene.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_cellxgene_es.png?raw=true" width="32%" height="32%" /></a>
<a href="es/enrichr.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_enrichr_es.png?raw=true" width="32%" height="32%" /></a></p>
<p><a href="es/info.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_info_es.png?raw=true" width="32%" height="32%" /></a>
<a href="es/muscle.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_muscle_es.png?raw=true" width="32%" height="32%" /></a>
<a href="es/pdb.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_pdb_es.png?raw=true" width="32%" height="32%" /></a></p>
<p><a href="es/ref.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_ref_es.png?raw=true" width="32%" height="32%" /></a>
<a href="es/search.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_search_es.png?raw=true" width="32%" height="32%" /></a>
<a href="es/seq.html"><img src="https://github.com/pachterlab/gget/blob/main/docs/assets/website_v2_gget_seq_es.png?raw=true" width="32%" height="32%" /></a> </p>
<h3 id="m√°s-tutoriales"><a class="header" href="#m√°s-tutoriales"><a href="https://github.com/pachterlab/gget_examples">M√°s tutoriales</a></a></h3>
<br>  
<p>Si usa <code>gget</code> en una publicaci√≥n, por favor <a href="es/cite.html">cite*</a>:</p>
<pre><code>Luebbert, L., &amp; Pachter, L. (2023). Efficient querying of genomic reference databases with gget. Bioinformatics. https://doi.org/10.1093/bioinformatics/btac836
</code></pre>
<p>Lea el art√≠culo aqu√≠: <a href="https://doi.org/10.1093/bioinformatics/btac836">https://doi.org/10.1093/bioinformatics/btac836</a></p>
<p>Gracias a <a href="https://github.com/victorg775">Victor Garcia-Ruiz</a> y <a href="https://www.linkedin.com/in/akorta/">Anna Karen Orta</a> por su ayuda con la traduccion del sitio web.</p>
<div style="break-before: page; page-break-before: always;"></div><h2 id="-lo-m√°s-reciente"><a class="header" href="#-lo-m√°s-reciente">‚ú® ¬°Lo m√°s reciente!</a></h2>
<p><strong>Versi√≥n ‚â• 0.28.3</strong> (22 de enero de 2024):</p>
<ul>
<li><strong><a href="es/./search.html"><code>gget search</code></a> y <a href="es/./ref.html"><code>gget ref</code></a> ahora tambi√©n admiten hongos üçÑ, protistas üåù y metazoos de invertebrados üêù üêú üêå üêô (adem√°s de vertebrados y plantas)</strong></li>
<li><strong>Nuevo m√≥dulo: <a href="es/./cosmic.html"><code>gget cosmic</code></a></strong></li>
<li><a href="es/./enrichr.html"><code>gget enrichr</code></a>: corrige puntos de dispersi√≥n duplicados en el gr√°fico cuando los nombres de las rutas est√°n duplicados</li>
<li><a href="es/./elm.html"><code>gget elm</code></a>:
<ul>
<li>Se cambi√≥ el nombre de la columna de resultados orto 'Ortholog_UniProt_ID' a 'Ortholog_UniProt_Acc' para reflejar correctamente el contenido de la columna, que son accesos de UniProt. 'UniProt ID' se cambi√≥ a 'UniProt Acc' en la documentaci√≥n para todos los m√≥dulos <code>gget</code>.</li>
<li>Se cambi√≥ el nombre de la columna de resultados ortogonales 'motif_in_query' a 'motif_inside_subject_query_overlap'.</li>
<li>Se agreg√≥ informaci√≥n del dominio de interacci√≥n a los resultados (nuevas columnas: &quot;InteractionDomainId&quot;, &quot;InteractionDomainDescription&quot;, &quot;InteractionDomainName&quot;).</li>
<li>La cadena de expresiones regulares para coincidencias de expresiones regulares se encapsul√≥ de la siguiente manera: &quot;(?=(regex))&quot; (en lugar de pasar directamente la cadena de expresiones regulares &quot;regex&quot;) para permitir capturar todas las apariciones de un motivo cuando la longitud del motivo es variable y hay son repeticiones en la secuencia (<a href="https://regex101.com/r/HUWLlZ/1">https://regex101.com/r/HUWLlZ/1</a>).</li>
</ul>
</li>
<li><a href="es/./setup.html"><code>gget setup</code></a>: utilice el argumento <code>out</code> para especificar un directorio en el que se descargar√° la base de datos ELM. Completa <a href="https://github.com/pachterlab/gget/issues/119">esta solicitud de funci√≥n</a>.</li>
<li><a href="es/./diamond.html"><code>gget Diamond</code></a>: El comando DIAMOND ahora se ejecuta con el indicador <code>--ignore-warnings</code>, lo que permite secuencias de nicho, como secuencias de amino√°cidos que solo contienen caracteres de nucle√≥tidos y secuencias repetidas. Esto tambi√©n es v√°lido para las alineaciones DIAMOND realizadas dentro de <a href="es/./elm.html"><code>gget elm</code></a>.</li>
<li>Cambio de back-end de <a href="es/./ref.html"><code>gget ref</code></a> y <a href="es/./search.html"><code>gget search</code></a>: la versi√≥n actual de Ensembl se obtiene del nuevo <a href="https://ftp.ensembl.org/pub/VERSION">archivo de versi√≥n</a> en el sitio FTP de Ensembl para evitar errores durante la carga de nuevos lanzamientos.</li>
<li><a href="es/./search.html"><code>gget search</code></a>:
<ul>
<li>Los resultados del enlace FTP (<code>--ftp</code>) se guardan en formato de archivo txt en lugar de json.</li>
<li>Se corrigieron enlaces URL al resumen de genes de Ensembl para especies con un nombre de subespecie e invertebrados.</li>
</ul>
</li>
<li><a href="es/./ref.html"><code>gget ref</code></a>:
<ul>
<li>Cambios de back-end para aumentar la velocidad.</li>
<li>Nuevo argumento: <code>list_iv_species</code> para enumerar todas las especies de invertebrados disponibles (se puede combinar con el argumento <code>release</code> para obtener todas las especies disponibles de una liberaci√≥n espec√≠fica de Ensembl)</li>
</ul>
</li>
</ul>
<p><strong>Versi√≥n ‚â• 0.28.2</strong> (15 de noviembre de 2023):</p>
<ul>
<li><a href="es/./info.html"><code>gget info</code></a>: devuelve un mensaje de error cuando el servidor NCBI falla por un motivo distinto a un error de recuperaci√≥n (esto es un error en el lado del servidor en lugar de un error con <code>gget</code>)</li>
<li>Reemplace el argumento obsoleto 'texto' para los m√©todos de tipo find() siempre que se usen con la dependencia <code>BeautifulSoup</code></li>
<li><a href="es/elm.html"><code>gget elm</code></a>: Elimina instancias de falsos positivos y verdaderos negativos de los resultados devueltos.</li>
<li><a href="es/elm.html"><code>gget elm</code></a>: agrega el argumento <code>expand</code></li>
</ul>
<p><strong>Versi√≥n ‚â• 0.28.0</strong> (5 de noviembre de 2023):</p>
<ul>
<li>Documentaci√≥n actualizada de <a href="es/./muscle.html"><code>gget muscle</code></a> para agregar un tutorial sobre c√≥mo visualizar secuencias con diferentes longitudes de nombres de secuencia + ligero cambio en la visualizaci√≥n devuelta para que sea un poco m√°s s√≥lida ante diferentes nombres de secuencia</li>
<li><a href="es/./muscle.html"><code>gget muscle</code></a> ahora tambi√©n permite una lista de secuencias como entrada (como alternativa a proporcionar la ruta a un archivo FASTA)</li>
<li>Permitir filtro de genes faltante para <a href="es/cellxgene.html"><code>gget cellxgene</code></a> (corrige <a href="https://github.com/pachterlab/gget/issues/110">error</a>)</li>
<li><a href="es/./seq.html"><code>gget seq</code></a>: permite nombres de genes faltantes (correccione [https://github.com/pachterlab/gget/issues/107](https://github.com/pachterlab/gget /n√∫meros/107))</li>
<li>Nuevos argumentos para <a href="es/enrichr.html"><code>gget enrichr</code></a>: use el argumento <code>kegg_out</code> y <code>kegg_rank</code> para crear una imagen de la v√≠a KEGG con los genes del an√°lisis de enriquecimiento resaltados (gracias a [este PR](https ://github.com/pachterlab/gget/pull/106) por <a href="https://github.com/noriakis">Noriaki Sato</a>)</li>
<li>Nuevos m√≥dulos: <a href="es/elm.html"><code>gget elm</code></a> y <a href="es/diamond.html"><code>gget Diamond</code></a></li>
</ul>
<p><strong>Versi√≥n ‚â• 0.27.9</strong> (7 de agosto de 2023):</p>
<ul>
<li>Nuevos argumentos para <a href="es/enrichr.html"><code>gget enrichr</code></a>: use el argumento <code>background_list</code> para proporcionar una lista de genes 'background'</li>
<li><a href="es/search.html"><code>gget search</code></a> ahora tambi√©n busca sin√≥nimos <a href="https://ensembl.org/">Ensembl</a> (adem√°s de nombres y descripciones de genes) para obtener resultados de b√∫squeda m√°s completos (gracias a <a href="https://github.com/KleinSamuel">Samuel Klein</a> por la <a href="https://github.com/pachterlab/gget/issu90">sugerencia</a>)</li>
</ul>
<p><strong>Versi√≥n ‚â• 0.27.8</strong> (12 de julio de 2023):</p>
<ul>
<li>Nuevo argumento para <a href="es/search.html"><code>gget search</code></a>: especifique la versi√≥n de Ensembl desde la cual se obtiene la informaci√≥n con <code>-r</code> <code>--release</code></li>
<li>Se corrigi√≥ un <a href="https://github.com/pachterlab/gget/issu91">error</a> en <a href="es/pdb.html"><code>gget pdb</code></a> (este error se introdujo en la versi√≥n 0.27.5)</li>
</ul>
<p><strong>Versi√≥n ‚â• 0.27.7</strong> (15 de mayo de 2023):</p>
<ul>
<li>Se movieron las dependencias para los m√≥dulos <a href="es/gpt.html"><code>gget gpt</code></a> y <a href="es/cellxgene.html"><code>gget cellxgene</code></a> de los requisitos instalados autom√°ticamente a <a href="es/setup.html"><code>gget setup</code></a></li>
<li>Dependencias <a href="es/alphafold.html"><code>gget alphafold</code></a> actualizadas para compatibilidad con Python &gt;= 3.10</li>
<li>Se agreg√≥ el argumento <code>census_version</code> a <a href="es/cellxgene.html"><code>gget cellxgene</code></a></li>
</ul>
<p><strong>Versi√≥n ‚â• 0.27.6</strong> (1 de mayo de 2023) (TIRO debido a problemas con las dependencias -&gt; reemplazada por la versi√≥n 0.27.7):</p>
<ul>
<li>Gracias a el PR de <a href="https://github.com/tdido">Tom√°s Di Domenico</a>: <a href="es/search.html"><code>gget search</code></a> ahora tambi√©n puede consultar los ID de plantas üå± Ensembl</li>
<li>Nuevo m√≥dulo: <a href="es/cellxgene.html"><code>gget cellxgene</code></a></li>
</ul>
<p><strong>Versi√≥n ‚â• 0.27.5</strong> (6 de abril de 2023):</p>
<ul>
<li>Se actualiz√≥ <a href="es/search.html"><code>gget search</code></a> para que funcione correctamente con la nueva versi√≥n de <a href="https://pypi.org/project/pandas/2.0.0/">Pandas</a> 2.0.0 (lanzado el 3 de abril de 2023), adem√°s de versiones anteriores de Pandas</li>
<li>Se actualiz√≥ <a href="es/info.html"><code>gget info</code></a> con nuevos banderas <code>uniprot</code> y <code>ncbi</code> que permiten desactivar los resultados de estas bases de datos de forma independiente para ahorrar tiempo de ejecuci√≥n (nota: el indicador <code>ensembl_only</code> qued√≥ obsoleto)</li>
<li>Todos los m√≥dulos gget ahora tienen una bandera <code>-q / --quiet</code> (para Python: <code>verbose=False</code>) para desactivar la informaci√≥n de progreso</li>
</ul>
<p><strong>Versi√≥n ‚â• 0.27.4</strong> (19 de marzo de 2023):</p>
<ul>
<li>Nuevo m√≥dulo: <a href="es/gpt.html"><code>gget gpt</code></a> </li>
</ul>
<p><strong>Versi√≥n ‚â• 0.27.3</strong> (11 de marzo de 2023):</p>
<ul>
<li><a href="es/info.html"><code>gget info</code></a> excluye los ID de PDB de forma predeterminada para aumentar la velocidad (los resultados de PDB se pueden incluir usando la marca <code>--pdb</code> / <code>pdb=True</code>).</li>
</ul>
<p><strong>Versi√≥n ‚â• 0.27.2</strong> (1 de enero de 2023):</p>
<ul>
<li>Se actualiz√≥ <a href="es/alphafold.html"><code>gget alphafold</code></a> a <a href="https://github.com/deepmind/alphafold/releases/tag/v2.3.0">DeepMind's AlphaFold v2.3.0</a> (incluidos los nuevos argumentos <code>multimer_for_monomer </code> y <code>multimer_recycles</code>)</li>
</ul>
<p><strong>Versi√≥n ‚â• 0.27.0</strong> (10 de diciembre de 2022):</p>
<ul>
<li>Se actualiz√≥ <a href="es/alphafold.html"><code>gget alphafold</code></a> para que coincida con los cambios recientes de DeepMind</li>
<li>N√∫mero de versi√≥n actualizado para que coincida con la edad de <a href="https://github.com/lauraluebbert">el creador de gget</a> siguiendo una larga tradici√≥n de laboratorio de Pachter</li>
</ul>
<p><strong>Versi√≥n ‚â• 0.3.13</strong> (11 de noviembre de 2022):</p>
<ul>
<li>Tiempo de ejecuci√≥n reducido para <a href="es/enrichr.html"><code>gget enrichr</code></a> y <a href="es/archs4.html"><code>gget archs4</code></a> cuando se usa con ID de Ensembl</li>
</ul>
<p><strong>Versi√≥n ‚â• 0.3.12</strong> (10 de noviembre de 2022):</p>
<ul>
<li><a href="es/info.html"><code>gget info</code></a> ahora tambi√©n devuelve datos de localizaci√≥n subcelular de UniProt</li>
<li>El nuevo indicador <a href="es/info.html"><code>gget info</code></a> <code>ensembl_only</code> devuelve solo los resultados de Ensembl</li>
<li>Tiempo de ejecuci√≥n reducido para <a href="es/info.html"><code>gget info</code></a> y <a href="es/seq.html"><code>gget seq</code></a></li>
</ul>
<p><strong>Versi√≥n ‚â• 0.3.11</strong> (7 de septiembre de 2022):</p>
<ul>
<li>Nuevo m√≥dulo: <a href="es/pdb.html"><code>gget pdb</code></a></li>
</ul>
<p><strong>Versi√≥n ‚â• 0.3.10</strong> (2 de septiembre de 2022):</p>
<ul>
<li><a href="es/alphafold.html"><code>gget alphafold</code></a> ahora tambi√©n devuelve valores pLDDT para generar gr√°ficos sin volver a ejecutar el programa (consulte tambi√©n las <a href="https://github.com/pachterlab/gget/discusion39">preguntas frecuentes de gget alphafold</a>)</li>
</ul>
<p><strong>Versi√≥n ‚â• 0.3.9</strong> (25 de agosto de 2022):</p>
<ul>
<li>Instrucciones de instalaci√≥n de openmm actualizadas para <a href="es/alphafold.html"><code>gget alphafold</code></a></li>
</ul>
<p><strong>Versi√≥n ‚â• 0.3.8</strong> (12 de agosto de 2022):</p>
<ul>
<li>Se corrigieron los requisitos de versi√≥n de mysql-connector-python</li>
</ul>
<p><strong>Versi√≥n ‚â• 0.3.7</strong> (9 de agosto de 2022):</p>
<ul>
<li><strong>NOTA:</strong> El <a href="http://ftp.ensembl.org/pub/">sitio FTP de Ensembl</a> cambi√≥ su estructura el 8 de agosto de 2022. Actualice a la versi√≥n <code>gget</code> ‚â• 0.3.7 si usa <a href="es/ref.html"><code>obtener ref</code></a></li>
</ul>
<p><strong>Versi√≥n ‚â• 0.3.5</strong> (6 de agosto de 2022):</p>
<ul>
<li>Nuevo m√≥dulo: <a href="es/alphafold.html"><code>gget alphafold</code></a></li>
</ul>
<p><strong>Versi√≥n ‚â• 0.2.6</strong> (7 de julio de 2022):</p>
<ul>
<li>¬°<a href="es/ref.html"><code>gget ref</code></a> ahora admite genomas de plantas! üå±</li>
</ul>
<p><strong>Versi√≥n ‚â• 0.2.5</strong> (30 de junio de 2022):</p>
<ul>
<li><strong>NOTA:</strong> <a href="https://www.uniprot.org/">UniProt</a> cambi√≥ la estructura de su API el 28 de junio de 2022. Actualice a la versi√≥n <code>gget</code> ‚â• 0.2.5 si usa alguno de los m√≥dulos que consultan datos de UniProt (<a href="es/info.html"><code>gget info</code></a> y <a href="es/seq.html"><code>gget seq</code></a>).</li>
</ul>
<p><strong>Versi√≥n ‚â• 0.2.3:</strong> (26 de junio de 2022):</p>
<ul>
<li>JSON ahora es el formato de regreso predeterminado para la Terminal para los m√≥dulos que anteriormente devolv√≠an el formato de data frame (CSV) (el formato se puede convertir a data frame/CSV usando la bandera <code>[-csv][--csv]</code>). El formato data frame/CSV sigue siendo el formato de regreso predeterminada para Python (Jupyter Lab/Google Colab) (y se puede convertir a JSON con <code>json=True</code>).</li>
<li>Para todos los m√≥dulos, el primer par√°metro requerido se convirti√≥ en un par√°metro posicional y ya no debe nombrarse en la l√≠nea de comandos, p. ej. <code>gget ref -s human</code> ‚Üí <code>gget ref human</code>.</li>
<li><code>gget info</code>: <code>[--expand]</code> est√° en desuso. El m√≥dulo ahora siempre devolver√° toda la informaci√≥n disponible.</li>
<li>Ligeros cambios en la salida devuelta por <code>gget info</code>, incluida la devoluci√≥n de los ID de Ensembl versionados.</li>
<li><code>gget info</code> y <code>gget seq</code> ahora son compatibles con las IDs de WormBase y FlyBase.</li>
<li>Ahora tambi√©n se pueden ingresar IDs de tipo Ensembl a <code>gget archs4</code> y <code>gget enrichr</code> con la bandera <code>[-e][--ensembl]</code> (<code>ensembl=True</code> para Python (Jupyter Lab / Google Colab)).</li>
<li>El par√°metro <code>seqtype</code> de <code>gget seq</code> fue reemplazado por la bandera <code>[-t][--translate]</code> (<code>translate=True/False</code> para Python (Jupyter Lab / Google Colab)) que devolver√° secuencias de nucle√≥tidos (<code>False</code>) o amino√°cidos (<code>True</code>).</li>
<li>El par√°metro <code>seqtype</code> de <code>gget search</code> se renombr√≥ a <code>id_type</code> (a√∫n tomando los mismos par√°metros 'gene' o 'transcript').</li>
</ul>
<div style="break-before: page; page-break-before: always;"></div><p><a href="https://pypi.org/project/gget"><img src="https://img.shields.io/pypi/v/gget" alt="pypi version" /></a>
<a href="https://anaconda.org/bioconda/gget"><img src="https://anaconda.org/bioconda/gget/badges/version.svg" alt="image" /></a></p>
<h2 id="instalaci√≥n"><a class="header" href="#instalaci√≥n">Instalaci√≥n</a></h2>
<pre><code class="language-bash">pip install --upgrade gget
</code></pre>
<p>Alternativa:</p>
<pre><code class="language-bash">conda install -c bioconda gget
</code></pre>
<p>Para uso en Jupyter Lab / Google Colab:</p>
<pre><code class="language-python">import gget
</code></pre>
<div style="break-before: page; page-break-before: always;"></div><h2 id="-gu√≠a-de-inicio-r√°pido"><a class="header" href="#-gu√≠a-de-inicio-r√°pido">ü™Ñ Gu√≠a de inicio r√°pido</a></h2>
<p>Terminal:</p>
<pre><code class="language-bash"># Obtenga todos los FTP de anotaciones y referencias de Homo sapiens de la √∫ltima versi√≥n de Ensembl
$ gget ref homo_sapiens

# Obtenga IDs de Ensembl de genes humanos con &quot;ace2&quot; o &quot;angiotensin converting enzyme 2&quot; en su nombre/descripci√≥n
$ gget search -s homo_sapiens 'ace2' 'angiotensin converting enzyme 2'

# Busque el gen ENSG00000130234 (ACE2) y su transcripci√≥n ENST00000252519
$ gget info ENSG00000130234 ENST00000252519

# Obtenga la secuencia de amino√°cidos de la transcripci√≥n can√≥nica del gen ENSG00000130234
$ gget seq --translate ENSG00000130234

# R√°pidamente encuentra la ubicaci√≥n gen√≥mica de la secuencia de amino√°cidos
$ gget blat MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS

# BLAST la secuencia de amino√°cidos
$ gget blast MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS

# Alinee m√∫ltiples secuencias de nucle√≥tidos o amino√°cidos entre s√≠ (tambi√©n acepta la ruta al archivo FASTA)  
$ gget muscle MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS

# Alinee una o m√°s secuencias de amino√°cidos con una referencia (que contiene una o m√°s secuencias) (BLAST local) (tambi√©n acepta rutas a archivos FASTA)  
$ gget diamond MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS -ref MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS

# Alinea secuencias de nucle√≥tidos o amino√°cidos en un archivo FASTA
$ gget muscle path/to/file.fa

# Use Enrichr para un an√°lisis de ontolog√≠a de una lista de genes
$ gget enrichr -db ontology ACE2 AGT AGTR1 ACE AGTRAP AGTR2 ACE3P

# Obtene la expresi√≥n en tejido humano del gen ACE2
$ gget archs4 -w tissue ACE2

# Obtenga la estructura de la prote√≠na (en formato PDB) de ACE2 (ID de PDB devuelta por gget info)
$ gget pdb 1R42 -o 1R42.pdb

# Encuentre motivos lineales eucari√≥ticos (ELM) en una secuencia de amino√°cidos
$ gget setup elm        # solo debe ejecutarse una vez
$ gget elm -o results MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS

# Obtene una matriz de recuento de scRNAseq (formato AnnData) basada en genes, tejidos y tipos de c√©lulas especificados (especie predeterminada: humano)
$ gget setup cellxgene  # solo debe ejecutarse una vez
$ gget cellxgene --gene ACE2 SLC5A1 --tissue lung --cell_type 'mucus secreting cell' -o example_adata.h5ad

# Predice la estructura proteica de GFP a partir de su secuencia de amino√°cidos
$ gget setup alphafold  # solo debe ejecutarse una vez
$ gget alphafold MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK
</code></pre>
<p>Python (Jupyter Lab / Google Colab):</p>
<pre><code class="language-python">import gget
gget.ref(&quot;homo_sapiens&quot;)
gget.search([&quot;ace2&quot;, &quot;angiotensin converting enzyme 2&quot;], &quot;homo_sapiens&quot;)
gget.info([&quot;ENSG00000130234&quot;, &quot;ENST00000252519&quot;])
gget.seq(&quot;ENSG00000130234&quot;, translate=True)
gget.blat(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)
gget.blast(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)
gget.muscle([&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;, &quot;MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS&quot;])
gget.diamond(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;, reference=&quot;MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS&quot;)
gget.enrichr([&quot;ACE2&quot;, &quot;AGT&quot;, &quot;AGTR1&quot;, &quot;ACE&quot;, &quot;AGTRAP&quot;, &quot;AGTR2&quot;, &quot;ACE3P&quot;], database=&quot;ontology&quot;, plot=True)
gget.archs4(&quot;ACE2&quot;, which=&quot;tissue&quot;)
gget.pdb(&quot;1R42&quot;, save=True)

gget.setup(&quot;elm&quot;)         # solo debe ejecutarse una vez
ortho_df, regex_df = gget.elm(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)

gget.setup(&quot;cellxgene&quot;)   # solo debe ejecutarse una vez
gget.cellxgene(gene = [&quot;ACE2&quot;, &quot;SLC5A1&quot;], tissue = &quot;lung&quot;, cell_type = &quot;mucus secreting cell&quot;)

gget.setup(&quot;alphafold&quot;)   # solo debe ejecutarse una vez
gget.alphafold(&quot;MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK&quot;)
</code></pre>
<p>Use a <code>gget</code> con R usando <a href="https://rstudio.github.io/reticulate/">reticulate</a>:</p>
<pre><code class="language-r">system(&quot;pip install gget&quot;)
install.packages(&quot;reticulate&quot;)
library(reticulate)
gget &lt;- import(&quot;gget&quot;)

gget$ref(&quot;homo_sapiens&quot;)
gget$search(list(&quot;ace2&quot;, &quot;angiotensin converting enzyme 2&quot;), &quot;homo_sapiens&quot;)
gget$info(list(&quot;ENSG00000130234&quot;, &quot;ENST00000252519&quot;))
gget$seq(&quot;ENSG00000130234&quot;, translate=TRUE)
gget$blat(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)
gget$blast(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)
gget$muscle(list(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;, &quot;MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS&quot;), out=&quot;out.afa&quot;)
gget$diamond(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;, reference=&quot;MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS&quot;)
gget$enrichr(list(&quot;ACE2&quot;, &quot;AGT&quot;, &quot;AGTR1&quot;, &quot;ACE&quot;, &quot;AGTRAP&quot;, &quot;AGTR2&quot;, &quot;ACE3P&quot;), database=&quot;ontology&quot;)
gget$archs4(&quot;ACE2&quot;, which=&quot;tissue&quot;)
gget$pdb(&quot;1R42&quot;, save=TRUE)
</code></pre>
<h4 id="m√°s-ejemplos"><a class="header" href="#m√°s-ejemplos"><a href="https://github.com/pachterlab/gget_examples">M√°s ejemplos</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-alphafold--1"><a class="header" href="#gget-alphafold--1">gget alphafold ü™¢</a></h2>
<p>Predice la estructura en 3D de cualquier prote√≠na derivada de su secuencia de amino√°cidos usando una versi√≥n simplificada del algoritmo <a href="https://github.com/deepmind/alphafold">AlphaFold2</a> de <a href="https://www.deepmind.com/">DeepMind</a>, originalmente producido y publicado para <a href="https://colab.research.google.com/github/deepmind/alphafold/blob/main/notebooks/AlphaFold.ipynb">AlphaFold Colab</a>.<br />
Resultado: Predicci√≥n de la estructura (en formato PDB) y el err√≥r de alineaci√≥n (en formato json).</p>
<p>Antes de usar <code>gget alphafold</code> por primera vez, corre <code>gget setup alphafold</code> / <code>gget.setup(&quot;alphafold&quot;)</code> (ver tambi√©n <a href="es/setup.html"><code>gget setup</code></a>).</p>
<p><strong>Par√°metro posicional</strong><br />
<code>sequence</code><br />
Secuencia de amino√°cidos (str), o una lista de secuencias (<em>gget alphafold automaticamente usa el algoritmo del mult√≠mero si m√∫ltiples secuencias son ingresadas</em>), o una ruta a un archivo formato FASTA.</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-mr</code> <code>--multimer_recycles</code><br />
El algoritmo de mult√≠mero se reciclara hasta que las predicciones dejen de cambiar, el limite de ciclos esta indicado aqui. Por defecto: 3<br />
Para obtener m√°s exactitud, ajusta este limite a 20 (al costo de ejecuciones mas tardadas).</p>
<p><code>-o</code> <code>--out</code><br />
Ruta a la carpeta para guardar los resultados de la predicci√≥n (str). Por defecto: &quot;./[fecha_tiempo]_gget_alphafold_prediction&quot;.</p>
<p><strong>Banderas</strong><br />
<code>-mfm</code> <code>--multimer_for_monomer</code><br />
Usa el algoritmo de mult√≠mero para un mon√≥mero.</p>
<p><code>-r</code> <code>--relax</code><br />
Relaja el mejor modelo con el algoritmo AMBER.</p>
<p><code>-q</code> <code>--quiet</code><br />
Uso limitado para Terminal. Impide la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.<br />
Para Python, usa <code>verbose=False</code>.</p>
<p><code>plot</code><br />
Solo para Python. <code>plot=True</code> prov√©e una visualizaci√≥n interactiva de la predicci√≥n con el err√≥r de alineaci√≥n en 3D con <a href="https://pypi.org/project/py3Dmol/">py3Dmol</a> y <a href="https://matplotlib.org/">matplotlib</a> (por defecto: True).</p>
<p><code>show_sidechains</code><br />
Solo para Python. <code>show_sidechains=True</code> incluye las cadenas laterales de prote√≠nas en el esquema (por defecto: True).</p>
<h3 id="ejemplo"><a class="header" href="#ejemplo">Ejemplo</a></h3>
<pre><code class="language-bash"># Predice la estructura de una prote√≠na derivada de su secuencia de amino√°cidos
gget alphafold MAAHKGAEHHHKAAEHHEQAAKHHHAAAEHHEKGEHEQAAHHADTAYAHHKHAEEHAAQAAKHDAEHHAPKPH

# Encuentra secuencias similares previamente depositadas en el PDB para an√°lisis comparativo
gget blast --database pdbaa MAAHKGAEHHHKAAEHHEQAAKHHHAAAEHHEKGEHEQAAHHADTAYAHHKHAEEHAAQAAKHDAEHHAPKPH

# Busca los archivos PDB de estructuras similares resultantes de gget blast para comparar y obtener una medida de calidad del modelo predecido.
gget pdb 3UQ3 -o 3UQ3.pdb
gget pdb 2K42 -o 2K42.pdb
</code></pre>
<pre><code class="language-python"># Python
# Predice la estructura de una prote√≠na derivada de su secuencia de amino√°cidos
gget.alphafold(&quot;MAAHKGAEHHHKAAEHHEQAAKHHHAAAEHHEKGEHEQAAHHADTAYAHHKHAEEHAAQAAKHDAEHHAPKPH&quot;)

# Encuentra secuencias similares previamente depositadas en el PDB para an√°lisis comparativo
gget.blast(&quot;MAAHKGAEHHHKAAEHHEQAAKHHHAAAEHHEKGEHEQAAHHADTAYAHHKHAEEHAAQAAKHDAEHHAPKPH&quot;, database=&quot;pdbaa&quot;)

# Busca los archivos PDB de estructuras similares resultantes de gget blast para comparar y obtener una medida de calidad del modelo predecido.
gget.pdb(&quot;3UQ3&quot;, save=True)
gget.pdb(&quot;2K42&quot;, save=True)
</code></pre>
<p>‚Üí <code>gget alphafold</code> produce la estructura predecida (en formato PDB) y el err√≥r de alineaci√≥n (en formato json) en una nueva carpeta (&quot;./[fecha_tiempo]_gget_alphafold_prediction&quot;). Este ejemplo demuestra como usar <a href="es/blast.html"><code>gget blast</code></a> y <a href="es/pdb.html"><code>gget pdb</code></a> para correr un an√°lisis comparativo. Los archivos PDB se pueden ver en 3D con <a href="https://rcsb.org/3d-view">RCSB 3D view</a>, o usando programas como <a href="https://pymol.org/">PyMOL</a> o <a href="https://www.blender.org/">Blender</a>. Para comparar m√∫ltiples archivos PDB, use <a href="https://rcsb.org/alignment">RCSB alignment</a>. Python tambi√©n produce <a href="https://twitter.com/NeuroLuebbert/status/1555968042948915200">esquemas interactivos</a>, los cuales se pueden generar de los archivos PDB y JSON, como es describido en <a href="https://github.com/pachterlab/gget/discussions/39">gget alphafold FAQ</a> Q4.</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/4qxGF1tbZ3I?si=mEqQ5oSnDYtg2OP7" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
<h3 id="ejemplo-en-google-colab"><a class="header" href="#ejemplo-en-google-colab"><a href="https://github.com/pachterlab/gget_examplblob/main/gget_alphafold.ipynb">Ejemplo en Google Colab</a></a></h3>
<h3 id="gget-alphafold---preguntas-m√°s-frecuentes"><a class="header" href="#gget-alphafold---preguntas-m√°s-frecuentes"><a href="https://github.com/pachterlab/gget/discussions/39">gget alphafold - preguntas m√°s frecuentes</a></a></h3>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no es especificado de otra manera. Las banderas son designadas como cierto o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede obtener desde Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-archs4--1"><a class="header" href="#gget-archs4--1">gget archs4 üêÅ</a></h2>
<p>Encuentra los genes m√°s correlacionados a un gen de inter√©s, o bi√©n, encuentra los tejidos donde un gen se expresa usando la base de datos <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.<br />
Produce: Resultados en formato JSON (Terminal) o Dataframe/CSV (Python).</p>
<p><strong>Par√°metro posicional</strong><br />
<code>gene</code><br />
Nombre corto (s√≠mbolo del gen) del gen de inter√©s, p. ej. STAT4.<br />
Alternativamente: usa la bandera <code>--ensembl</code> para ingresar un ID tipo Ensembl, p. ej. ENSG00000138378.</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-w</code> <code>--which</code><br />
'correlation' (correlaci√≥n; se usa por defecto) o 'tissue' (tejido).<br />
'correlation' produce una tabla que contiene los 100 genes m√°s correlacionados con el gen de inter√©s. La correlaci√≥n de Pearson se calcula de todas las muestras y tejidos en <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.<br />
'tissue' produce un atlas de expresi√≥n tisular calculado de todas las muestras humanas o de rat√≥n (seg√∫n lo definido usando el par√°metro <code>--species</code> (especies)) en <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.</p>
<p><code>-s</code> <code>--species</code><br />
'human' (humano; se usa por defecto) o 'mouse' (rat√≥n).<br />
Define si se usan muestras humanas o de rat√≥n de <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.<br />
(Solo aplica para el atlas de expresi√≥n tisular.)</p>
<p><code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados, p. ej. ruta/al/directorio/resultados.csv (o .json). Por defecto: salida est√°ndar (STDOUT).<br />
Para Python, use <code>save=True</code> para guardar los resultados en el directorio de trabajo actual.</p>
<p><strong>Banderas</strong><br />
<code>-e</code> <code>--ensembl</code><br />
Usa esta bandera si <code>gene</code> se ingresa como ID tipo Ensembl.</p>
<p><code>-csv</code> <code>--csv</code><br />
Solo para Terminal. Produce los resultados en formato CSV.<br />
Para Python, usa <code>json=True</code> para obtener los resultados en formato JSON.</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para Terminal. Impide la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.<br />
Para Python, usa <code>verbose=False</code> para impedir la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.</p>
<h3 id="ejemplo-1"><a class="header" href="#ejemplo-1">Ejemplo</a></h3>
<pre><code class="language-bash">gget archs4 ACE2
</code></pre>
<pre><code class="language-python"># Python
gget.archs4(&quot;ACE2&quot;)
</code></pre>
<p>‚Üí Produce los 100 genes m√°s correlacionados con el gen ACE2:</p>
<div class="table-wrapper"><table><thead><tr><th>gene_symbol</th><th>pearson_correlation</th></tr></thead><tbody>
<tr><td>SLC5A1</td><td>0.579634</td></tr>
<tr><td>CYP2C18</td><td>0.576577</td></tr>
<tr><td>. . .</td><td>. . .</td></tr>
</tbody></table>
</div>
<p><br/><br/></p>
<pre><code class="language-bash">gget archs4 -w tissue ACE2
</code></pre>
<pre><code class="language-python"># Python
gget.archs4(&quot;ACE2&quot;, which=&quot;tissue&quot;)
</code></pre>
<p>‚Üí Produce la expresi√≥n tisular de ACE2 (por defecto, se utilizan datos humanos):</p>
<div class="table-wrapper"><table><thead><tr><th>id</th><th>min</th><th>q1</th><th>median</th><th>q3</th><th>max</th></tr></thead><tbody>
<tr><td>System.Urogenital/Reproductive System.Kidney.RENAL CORTEX</td><td>0.113644</td><td>8.274060</td><td>9.695840</td><td>10.51670</td><td>11.21970</td></tr>
<tr><td>System.Digestive System.Intestine.INTESTINAL EPITHELIAL CELL</td><td>0.113644</td><td>5.905560</td><td>9.570450</td><td>13.26470</td><td>13.83590</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td></tr>
</tbody></table>
</div>
<p><br/><br/>
Consulte <a href="https://davetang.org/muse/2023/05/16/check-where-a-gene-is-expressed-from-the-command-line/">este tutorial</a> de Dave Tang, quien escribi√≥ un script R para crear esta visualizaci√≥n con los resultados de <code>gget archs4</code> en formato JSON:</p>
<p><img src="https://github.com/pachterlab/gget/assets/56094636/f2a34a9e-beaa-45a5-a678-d38399dd3017" alt="image" /></p>
<h4 id="m√°s-ejemplos-1"><a class="header" href="#m√°s-ejemplos-1"><a href="https://github.com/pachterlab/gget_examples">M√°s ejemplos</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Las banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-blast--1"><a class="header" href="#gget-blast--1">gget blast üí•</a></h2>
<p>BLAST una secuencia de nucle√≥tidos o amino√°cidos a cualquier base de datos <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a>.<br />
Produce: Resultados en formato JSON (Terminal) o Dataframe/CSV (Python).</p>
<p><strong>Par√°metro posicional</strong><br />
<code>sequence</code><br />
Secuencia de nucle√≥tidos o amino√°cidos, o una ruta a un archivo tipo FASTA o .txt.</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-p</code> <code>--program</code><br />
'blastn', 'blastp', 'blastx', 'tblastn', o 'tblastx'.<br />
Por defecto: 'blastn' para secuencias de nucle√≥tidos; 'blastp' para secuencias de amino√°cidos.</p>
<p><code>-db</code> <code>--database</code><br />
'nt', 'nr', 'refseq_rna', 'refseq_protein', 'swissprot', 'pdbaa', o 'pdbnt'.<br />
Por defecto: 'nt' para secuencias de nucle√≥tidos; 'nr' para secuencias de amino√°cidos.<br />
<a href="https://ncbi.github.io/blast-cloud/blastdb/available-blastdbs.html">M√°s informaci√≥n sobre los bases de datos BLAST</a></p>
<p><code>-l</code> <code>--limit</code><br />
Limita el n√∫mero de resultados producidos. Por defecto: 50.</p>
<p><code>-e</code> <code>--expect</code><br />
Define el umbral de <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=FAQ#expect">'expect value'</a>. Por defecto: 10.0.</p>
<p><code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados, p. ej. ruta/al/directorio/resultados.csv (o .json). Por defecto: salida est√°ndar (STDOUT).<br />
Para Python, usa <code>save=True</code> para guardar los resultados en el directorio de trabajo actual.</p>
<p><strong>Banderas</strong><br />
<code>-lcf</code> <code>--low_comp_filt</code><br />
Activa el <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=FAQ#LCR">'low complexity filter'</a> (filtro de baja complejidad).</p>
<p><code>-mbo</code> <code>--megablast_off</code><br />
Desactiva el algoritmo MegaBLAST. Por defecto: MegaBLAST esta activado (solo aplicable para blastn).</p>
<p><code>-csv</code> <code>--csv</code><br />
Solo para Terminal. Produce los resultados en formato CSV.<br />
Para Python, usa <code>json=True</code> para producir los resultados en formato JSON.</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para Terminal. Impide la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.<br />
Para Python, usa <code>verbose=False</code> para imipidir la informacion de progreso de ser exhibida durante la ejecuci√≥n del programa.</p>
<p><code>wrap_text</code><br />
Solo para Python. <code>wrap_text=True</code> muestra los resultados con texto envuelto para facilitar la lectura (por defecto: False).</p>
<h3 id="por-ejemplo"><a class="header" href="#por-ejemplo">Por ejemplo</a></h3>
<pre><code class="language-bash">gget blast MKWMFKEDHSLEHRCVESAKIRAKYPDRVPVIVEKVSGSQIVDIDKRKYLVPSDITVAQFMWIIRKRIQLPSEKAIFLFVDKTVPQSR
</code></pre>
<pre><code class="language-python"># Python
gget.blast(&quot;MKWMFKEDHSLEHRCVESAKIRAKYPDRVPVIVEKVSGSQIVDIDKRKYLVPSDITVAQFMWIIRKRIQLPSEKAIFLFVDKTVPQSR&quot;)
</code></pre>
<p>‚Üí Produce los resultados BLAST de la secuencia de inter√©s. <code>gget blast</code> autom√°ticamente detecta esta secuencia como una secuencia de amino√°cidos y, por lo tanto, establece el programa BLAST en <em>blastp</em> con la base de datos <em>nr</em>.</p>
<div class="table-wrapper"><table><thead><tr><th>Description</th><th>Scientific Name</th><th>Common Name</th><th>Taxid</th><th>Max Score</th><th>Total Score</th><th>Query Cover</th><th>...</th></tr></thead><tbody>
<tr><td>PREDICTED: gamma-aminobutyric acid receptor-as...</td><td>Colobus angolensis palliatus</td><td>NaN</td><td>336983</td><td>180</td><td>180</td><td>100%</td><td>...</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>...</td></tr>
</tbody></table>
</div>
<p><strong>BLAST desde un archivo .fa o .txt:</strong></p>
<pre><code class="language-bash">gget blast fasta.fa
</code></pre>
<pre><code class="language-python"># Python
gget.blast(&quot;fasta.fa&quot;)
</code></pre>
<p>‚Üí Produce los resultados BLAST de la primera secuencia contenida en el archivo 'fasta.fa'.</p>
<h4 id="m√°s-ejemplos-2"><a class="header" href="#m√°s-ejemplos-2"><a href="https://github.com/pachterlab/gget_examples">M√°s ejemplos</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-blat--1"><a class="header" href="#gget-blat--1">gget blat üéØ</a></h2>
<p>Encuentra la ubicaci√≥n gen√≥mica de una secuencia de nucle√≥tidos o amino√°cidos usando <a href="https://genome.ucsc.edu/cgi-bin/hgBlat">BLAT</a>.<br />
Produce: Resultados en formato JSON (Terminal) o Dataframe/CSV (Python).</p>
<p><strong>Par√°metro posicional</strong><br />
<code>sequence</code><br />
Secuencia de nucle√≥tidos o amino√°cidos, o una ruta a un archivo tipo FASTA o .txt.</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-st</code> <code>--seqtype</code><br />
'DNA', 'protein', 'translated%20RNA', o 'translated%20DNA'.<br />
Por defecto: 'DNA' para secuencias de nucle√≥tidos; 'protein' para secuencias de amino√°cidos.</p>
<p><code>-a</code> <code>--assembly</code><br />
Ensamblaje del genoma. 'human' (hg38) (se usa por defecto), 'mouse' (mm39) (rat√≥n), 'zebrafish' (taeGut2) (pinz√≥n cebra),<br />
o cualquiera de los ensamblajes de especies disponibles <a href="https://genome.ucsc.edu/cgi-bin/hgBlat">aqu√≠</a> (use el nombre corto del ensamblado, p. ej. 'hg38').</p>
<p><code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados, p. ej. ruta/al/directorio/resultados.csv (o .json). Por defecto: salida est√°ndar (STDOUT).<br />
Para Python, usa <code>save=True</code> para guardar los resultados en el directorio de trabajo actual.</p>
<p><strong>Banderas</strong><br />
<code>-csv</code> <code>--csv</code><br />
Solo para Terminal. Produce los resultados en formato CSV.<br />
Para Python, usa <code>json=True</code> para producir los resultados en formato JSON.</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para Terminal. Impide la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.<br />
Para Python, usa <code>verbose=False</code> para impedir la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.</p>
<h3 id="ejemplo-2"><a class="header" href="#ejemplo-2">Ejemplo</a></h3>
<pre><code class="language-bash">gget blat -a taeGut2 MKWMFKEDHSLEHRCVESAKIRAKYPDRVPVIVEKVSGSQIVDIDKRKYLVPSDITVAQFMWIIRKRIQLPSEKAIFLFVDKTVPQSR
</code></pre>
<pre><code class="language-python"># Python
gget.blat(&quot;MKWMFKEDHSLEHRCVESAKIRAKYPDRVPVIVEKVSGSQIVDIDKRKYLVPSDITVAQFMWIIRKRIQLPSEKAIFLFVDKTVPQSR&quot;, assembly=&quot;taeGut2&quot;)
</code></pre>
<p>‚Üí Produce los resultados de BLAT para el ensamblaje taeGut2 (pinz√≥n cebra). En este ejemplo, <code>gget blat</code> autom√°ticamente detecta esta secuencia como una secuencia de amino√°cidos y, por lo tanto, establece el tipo de secuencia (<code>--seqtype</code>) como <em>prote√≠na</em>. </p>
<div class="table-wrapper"><table><thead><tr><th>genome</th><th>query_size</th><th>aligned_start</th><th>aligned_end</th><th>matches</th><th>mismatches</th><th>%_aligned</th><th>...</th></tr></thead><tbody>
<tr><td>taeGut2</td><td>88</td><td>12</td><td>88</td><td>77</td><td>0</td><td>87.5</td><td>...</td></tr>
</tbody></table>
</div>
<h4 id="m√†s-ejemplos"><a class="header" href="#m√†s-ejemplos"><a href="https://github.com/pachterlab/gget_examples">M√†s ejemplos</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Las banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-cellxgene--1"><a class="header" href="#gget-cellxgene--1">gget cellxgene üç±</a></h2>
<p>Query data de la base de datos <a href="https://cellxgene.cziscience.com/">CZ CELLxGENE Discover</a> usando <a href="https://github.com/chanzuckerberg/cellxgene-census">CZ CELLxGENE Discover Census</a>.<br />
Produce: Un objeto AnnData que contiene la matriz de recuentos de genes y los metadatos de resultados de single cell RNA-seq de los tejidos/genes/etcetera previamente definidos.</p>
<p>Antes de usar <code>gget cellxgene</code> por primera vez, corre <code>gget setup cellxgene</code> / <code>gget.setup(&quot;cellxgene&quot;)</code> (ver tambi√©n <a href="es/setup.html"><code>gget setup</code></a>).</p>
<p><strong>Par√°metros opcionales</strong><br />
<code>-s</code> <code>--species</code><br />
'homo_sapiens' o 'mus_musculus'. Por defecto: 'homo_sapiens'.</p>
<p><code>-g</code> <code>--gene</code><br />
Str o lista de genes de inter√©s o ID(s) tipo Ensembl. Por defecto: None (ninguno).<br />
Atenci√≥n: Utilice la bandera <code>-e / --ensembl</code> (Python: <code>ensembl=True</code>) cuando ingrese ID(s) tipo Ensembl.<br />
Ver https://cellxgene.cziscience.com/gene-expression para ejemplos de genes.</p>
<p><code>-cv</code> <code>--census_version</code><br />
Versi√≥n del CZ CELLxGENE Discover Census (str), p. ej. &quot;2023-05-15&quot;, o &quot;latest&quot; (ultima) o &quot;stable&quot; (estable). Por defecto: &quot;stable&quot; (estable).</p>
<p><code>-cn</code> <code>--column_names</code><br />
Lista de columnas de metadatos a obtener (almacenadas en AnnData.obs).<br />
Por defecto: ['dataset_id', 'assay', 'suspension_type', 'sex', 'tissue_general', 'tissue', 'cell_type']<br />
Para m√°s opciones, ver: https://api.cellxgene.cziscience.com/curation/ui/#/ -&gt; 'Schemas' -&gt; 'dataset'</p>
<p><code>-o</code> <code>--out</code><br />
Ruta al archivo para guardar el objeto AnnData formato .h5ad (o .csv con bandera <code>-mo / --meta_only</code>).<br />
¬°Requerido cuando se usa desde Terminal!</p>
<p><strong>Banderas</strong><br />
<code>-e</code> <code>--ensembl</code><br />
Usa esta bandera si <code>gene</code> se ingresa como ID tipo Ensembl.</p>
<p><code>-mo</code> <code>--meta_only</code><br />
Solo produce la tabla (Dataframe) con metadatos (corresponde a AnnData.obs).</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para Terminal. Impide la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.<br />
Para Python, usa <code>verbose=False</code> para impedir la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.</p>
<p><strong>Par√°metros opcionales correspondientes a los atributos de metadatos de CZ CELLxGENE Discover</strong><br />
<code>--tissue</code><br />
Str o lista de tejido(s), p. ej. ['lung', 'blood']. Por defecto: None.<br />
Ver https://cellxgene.cziscience.com/gene-expression para ejemplos de tejidos.</p>
<p><code>--cell_type</code><br />
Str o lista de tipo(s) de c√©lula(s), p. ej. ['mucus secreting cell', 'neuroendocrine cell']. Por defecto: None.<br />
Ver https://cellxgene.cziscience.com/gene-expression y seleccione un tejido para ejemplos de tipos de c√©lulas.</p>
<p><code>--development_stage</code><br />
Str o lista de etapa(s) de desarrollo. Por defecto: None.</p>
<p><code>--disease</code><br />
Str o lista de enfermedad(es). Por defecto: None.</p>
<p><code>--sex</code><br />
Str o lista de sexo(s), p. ej. 'female' (femenina). Por defecto: None.</p>
<p><code>--dataset_id</code><br />
Str o lista de CELLxGENE ID(s). Por defecto: None.</p>
<p><code>--tissue_general_ontology_term_id</code><br />
Str o lista de tejido(s) del tipo high-level UBERON ID. Por defecto: None.<br />
Tejidos y sus IDs tipo UBERON se enumeran <a href="https://github.com/chanzuckerberg/single-cell-data-portal/blob/9b94ccb0a2e0a8f6182b213aa4852c491f6f6aff/backend/wmg/data/tissue_mapper.py">aqu√≠</a>.</p>
<p><code>--tissue_general</code><br />
Str o lista de tejido(s) del tipo high-level. Por defecto: None.<br />
Tejidos y sus IDs de UBERON se enumeran <a href="https://github.com/chanzuckerberg/single-cell-data-portal/blob/9b94ccb0a2e0a8f6182b213aa4852c491f6f6aff/backend/wmg/data/tissue_mapper.py">aqu√≠</a>.</p>
<p><code>--tissue_ontology_term_id</code><br />
Str o lista de ID(s) de 'tissue ontology term' como est√°n definidos en el <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">esquema de datos del CELLxGENE</a>. Por defecto: None.</p>
<p><code>--assay_ontology_term_id</code><br />
Str o lista de ID(s) de 'assay ontology term' como est√°n definidos en el <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">esquema de datos del CELLxGENE</a>. Por defecto: None.</p>
<p><code>--assay</code><br />
Str o lista de 'assays' (m√©todos) como est√°n definidos en el <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">esquema de datos del CELLxGENE</a>. Por defecto: None.</p>
<p><code>--cell_type_ontology_term_id</code><br />
Str o lista de ID(s) de 'celltype ontology term' como est√°n definidos en el <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">esquema de datos del CELLxGENE</a>. Por defecto: None.</p>
<p><code>--development_stage_ontology_term_id</code><br />
Str o lista de ID(s) de 'development stage ontology term' como est√°n definidos en el <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">esquema de datos del CELLxGENE</a>. Por defecto: None.</p>
<p><code>--disease_ontology_term_id</code><br />
Str o lista de ID(s) de 'disease ontology term' como est√°n definidos en el <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">esquema de datos del CELLxGENE</a>. Por defecto: None.</p>
<p><code>--donor_id</code><br />
Str o lista de ID(s) de 'donor' (donador) como est√°n definidos en el <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">esquema de datos del CELLxGENE</a>. Por defecto: None.</p>
<p><code>--self_reported_ethnicity_ontology_term_id</code><br />
Str o lista de ID(s) de 'self-reported ethnicity ontology' como est√°n definidos en el <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">esquema de datos del CELLxGENE</a>. Por defecto: None.</p>
<p><code>--self_reported_ethnicity</code><br />
Str o lista de etnias autoinformadas como est√°n definidas en el <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">esquema de datos del CELLxGENE</a>. Por defecto: None.</p>
<p><code>--sex_ontology_term_id</code><br />
Str o lista de ID(s) de 'sex ontology' como est√°n definidos en el <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">esquema de datos del CELLxGENE</a>. Por defecto: None.</p>
<p><code>--suspension_type</code><br />
Str o lista de tipo(s) de suspensi√≥n como est√°n definidos en el <a href="https://github.com/chanzuckerberg/single-cell-curation/tree/main/schema">esquema de datos del CELLxGENE</a>. Por defecto: None.</p>
<h3 id="ejemplo-3"><a class="header" href="#ejemplo-3">Ejemplo</a></h3>
<pre><code class="language-bash">gget cellxgene --gene ACE2 ABCA1 SLC5A1 --tissue lung --cell_type 'mucus secreting cell' 'neuroendocrine cell' -o example_adata.h5ad
</code></pre>
<pre><code class="language-python"># Python
adata = gget.cellxgene(
    gene = [&quot;ACE2&quot;, &quot;ABCA1&quot;, &quot;SLC5A1&quot;],
    tissue = &quot;lung&quot;,
    cell_type = [&quot;mucus secreting cell&quot;, &quot;neuroendocrine cell&quot;]
)
adata
</code></pre>
<p>‚Üí Produce un objeto AnnData que contiene la matriz de recuentos de scRNAseq de los genes ACE2, ABCA1 y SLC5A1 en 3322 c√©lulas secretoras de mucosidad y neuroendocrinas pulmonares humanas y sus metadatos correspondientes.</p>
<p><br/><br/></p>
<p>Obtiene solo los metadatos (corresponde a AnnData.obs):</p>
<pre><code class="language-bash">gget cellxgene --meta_only --gene ENSMUSG00000015405 --ensembl --tissue lung --species mus_musculus -o example_meta.csv
</code></pre>
<pre><code class="language-python"># Python
df = gget.cellxgene(
    meta_only = True,
    gene = &quot;ENSMUSG00000015405&quot;,
    ensembl = True,
    tissue = &quot;lung&quot;,  
    species = &quot;mus_musculus&quot;
)
df
</code></pre>
<p>‚Üí Produce solo los metadatos de los conjuntos de datos de ENSMUSG00000015405 (ACE2), los cuales corresponden a c√©lulas pulmonares murinas.</p>
<p>Ver tambi√©n: <a href="https://chanzuckerberg.github.io/cellxgene-census/notebooks/api_demo/census_gget_demo.html">https://chanzuckerberg.github.io/cellxgene-census/notebooks/api_demo/census_gget_demo.html</a></p>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Las banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-cosmic--1"><a class="header" href="#gget-cosmic--1">gget cosmic ü™ê</a></h2>
<p>Busque genes, mutaciones, etc. asociados con c√°nceres utilizando la base de datos <a href="https://cancer.sanger.ac.uk/cosmic">COSMIC</a> (Cat√°logo de mutaciones som√°ticas en c√°ncer).<br />
Produce: Resultados en formato JSON (Terminal) o Dataframe/CSV (Python).<br />
<code>gget cosmic</code> fue escrito por <a href="https://github.com/AubakirovArman">@AubakirovArman</a>.</p>
<p><strong>Par√°metro posicional</strong><br />
<code>searchterm</code><br />
T√©rmino de b√∫squeda. Puede ser una mutaci√≥n, un gen (o ID de Ensembl), una muestra, etc., tal como se define con el argumento <code>entity</code>. Ejemplo: 'EGFR'</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-e</code> <code>--entity</code><br />
'mutations', 'genes', 'cancer', 'tumour site', 'studies', 'pubmed', o 'samples'. Por defecto: 'mutations'.<br />
Define el tipo de t√©rmino de b√∫squeda (<code>searchterm</code>).</p>
<p><code>-l</code> <code>--limit</code><br />
Limita el n√∫mero de resultados producidos. Por defecto: 100.</p>
<p><code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados, p. ej. ruta/al/directorio/resultados.csv (o .json). Por defecto: salida est√°ndar (STDOUT).<br />
Para Python, usa <code>save=True</code> para guardar los resultados en el directorio de trabajo actual.</p>
<p><strong>Banderas</strong><br />
<code>-csv</code> <code>--csv</code>
Solo para Terminal. Produce los resultados en formato CSV.<br />
Para Python, usa json=True para producir los resultados en formato JSON.</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para Terminal. Impide la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.<br />
Para Python, usa <code>verbose=False</code> para imipidir la informacion de progreso de ser exhibida durante la ejecuci√≥n del programa.</p>
<h3 id="por-ejemplo-1"><a class="header" href="#por-ejemplo-1">Por ejemplo</a></h3>
<pre><code class="language-bash">gget cosmic -e genes EGFR
</code></pre>
<pre><code class="language-python"># Python
gget.cosmic(&quot;EGFR&quot;, entity=&quot;genes&quot;)
</code></pre>
<p>‚Üí Produce los resultados COSMIC para el gen 'EGFR':</p>
<div class="table-wrapper"><table><thead><tr><th>Gene</th><th>Alternate IDs</th><th>Tested samples</th><th>Simple Mutations</th><th>Fusions</th><th>Coding Mutations</th><th>...</th></tr></thead><tbody>
<tr><td>EGFR</td><td>EGFR,ENST00000275493.6,...</td><td>210280</td><td>31900</td><td>0</td><td>31900</td><td>...</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td></tr>
</tbody></table>
</div><div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Las banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-diamond--1"><a class="header" href="#gget-diamond--1">gget diamond üíé</a></h2>
<p>Alinee m√∫ltiples prote√≠nas o secuencias de ADN traducidas usando <a href="https://www.nature.com/articles/nmeth.3176">DIAMOND</a> (DIAMOND es similar a BLAST, pero este es un c√°lculo local).<br />
Produce: Resultados en formato JSON (Terminal) o Dataframe/CSV (Python).</p>
<p><strong>Par√°metro posicional</strong><br />
<code>query</code><br />
Secuencia(s) (str o lista) de amino√°cidos, o una ruta a un archivo tipo FASTA.</p>
<p><strong>Par√°metro requerido</strong><br />
<code>-ref</code> <code>--reference</code><br />
Secuencias de amino√°cidos de referencia (str o lista), o una ruta a un archivo tipo FASTA.</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-db</code> <code>--diamond_db</code><br />
Ruta para guardar la base de datos DIAMOND creada a partir de <code>reference</code> (str).<br />
Por defecto: None -&gt; El archivo de base de datos DIAMOND temporal se eliminar√° despu√©s de la alineaci√≥n o se guardar√° en <code>out</code> si se proporciona <code>out</code>.</p>
<p><code>-s</code> <code>--sensitivity</code><br />
Sensibilidad de la alineaci√≥n (str). Por defecto: &quot;very-sensitive&quot; (muy sensible).<br />
Uno de los siguientes: fast, mid-sensitive, sensitive, more-sensitive, very-sensitive, or ultra-sensitive.</p>
<p><code>-t</code> <code>--threads</code><br />
N√∫mero de hilos de procesamiento utilizados (int). Por defecto: 1.</p>
<p><code>-db</code> <code>--diamond_binary</code><br />
Ruta al binario DIAMOND (str). Por defecto: None -&gt; Utiliza el binario DIAMOND instalado autom√°ticamente con <code>gget</code>.</p>
<p><code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados (str), p. ej. &quot;ruta/al/directorio&quot;. Por defecto: salida est√°ndar (STDOUT); los archivos temporales se eliminan.</p>
<p><strong>Banderas</strong><br />
<code>-u</code> <code>--uniprot</code><br />
Use esta bandera cuando <code>sequence</code> es un ID de Uniprot en lugar de una secuencia de amino√°cidos.</p>
<p><code>-csv</code> <code>--csv</code><br />
Solo para Terminal. Produce los resultados en formato CSV.<br />
Para Python, usa <code>json=True</code> para producir los resultados en formato JSON.</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para Terminal. Impide la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.<br />
Para Python, usa <code>verbose=False</code> para impedir la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.</p>
<h3 id="ejemplo-4"><a class="header" href="#ejemplo-4">Ejemplo</a></h3>
<pre><code class="language-bash"># !!! Aseg√∫rese de enumerar primero el argumento posicional aqu√≠ para que no se agregue como secuencia de referencia
gget diamond GGETISAWESQME ELVISISALIVE LQVEFRANKLIN PACHTERLABRQCKS -ref GGETISAWESQMEELVISISALIVELQVEFRANKLIN PACHTERLABRQCKS
</code></pre>
<pre><code class="language-python"># Python
gget.diamond([&quot;GGETISAWESQME&quot;, &quot;ELVISISALIVE&quot;, &quot;LQVEFRANKLIN&quot;, &quot;PACHTERLABRQCKS&quot;], reference=[&quot;GGETISAWESQMEELVISISALIVELQVEFRANKLIN&quot;, &quot;PACHTERLABRQCKS&quot;])
</code></pre>
<p>‚Üí Produce los resultados de la alineaci√≥n en formato JSON (Terminal) o Dataframe/CSV:</p>
<div class="table-wrapper"><table><thead><tr><th>query_accession</th><th>subject_accession</th><th>identity_percentage</th><th>query_seq_length</th><th>subject_seq_length</th><th>length</th><th>mismatches</th><th>gap_openings</th><th>query_start</th><th>query_end</th><th>subject_start</th><th>subject_end</th><th>e-value</th><th>bit_score</th></tr></thead><tbody>
<tr><td>Seq0</td><td>Seq0</td><td>100</td><td>13</td><td>37</td><td>13</td><td>0</td><td>0</td><td>1</td><td>13</td><td>1</td><td>13</td><td>2.82e-09</td><td>30.8</td></tr>
<tr><td>Seq2</td><td>Seq0</td><td>100</td><td>12</td><td>37</td><td>12</td><td>0</td><td>0</td><td>1</td><td>12</td><td>26</td><td>37</td><td>4.35e-08</td><td>27.7</td></tr>
<tr><td>Seq3</td><td>Seq1</td><td>100</td><td>15</td><td>15</td><td>15</td><td>0</td><td>0</td><td>1</td><td>15</td><td>1</td><td>15</td><td>2.01e-11</td><td>36.2</td></tr>
</tbody></table>
</div>
<h4 id="m√†s-ejemplos-1"><a class="header" href="#m√†s-ejemplos-1"><a href="https://github.com/pachterlab/gget_examples">M√†s ejemplos</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-elm--1"><a class="header" href="#gget-elm--1">gget elm üé≠</a></h2>
<p>Prediga localmente motivos lineales eucari√≥ticos (ELMs) a partir de una secuencia de amino√°cidos o UniProt Acc utilizando datos de la <a href="http://elm.eu.org/">base de datos ELM</a>. Los datos de ELM se pueden descargar y distribuir para uso no comercial de acuerdo con el <a href="http://elm.eu.org/media/Elm_academic_license.pdf">Acuerdo de licencia de software de ELM</a>.<br />
Produce: Resultados en formato JSON (Terminal) o Dataframe/CSV (Python). Este m√≥dulo devuelve dos tipos de resultados (ver ejemplos).</p>
<p>Antes de usar <code>gget elm</code> por primera vez, ejecute <code>gget setup elm</code> / <code>gget.setup(&quot;elm&quot;)</code> una vez (consulte tambi√©n <a href="es/setup.html"><code>gget setup</code></a>).</p>
<p><strong>Par√°metro posicional</strong><br />
<code>sequence</code><br />
Secuencia de amino√°cidos o Uniprot Acc (str).<br />
Al proporcionar una Uniprot Acc, use la bandera <code>--uniprot</code> (Python: <code>uniprot=True</code>).</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-s</code> <code>sensitivity</code><br />
Sensibilidad de la alineaci√≥n DIAMOND (str). Por defecto: &quot;very-sensitive&quot; (muy sensible).<br />
Uno de los siguientes: fast, mid-sensitive, sensitive, more-sensitive, very-sensitive, or ultra-sensitive.</p>
<p><code>-t</code> <code>threads</code><br />
N√∫mero de hilos de procesamiento utilizados en la alineaci√≥n de secuencias con DIAMOND (int). Por defecto: 1.</p>
<p><code>-bin</code> <code>diamond_binary</code><br />
Ruta al binario DIAMOND (str). Por defecto: None -&gt; Utiliza el binario DIAMOND instalado autom√°ticamente con <code>gget</code>.</p>
<p><code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados (str), p. ej. &quot;ruta/al/directorio&quot;. Por defecto: salida est√°ndar (STDOUT); los archivos temporales se eliminan.</p>
<p><strong>Banderas</strong><br />
<code>-u</code> <code>--uniprot</code><br />
Use esta bandera cuando <code>sequence</code> es una Uniprot Acc en lugar de una secuencia de amino√°cidos.</p>
<p><code>-e</code> <code>--expand</code><br />
Ampl√≠e la informaci√≥n devuelta en el marco de datos de expresiones regulares para incluir los nombres de prote√≠nas, los organismos y las referencias en las que se valid√≥ originalmente el motivo.</p>
<p><code>-csv</code> <code>--csv</code><br />
Solo para Terminal. Produce los resultados en formato CSV.<br />
Para Python, usa <code>json=True</code> para producir los resultados en formato JSON.</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para Terminal. Impide la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.<br />
Para Python, usa <code>verbose=False</code> para impedir la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.</p>
<h3 id="ejemplo-5"><a class="header" href="#ejemplo-5">Ejemplo</a></h3>
<p>Encuentre ELM en una secuencia de amino√°cidos:</p>
<pre><code class="language-bash">gget setup elm          # Descarga/actualiza la base de datos ELM local
gget elm -o gget_elm_results LIAQSIGQASFV
</code></pre>
<pre><code class="language-python"># Python
gget.setup(‚Äúelm‚Äù)      # Descarga/actualiza la base de datos ELM local
ortholog_df, regex_df = gget.elm(&quot;LIAQSIGQASFV&quot;)
</code></pre>
<p>Encuentre ELM que proporcionen a una UniProt Acc: </p>
<pre><code class="language-bash">gget setup elm          # Descarga/actualiza la base de datos ELM local
gget elm -o gget_elm_results --uniprot Q02410 -e
</code></pre>
<pre><code class="language-python"># Python
gget.setup(‚Äúelm‚Äù)      # Descarga/actualiza la base de datos ELM local
ortholog_df, regex_df = gget.elm(&quot;Q02410&quot;, uniprot=True, expand=True)
</code></pre>
<p>‚Üí Produce dos resultados con informaci√≥n extensa sobre ELMs asociados con prote√≠nas ort√≥logas y motivos encontrados en la secuencia de entrada directamente en funci√≥n de sus expresiones regex:</p>
<p>ortholog_df:</p>
<div class="table-wrapper"><table><thead><tr><th style="text-align: center">Ortholog_UniProt_Acc</th><th style="text-align: center">ProteinName</th><th style="text-align: center">class_accession</th><th style="text-align: center">ELMIdentifier</th><th style="text-align: center">FunctionalSiteName</th><th style="text-align: center">Description</th><th style="text-align: center">Organism</th><th style="text-align: center">‚Ä¶</th></tr></thead><tbody>
<tr><td style="text-align: center">Q02410</td><td style="text-align: center">APBA1_HUMAN</td><td style="text-align: center">ELME000357</td><td style="text-align: center">LIG_CaMK_CASK_1</td><td style="text-align: center">CASK CaMK domain binding ligand motif</td><td style="text-align: center">Motif that mediates binding to the calmodulin-dependent protein kinase (CaMK) domain of the peripheral plasma membrane protein CASK/Lin2.</td><td style="text-align: center">Homo sapiens</td><td style="text-align: center">‚Ä¶</td></tr>
<tr><td style="text-align: center">Q02410</td><td style="text-align: center">APBA1_HUMAN</td><td style="text-align: center">ELME000091</td><td style="text-align: center">LIG_PDZ_Class_2</td><td style="text-align: center">PDZ domain ligands</td><td style="text-align: center">The C-terminal class 2 PDZ-binding motif is classically represented by a pattern such as</td><td style="text-align: center">Homo sapiens</td><td style="text-align: center">‚Ä¶</td></tr>
</tbody></table>
</div>
<p>regex_df:</p>
<div class="table-wrapper"><table><thead><tr><th style="text-align: center">Instance_accession</th><th style="text-align: center">ELMIdentifier</th><th style="text-align: center">FunctionalSiteName</th><th style="text-align: center">ELMType</th><th style="text-align: center">Description</th><th style="text-align: center">Instances (Matched Sequence)</th><th style="text-align: center">Organism</th><th style="text-align: center">‚Ä¶</th></tr></thead><tbody>
<tr><td style="text-align: center">ELME000321</td><td style="text-align: center">CLV_C14_Caspase3-7</td><td style="text-align: center">Caspase cleavage motif</td><td style="text-align: center">CLV</td><td style="text-align: center">Caspase-3 and Caspase-7 cleavage site.</td><td style="text-align: center">ERSDG</td><td style="text-align: center">Mus musculus</td><td style="text-align: center">‚Ä¶</td></tr>
<tr><td style="text-align: center">ELME000102</td><td style="text-align: center">CLV_NRD_NRD_1</td><td style="text-align: center">NRD cleavage site</td><td style="text-align: center">CLV</td><td style="text-align: center">N-Arg dibasic convertase (NRD/Nardilysin) cleavage site.</td><td style="text-align: center">RRA</td><td style="text-align: center">Rattus norvegicus</td><td style="text-align: center">‚Ä¶</td></tr>
<tr><td style="text-align: center">ELME000100</td><td style="text-align: center">CLV_PCSK_PC1ET2_1</td><td style="text-align: center">PCSK cleavage site</td><td style="text-align: center">CLV</td><td style="text-align: center">NEC1/NEC2 cleavage site.</td><td style="text-align: center">KRD</td><td style="text-align: center">Mus musculus</td><td style="text-align: center">‚Ä¶</td></tr>
<tr><td style="text-align: center">ELME000146</td><td style="text-align: center">CLV_PCSK_SKI1_1</td><td style="text-align: center">PCSK cleavage site</td><td style="text-align: center">CLV</td><td style="text-align: center">Subtilisin/kexin isozyme-1 (SKI1) cleavage site.</td><td style="text-align: center">RLLTA</td><td style="text-align: center">Homo sapiens</td><td style="text-align: center">‚Ä¶</td></tr>
<tr><td style="text-align: center">ELME000231</td><td style="text-align: center">DEG_APCC_DBOX_1</td><td style="text-align: center">APCC-binding Destruction motifs</td><td style="text-align: center">DEG</td><td style="text-align: center">An RxxL-based motif that binds to the Cdh1 and Cdc20 components of APC/C thereby targeting the protein for destruction in a cell cycle dependent manner</td><td style="text-align: center">SRVKLNIVR</td><td style="text-align: center">Saccharomyces cerevisiae S288c</td><td style="text-align: center">‚Ä¶</td></tr>
<tr><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td><td style="text-align: center">‚Ä¶</td></tr>
</tbody></table>
</div>
<p>(Los motivos que aparecen en muchas especies diferentes pueden parecer repetidos, pero todas las filas deben ser √∫nicas.)</p>
<h4 id="m√†s-ejemplos-2"><a class="header" href="#m√†s-ejemplos-2"><a href="https://github.com/pachterlab/gget_examples">M√†s ejemplos</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Las banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-enrichr--1"><a class="header" href="#gget-enrichr--1">gget enrichr üí∞</a></h2>
<p>Realice un an√°lisis de enriquecimiento de una lista de genes utilizando <a href="https://maayanlab.cloud/Enrichr/">Enrichr</a>.<br />
Produce: Resultados en formato JSON (Terminal) o Dataframe/CSV (Python).</p>
<p><strong>Par√°metro posicional</strong><br />
<code>genes</code><br />
Lista de nombres cortos (s√≠mbolos) de los genes de inter√©s para realizar el an√°lisis de enriquecimiento, p. PHF14 RBM3 MSL1 PHF21A.<br />
Alternativamente: use la bandera <code>--ensembl</code> para ingresar IDs tipo Ensembl, p. ENSG00000106443 ENSG00000102317 ENSG00000188895.</p>
<p><strong>Otros par√°metros requeridos</strong><br />
<code>-db</code> <code>--database</code><br />
Base de datos que ser√° utilizada como referencia para el an√°lisis de enriquecimiento.<br />
Admite cualquier base de datos enumerada <a href="https://maayanlab.cloud/Enrichr/#libraries">aqu√≠</a> o uno de los siguientes accesos directos:<br />
'pathway'‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ (KEGG_2021_Human)<br />
'transcription'‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ (ChEA_2016)<br />
'ontology'‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ(GO_Biological_Process_2021)<br />
'diseases_drugs'‚ÄÉ‚ÄÉ‚ÄÉ(GWAS_Catalog_2019)<br />
'celltypes'‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ‚ÄÉ(PanglaoDB_Augmented_2021)<br />
'kinase_interactions'‚ÄÉ  (KEA_2015)</p>
<p><strong>Par√°metros opcionales</strong><br />
<code>-bkg_l</code> <code>--background_list</code><br />
Lista de nombres cortos (s√≠mbolos) de genes de 'background' (de fondo/control), p. NSUN3 POLRMT NLRX1.<br />
Alternativamente: usa la bandera <code>--ensembl_background</code> para ingresar IDs tipo Ensembl.</p>
<p><code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados, p. ruta/al/directorio/resultados.csv (o .json). Por defecto: salida est√°ndar (STDOUT).<br />
Para Python, usa <code>save=True</code> para guardar los resultados en el directorio de trabajo actual.</p>
<p><code>-ko</code> <code>--kegg_out</code><br />
Ruta al archivo png en el que se guardar√° la im√°gen de la v√≠a de se√±alizaci√≥n celular KEGG, p. ej. ruta/al/directorio/KEGG.png. (Por defecto: None)</p>
<p><code>-kr</code> <code>--kegg_rank</code><br />
Rango de la ruta KEGG que se va a trazar. (Por defecto: 1)</p>
<p><code>figsize</code><br />
Solo para Python. (ancho, alto) de la visualizaci√≥n en pulgadas. (Por defecto: (10,10))</p>
<p><code>ax</code><br />
Solo para Python. Ingresa un objeto de ejes matplotlib para personalizar la visualizaci√≥n.(Por defecto: None)</p>
<p><strong>Banderas</strong><br />
<code>-e</code> <code>--ensembl</code><br />
Usa esta bandera si <code>genes</code> se ingresa como una lista de IDs tipo Ensembl.</p>
<p><code>-e_b</code> <code>--ensembl_background</code><br />
Usa esta bandera si <code>background_list</code> se ingresa como una lista de IDs tipo Ensembl.</p>
<p><code>-bkg</code> <code>--background</code><br />
Use un conjunto de 20,625 genes 'background' 
listados <a href="https://github.com/pachterlab/gget/blob/main/gget/constants/enrichr_bkg_genes.txt">aqu√≠</a>.</p>
<p><code>-csv</code> <code>--csv</code><br />
Solo para Terminal. Produce los resultados en formato CSV.<br />
Para Python, usa <code>json=True</code> produce los resultados en formato JSON.</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para Terminal. Impide la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.<br />
Para Python, usa <code>verbose=False</code> para imipidir la informaci√≥n de progreso de ser exhibida durante la ejecuci√≥n del programa.</p>
<p><code>plot</code><br />
Solo para Python. <code>plot=True</code> prov√©e la visualizaci√≥n de los primeros 15 resultados (por defecto: False).</p>
<h3 id="ejemplo-6"><a class="header" href="#ejemplo-6">Ejemplo</a></h3>
<pre><code class="language-bash">gget enrichr -db ontology ACE2 AGT AGTR1
</code></pre>
<pre><code class="language-python"># Python
gget.enrichr([&quot;ACE2&quot;, &quot;AGT&quot;, &quot;AGTR1&quot;], database=&quot;ontology&quot;, plot=True)
</code></pre>
<p>‚Üí Produce v√≠as/funciones celulares relacionadas con los genes ACE2, AGT y AGTR1 de la base de datos <em>GO Biological Process 2021</em>. En Python, <code>plot=True</code> provee la visualizaci√≥n de resultados:</p>
<p><img src="https://github.com/pachterlab/gget/blob/main/figures/gget_enrichr_results.png?raw=true" alt="alt text" /></p>
<p><br/><br/></p>
<p><strong>Use <code>gget enrichr</code> con una lista de genes 'background':</strong></p>
<pre><code class="language-bash"># Aqu√≠, primero ingresamos los genes de inter√©s (par√°metro posicional 'genes'), para que no se agreguen a la lista de genes 'background' detr√°s del par√°metro '-bkgr_l'
gget enrichr \
	PHF14 RBM3 MSL1 PHF21A ARL10 INSR JADE2 P2RX7 LINC00662 CCDC101 PPM1B KANSL1L CRYZL1 ANAPC16 TMCC1 CDH8 RBM11 CNPY2 HSPA1L CUL2 PLBD2 LARP7 TECPR2 ZNF302 CUX1 MOB2 CYTH2 SEC22C EIF4E3 ROBO2 ADAMTS9-AS2 CXXC1 LINC01314 ATF7 ATP5F1 \
	-db ChEA_2022 \
	-bkg_l NSUN3 POLRMT NLRX1 SFXN5 ZC3H12C SLC25A39 ARSG DEFB29 PCMTD2 ACAA1A LRRC1 2810432D09RIK SEPHS2 SAC3D1 TMLHE LOC623451 TSR2 PLEKHA7 GYS2 ARHGEF12 HIBCH LYRM2 ZBTB44 ENTPD5 RAB11FIP2 LIPT1 INTU ANXA13 KLF12 SAT2 GAL3ST2 VAMP8 FKBPL AQP11 TRAP1 PMPCB TM7SF3 RBM39 BRI3 KDR ZFP748 NAP1L1 DHRS1 LRRC56 WDR20A STXBP2 KLF1 UFC1 CCDC16 9230114K14RIK RWDD3 2610528K11RIK ACO1 CABLES1 LOC100047214 YARS2 LYPLA1 KALRN GYK ZFP787 ZFP655 RABEPK ZFP650 4732466D17RIK EXOSC4 WDR42A GPHN 2610528J11RIK 1110003E01RIK MDH1 1200014M14RIK AW209491 MUT 1700123L14RIK 2610036D13RIK PHF14 RBM3 MSL1 PHF21A ARL10 INSR JADE2 P2RX7 LINC00662 CCDC101 PPM1B KANSL1L CRYZL1 ANAPC16 TMCC1 CDH8 RBM11 CNPY2 HSPA1L CUL2 PLBD2 LARP7 TECPR2 ZNF302 CUX1 MOB2 CYTH2 SEC22C EIF4E3 ROBO2 ADAMTS9-AS2 CXXC1 LINC01314 ATF7 ATP5F1COX15 TMEM30A NSMCE4A TM2D2 RHBDD3 ATXN2 NFS1 3110001I20RIK BC038156 C330002I19RIK ZFYVE20 POLI TOMM70A LOC100047782 2410012H22RIK RILP A230062G08RIK PTTG1IP RAB1 AFAP1L1 LYRM5 2310026E23RIK SLC7A6OS MAT2B 4932438A13RIK LRRC8A SMO NUPL2
</code></pre>
<pre><code class="language-python"># Python
gget.enrichr(
	genes = [
		&quot;PHF14&quot;, &quot;RBM3&quot;, &quot;MSL1&quot;, &quot;PHF21A&quot;, &quot;ARL10&quot;, &quot;INSR&quot;, &quot;JADE2&quot;, &quot;P2RX7&quot;,
		&quot;LINC00662&quot;, &quot;CCDC101&quot;, &quot;PPM1B&quot;, &quot;KANSL1L&quot;, &quot;CRYZL1&quot;, &quot;ANAPC16&quot;, &quot;TMCC1&quot;,
		&quot;CDH8&quot;, &quot;RBM11&quot;, &quot;CNPY2&quot;, &quot;HSPA1L&quot;, &quot;CUL2&quot;, &quot;PLBD2&quot;, &quot;LARP7&quot;, &quot;TECPR2&quot;, 
		&quot;ZNF302&quot;, &quot;CUX1&quot;, &quot;MOB2&quot;, &quot;CYTH2&quot;, &quot;SEC22C&quot;, &quot;EIF4E3&quot;, &quot;ROBO2&quot;,
		&quot;ADAMTS9-AS2&quot;, &quot;CXXC1&quot;, &quot;LINC01314&quot;, &quot;ATF7&quot;, &quot;ATP5F1&quot;
	], 
	database = &quot;ChEA_2022&quot;,
	background_list = [
		&quot;NSUN3&quot;,&quot;POLRMT&quot;,&quot;NLRX1&quot;,&quot;SFXN5&quot;,&quot;ZC3H12C&quot;,&quot;SLC25A39&quot;,&quot;ARSG&quot;,
		&quot;DEFB29&quot;,&quot;PCMTD2&quot;,&quot;ACAA1A&quot;,&quot;LRRC1&quot;,&quot;2810432D09RIK&quot;,&quot;SEPHS2&quot;,
		&quot;SAC3D1&quot;,&quot;TMLHE&quot;,&quot;LOC623451&quot;,&quot;TSR2&quot;,&quot;PLEKHA7&quot;,&quot;GYS2&quot;,&quot;ARHGEF12&quot;,
		&quot;HIBCH&quot;,&quot;LYRM2&quot;,&quot;ZBTB44&quot;,&quot;ENTPD5&quot;,&quot;RAB11FIP2&quot;,&quot;LIPT1&quot;,
		&quot;INTU&quot;,&quot;ANXA13&quot;,&quot;KLF12&quot;,&quot;SAT2&quot;,&quot;GAL3ST2&quot;,&quot;VAMP8&quot;,&quot;FKBPL&quot;,
		&quot;AQP11&quot;,&quot;TRAP1&quot;,&quot;PMPCB&quot;,&quot;TM7SF3&quot;,&quot;RBM39&quot;,&quot;BRI3&quot;,&quot;KDR&quot;,&quot;ZFP748&quot;,
		&quot;NAP1L1&quot;,&quot;DHRS1&quot;,&quot;LRRC56&quot;,&quot;WDR20A&quot;,&quot;STXBP2&quot;,&quot;KLF1&quot;,&quot;UFC1&quot;,
		&quot;CCDC16&quot;,&quot;9230114K14RIK&quot;,&quot;RWDD3&quot;,&quot;2610528K11RIK&quot;,&quot;ACO1&quot;,
		&quot;CABLES1&quot;, &quot;LOC100047214&quot;,&quot;YARS2&quot;,&quot;LYPLA1&quot;,&quot;KALRN&quot;,&quot;GYK&quot;,
		&quot;ZFP787&quot;,&quot;ZFP655&quot;,&quot;RABEPK&quot;,&quot;ZFP650&quot;,&quot;4732466D17RIK&quot;,&quot;EXOSC4&quot;,
		&quot;WDR42A&quot;,&quot;GPHN&quot;,&quot;2610528J11RIK&quot;,&quot;1110003E01RIK&quot;,&quot;MDH1&quot;,&quot;1200014M14RIK&quot;,
		&quot;AW209491&quot;,&quot;MUT&quot;,&quot;1700123L14RIK&quot;,&quot;2610036D13RIK&quot;,
		&quot;PHF14&quot;, &quot;RBM3&quot;, &quot;MSL1&quot;, &quot;PHF21A&quot;, &quot;ARL10&quot;, &quot;INSR&quot;, &quot;JADE2&quot;, 
		&quot;P2RX7&quot;, &quot;LINC00662&quot;, &quot;CCDC101&quot;, &quot;PPM1B&quot;, &quot;KANSL1L&quot;, &quot;CRYZL1&quot;, 
		&quot;ANAPC16&quot;, &quot;TMCC1&quot;,&quot;CDH8&quot;, &quot;RBM11&quot;, &quot;CNPY2&quot;, &quot;HSPA1L&quot;, &quot;CUL2&quot;, 
		&quot;PLBD2&quot;, &quot;LARP7&quot;, &quot;TECPR2&quot;, &quot;ZNF302&quot;, &quot;CUX1&quot;, &quot;MOB2&quot;, &quot;CYTH2&quot;, 
		&quot;SEC22C&quot;, &quot;EIF4E3&quot;, &quot;ROBO2&quot;, &quot;ADAMTS9-AS2&quot;, &quot;CXXC1&quot;, &quot;LINC01314&quot;, &quot;ATF7&quot;, 
		&quot;ATP5F1&quot;&quot;COX15&quot;,&quot;TMEM30A&quot;,&quot;NSMCE4A&quot;,&quot;TM2D2&quot;,&quot;RHBDD3&quot;,&quot;ATXN2&quot;,&quot;NFS1&quot;,
		&quot;3110001I20RIK&quot;,&quot;BC038156&quot;,&quot;C330002I19RIK&quot;,&quot;ZFYVE20&quot;,&quot;POLI&quot;,&quot;TOMM70A&quot;,
		&quot;LOC100047782&quot;,&quot;2410012H22RIK&quot;,&quot;RILP&quot;,&quot;A230062G08RIK&quot;,
		&quot;PTTG1IP&quot;,&quot;RAB1&quot;,&quot;AFAP1L1&quot;, &quot;LYRM5&quot;,&quot;2310026E23RIK&quot;,
		&quot;SLC7A6OS&quot;,&quot;MAT2B&quot;,&quot;4932438A13RIK&quot;,&quot;LRRC8A&quot;,&quot;SMO&quot;,&quot;NUPL2&quot;
	],
	plot=True
)
</code></pre>
<p>‚Üí Prov√©e factores de transcripci√≥n relacionados a los genes de inter√©s y controlados con la lista de genes background de la base de datos <em>ChEA 2022</em>. En Python, <code>plot=True</code> permite la visualizaci√≥n de resultados:</p>
<p><img src="https://github.com/pachterlab/gget/blob/main/figures/gget_enrichr_results_2.png?raw=true" alt="alt text" /></p>
<p><br/><br/></p>
<p><strong>Genere una imagen de la v√≠a de se√±alizaci√≥n de c√©lulas KEGG con los genes del an√°lisis de enriquecimiento resaltados:</strong><br />
Esta funci√≥n est√° disponible gracias a un <a href="https://github.com/pachterlab/gget/pull/106">PR</a> de <a href="https://github.com/noriakis">Noriaki Sato</a>.</p>
<pre><code class="language-bash">gget enrichr -db pathway --kegg_out kegg.png --kegg_rank 1 ZBP1 IRF3 RIPK1
</code></pre>
<pre><code class="language-python"># Python
gget.enrichr([&quot;ZBP1&quot;, &quot;IRF3&quot;, &quot;RIPK1&quot;], database=&quot;pathway&quot;, kegg_out=&quot;kegg.png&quot;, kegg_rank=1)
</code></pre>
<p>‚Üí Adem√°s de los resultados est√°ndar <code>gget enrichr</code>, el argumento <code>kegg_out</code> guarda una imagen con los genes del an√°lisis de enriquecimiento resaltados guardado como kegg.png:</p>
<p><img src="https://github.com/pachterlab/gget/assets/56094636/b0aa5a64-69d0-4a6a-85db-3e7baf9cb2a4" alt="kegg" /></p>
<p><br/><br/></p>
<p>El siguiente ejemplo fue enviado por <a href="https://github.com/DylanLawless">Dylan Lawless</a> a trav√©s de un <a href="https://github.com/pachterlab/gget/pull/54">PR</a> (con ajustes de <a href="https://github.com/lauraluebbert">Laura Luebbert</a>):<br />
<strong>Use <code>gget enrichr</code> en R y cree unq visualizaci√≥n similar usando <a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a>.</strong><br />
TENGA EN CUENTA el cambio de ejes en comparaci√≥n con la visualizaci√≥n en Python. </p>
<pre><code class="language-r">system(&quot;pip install gget&quot;)
install.packages(&quot;reticulate&quot;)
library(reticulate)
gget &lt;- import(&quot;gget&quot;)

# Perform enrichment analysis on a list of genes
df &lt;- gget$enrichr(list(&quot;ACE2&quot;, &quot;AGT&quot;, &quot;AGTR1&quot;), database = &quot;ontology&quot;)

# Count number of overlapping genes
df$overlapping_genes_count &lt;- lapply(df$overlapping_genes, length) |&gt; as.numeric()

# Only keep the top 15 results
df &lt;- df[1:15, ]

# Plot
library(ggplot2)

df |&gt;
	ggplot() +
	geom_bar(aes(
		x = -log10(adj_p_val),
		y = reorder(path_name, -adj_p_val)
	),
	stat = &quot;identity&quot;,
  	fill = &quot;lightgrey&quot;,
  	width = 0.5,
	color = &quot;black&quot;) +
	geom_text(
		aes(
			y = path_name,
			x = (-log10(adj_p_val)),
			label = overlapping_genes_count
		),
		nudge_x = 0.75,
		show.legend = NA,
		color = &quot;red&quot;
	) +
  	geom_text(
		aes(
			y = Inf,
			x = Inf,
      			hjust = 1,
      			vjust = 1,
			label = &quot;# of overlapping genes&quot;
		),
		show.legend = NA,
		color = &quot;red&quot;
	) +
	geom_vline(linetype = &quot;dotted&quot;, linewidth = 1, xintercept = -log10(0.05)) +
	ylab(&quot;Pathway name&quot;) +
	xlab(&quot;-log10(adjusted P value)&quot;)
</code></pre>
<h4 id="m√°s-ejemplos-3"><a class="header" href="#m√°s-ejemplos-3"><a href="https://github.com/pachterlab/gget_examples">M√°s ejemplos</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-gpt--1"><a class="header" href="#gget-gpt--1">gget gpt üí¨</a></h2>
<p>Genera texto en lenguaje natural basado en mensaje de entrada. <code>gget gpt</code> use la API 'openai.ChatCompletion.create' de <a href="https://openai.com/">OpenAI</a>.
Este m√≥dulo, incluido su c√≥digo, documentaci√≥n y pruebas unitarias, fue escrito en parte por Chat-GTP3 de OpenAI.</p>
<p>TENGA EN CUENTA:<br />
Las llamadas a la API de OpenAI solo son 'gratuitas' durante los primeros tres meses despu√©s de generar su cuenta de OpenAI (OpenAI proporciona un cr√©dito de $5 que vence).<br />
Puede definir un l√≠mite de facturaci√≥n mensual estricto (por ejemplo, $1) <a href="https://platform.openai.com/account/billing/limits">aqu√≠</a>.<br />
Vea sus precios y preguntas frecuentes <a href="https://openai.com/pricing">aqu√≠</a>.<br />
Obtenga su clave API de OpenAI <a href="https://platform.openai.com/account/api-keys">aqu√≠</a>.</p>
<p>Regresa: El texto generado (str).</p>
<p>Antes de usar  <code>gget gpt</code> por primera vez, corre <code>gget setup gpt</code> / <code>gget.setup(&quot;gpt&quot;)</code> (ver tambi√©n <a href="es/setup.html"><code>gget setup</code></a>).</p>
<p><strong>Par√°metros posicionales</strong><br />
<code>prompt</code><br />
Mensaje de entrada basado en el cual generar texto (str).</p>
<p><code>api_key</code><br />
Su clave API de OpenAI (str) (<a href="https://platform.openai.com/account/api-keys">obtenga su clave API</a>).</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-m</code> <code>--model</code><br />
El nombre del algoritmo GPT que se usar√° para generar el texto (str). Por defecto: &quot;gpt-3.5-turbo&quot;.<br />
See https://platform.openai.com/docs/models/gpt-4 for more information on the available models.</p>
<p><code>-temp</code> <code>--temperature</code><br />
Valor entre 0 y 2 que controla el nivel de aleatoriedad y creatividad en el texto generado (float).<br />
Los valores m√°s altos resultan en un texto m√°s creativo y variado. Por defecto: 1.</p>
<p><code>-tp</code> <code>--top_p</code><br />
Controla la diversidad del texto generado como alternativa al muestreo con <code>--temperature</code> (float).<br />
Los valores m√°s altos resultan en un texto m√°s diverso e inesperado. Por defecto: 1.<br />
Tenga en cuenta que OpenAI recomienda modificar <code>--top_p</code> o el par√°metro <code>--temperature</code>, pero no ambas.</p>
<p><code>-s</code> <code>--stop</code><br />
Una secuencia de tokens para marcar el final del texto generado (str). Por defecto: None.</p>
<p><code>-mt</code> <code>--max_tokens</code><br />
Controla la longitud m√°xima del texto generado, en tokens (int). Por defecto: 200.</p>
<p><code>-pp</code> <code>--presence_penalty</code><br />
N√∫mero entre -2.0 y 2.0. Los valores m√°s altos aumentan la probabilidad de que el modelo hable sobre temas nuevos (float). Por defecto: 0.</p>
<p><code>-fp</code> <code>--frequency_penalty</code><br />
N√∫mero entre -2.0 y 2.0. Los valores m√°s altos reducen la probabilidad de que el modelo repita la misma l√≠nea palabra por palabra (float). Por defecto: 0.</p>
<p><code>-lb</code> <code>--logit_bias</code><br />
Un diccionario que especifica un sesgo hacia ciertos tokens en el texto generado (dict). Por defecto: None.</p>
<p><code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados, p. ej. ruta/al/directorio/resultados.txt. Por defecto: salida est√°ndar (STDOUT).</p>
<h3 id="por-ejemplo-2"><a class="header" href="#por-ejemplo-2">Por ejemplo</a></h3>
<pre><code class="language-bash">gget gpt &quot;C√≥mo est√°s hoy GPT?&quot; su_clave_api
</code></pre>
<pre><code class="language-python"># Python
print(gget.gpt(&quot;C√≥mo est√°s hoy GPT?&quot;, &quot;su_clave_api&quot;))
</code></pre>
<br>
<img width="725" alt="Screen Shot 2023-03-18 at 3 42 32 PM" src="https://user-images.githubusercontent.com/56094636/226143902-6fa2d0c7-7eea-4382-b1d2-df6c3f0d5fd5.png">
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-info--1"><a class="header" href="#gget-info--1">gget info üí°</a></h2>
<p>Obtenga informaci√≥n detallada sobre genes y transcripciones de <a href="https://www.ensembl.org/">Ensembl</a>, <a href="https://www.uniprot.org/">UniProt</a> y <a href="https://www.ncbi.nlm.nih.gov/">NCBI</a> utilizando sus IDs del tipo Ensembl.<br />
Regresa: Resultados en formato JSON (Terminal) o Dataframe/CSV (Python).</p>
<p><strong>Par√°metro posicional</strong><br />
<code>ens_ids</code><br />
Uno o m√°s ID del tipo Ensembl.</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados, p. ej. ruta/al/directorio/resultados.csv (o .json). Por defecto: salida est√°ndar (STDOUT).<br />
Para Python, usa <code>save=True</code> para guardar los resultados en el directorio de trabajo actual.</p>
<p><strong>Banderas</strong><br />
<code>-n</code> <code>--ncbi</code><br />
DESACTIVA los resultados de <a href="https://www.ncbi.nlm.nih.gov/">NCBI</a>.<br />
Para Python: <code>ncbi=False</code> evita la incluida de datos de NCBI (por defecto: True).</p>
<p><code>-u</code> <code>--uniprot</code><br />
DESACTIVA los resultados de <a href="https://www.uniprot.org/">UniProt</a>.<br />
Para Python: <code>uniprot=False</code> evita la incluida de datos de UniProt (por defecto: True).</p>
<p><code>-pdb</code> <code>--pdb</code><br />
INCLUYE <a href="https://www.ebi.ac.uk/pdbe/">PDB</a> IDs en los resultados (podr√≠a aumentar el tiempo de ejecuci√≥n).<br />
Para Python: <code>pdb=True</code> incluye IDs de PDB en los resultados (por defecto: False). </p>
<p><code>-csv</code> <code>--csv</code><br />
Solo para la Terminal. Regresa los resultados en formato CSV.<br />
Para Python, usa <code>json=True</code> para regresar los resultados en formato JSON.</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para la Terminal. Impide la informacion de progreso de ser exhibida durante la corrida.<br />
Para Python, usa <code>verbose=False</code> para imipidir la informacion de progreso de ser exhibida durante la corrida.</p>
<p><code>wrap_text</code><br />
Solo para Python. <code>wrap_text=True</code> muestra los resultados con texto envuelto para facilitar la lectura (por defecto: False).</p>
<h3 id="por-ejemplo-3"><a class="header" href="#por-ejemplo-3">Por ejemplo</a></h3>
<pre><code class="language-bash">gget info ENSG00000034713 ENSG00000104853 ENSG00000170296
</code></pre>
<pre><code class="language-python"># Python
gget.info([&quot;ENSG00000034713&quot;, &quot;ENSG00000104853&quot;, &quot;ENSG00000170296&quot;])
</code></pre>
<p>‚Üí Regresa informaci√≥n detallada sobre cada ID de Ensembl ingresada:</p>
<div class="table-wrapper"><table><thead><tr><th></th><th>uniprot_id</th><th>ncbi_gene_id</th><th>primary_gene_name</th><th>synonyms</th><th>protein_names</th><th>ensembl_description</th><th>uniprot_description</th><th>ncbi_description</th><th>biotype</th><th>canonical_transcript</th><th>...</th></tr></thead><tbody>
<tr><td>ENSG00000034713</td><td>P60520</td><td>11345</td><td>GABARAPL2</td><td>[ATG8, ATG8C, FLC3A, GABARAPL2, GATE-16, GATE16, GEF-2, GEF2]</td><td>Gamma-aminobutyric acid receptor-associated protein like 2 (GABA(A) receptor-associated protein-like 2)...</td><td>GABA type A receptor associated protein like 2 [Source:HGNC Symbol;Acc:HGNC:13291]</td><td>FUNCTION: Ubiquitin-like modifier involved in intra- Golgi traffic (By similarity). Modulates intra-Golgi transport through coupling between NSF activity and ...</td><td>Enables ubiquitin protein ligase binding activity. Involved in negative regulation of proteasomal protein catabolic process and protein...</td><td>protein_coding</td><td>ENST00000037243.7</td><td>...</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>...</td></tr>
</tbody></table>
</div>
<h4 id="more-examples-13"><a class="header" href="#more-examples-13"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-muscle--1"><a class="header" href="#gget-muscle--1">gget muscle ü¶æ</a></h2>
<p>Alinea m√∫ltiples secuencias de nucle√≥tidos o amino√°cidos usando el algoritmo <a href="https://www.drive5.com/muscle/">Muscle5</a>.<br />
Regresa: Salida est√°ndar (STDOUT) en formato ClustalW o archivo de tipo 'aligned FASTA' (.afa).</p>
<p><strong>Par√°metro posicional</strong><br />
<code>fasta</code><br />
Lista de secuencias o ruta al archivo FASTA o .txt que contiene las secuencias de nucle√≥tidos o amino√°cidos que se van a alinear.</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados, p. ej. ruta/al/directorio/resultados.afa. Por defecto: salida est√°ndar (STDOUT).<br />
Para Python, usa <code>save=True</code> para guardar los resultados en el directorio de trabajo actual.</p>
<p><strong>Banderas</strong><br />
<code>-s5</code> <code>--super5</code><br />
Alinea las secuencies usando el algoritmo <a href="https://drive5.com/muscle5/Muscle5_SuppMat.pdf">Super5</a> en lugar del algoritmo <a href="https://drive5.com/muscle5/Muscle5_SuppMat.pdf">Parallel Perturbed Probcons (PPP)</a> para disminuir el tiempo y la memoria usada durante la corrida.<br />
Use para ingresos grandes (unos cientos secuencias).</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para la Terminal. Impide la informacion de progreso de ser exhibida durante la corrida.<br />
Para Python, usa <code>verbose=False</code> para imipidir la informacion de progreso de ser exhibida durante la corrida.</p>
<h3 id="por-ejemplo-4"><a class="header" href="#por-ejemplo-4">Por ejemplo</a></h3>
<pre><code class="language-bash">gget muscle MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS
</code></pre>
<pre><code class="language-python"># Python
gget.muscle([&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;, &quot;MSSSSWLLLSLVEVTAAQSTIEQQAKTFLDKFHEAEDLFYQSLLAS&quot;])
</code></pre>
<pre><code class="language-bash">gget muscle fasta.fa
</code></pre>
<pre><code class="language-python"># Python
gget.muscle(&quot;fasta.fa&quot;)
</code></pre>
<p>‚Üí Regresa las secuencias alineadas con coloraci√≥n ClustalW. (Para devolver un archivo FASTA alineado (.afa), use el argumento <code>--out</code> (o <code>save=True</code> en Python).) En este ejemplo, el archivo 'fasta.fa' incluye varias secuencias para alineaci√≥n (por ejemplo, isoformas devueltas desde <code>gget seq</code>).</p>
<p><img src="https://github.com/pachterlab/gget/blob/main/figures/example_muscle_return.png?raw=true" alt="alt text" /></p>
<p>Tambi√©n puede ver archivos FASTA alineados devueltos por <code>gget.muscle</code> usando programas como <a href="https://github.com/arvestad/alv"><code>alv</code></a>:</p>
<pre><code class="language-python"># Python
!pip install biopython
!pip install alv
from Bio import AlignIO
import alv

gget.muscle(&quot;fasta.fa&quot;, out=&quot;fasta_aligned.afa&quot;)
msa = AlignIO.read(&quot;fasta_aligned.afa&quot;, &quot;fasta&quot;)
alv.view(msa)
</code></pre>
<h4 id="more-examples-14"><a class="header" href="#more-examples-14"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-pdb--1"><a class="header" href="#gget-pdb--1">gget pdb üîÆ</a></h2>
<p>Obtenga la estructura o los metadatos de una prote√≠na usando data de <a href="https://www.rcsb.org/">RCSB Protein Data Bank (PDB)</a>.<br />
Regresa: El archivo 'pdb' se regresa en formato PDB. Todos los dem√°s datos se regresan en formato JSON. </p>
<p><strong>Par√°metro posicional</strong><br />
<code>pdb_id</code><br />
ID del tipo PDB, p. ej. '7S7U'.</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-r</code> <code>--resource</code><br />
Define el tipo de informaci√≥n a regresar. Uno de los siguientes:<br />
'pdb': Regresa la estructura de la prote√≠na en formato PDB (regresa por defecto).<br />
'entry': Regresa informaci√≥n sobre las estructuras PDB en el nivel superior de la organizaci√≥n de datos PDB jer√°rquicos.<br />
'pubmed': Regresa anotaciones de PubMed (datos integrados de PubMed) para la cita principal de un ID PDB.<br />
'assembly': Regresa informaci√≥n sobre estructuras PDB en el nivel de estructura cuaternaria.<br />
'branched_entity': Regresa la descripci√≥n de la entidad ramificada (defina el ID de la entidad como <code>identifier</code>).<br />
'nonpolymer_entity': Regresa datos de entidades no polim√©ricas (defina el ID de la entidad como <code>identifier</code>).<br />
'polymer_entity': Regresa datos de entidades polim√©ricas (defina el ID de la entidad como <code>identifier</code>).<br />
'uniprot': Regresa anotaciones UniProt para una entidad macromolecular (defina el ID de la entidad como <code>identifier</code>).<br />
'branched_entity_instance': Regresa la descripci√≥n de instancia de entidad ramificada (defina el ID de cadena como <code>identifier</code>).<br />
'polymer_entity_instance': Regresa datos de instancia de entidad polim√©rica (tambi√©n conocida como cadena) (defina el ID de cadena como <code>identifier</code>).<br />
'nonpolymer_entity_instance': Regresa datos de instancia de entidad no polim√©rica (defina el ID de cadena como <code>identifier</code>). </p>
<p><code>-i</code> <code>--identifier</code><br />
Este par√°metro se puede utilizar para definir el ID de ensamblaje, entidad o cadena (po defecto: None). Los IDs de ensamblaje/entidad son n√∫meros (p. ej., 1) y los IDs de cadena son letras (p. ej., 'A').</p>
<p><code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados, p. ej. ruta/al/directorio/7S7U.pdb (o 7S7U_entry.json). Por defecto: salida est√°ndar (STDOUT).<br />
Para Python, usa <code>save=True</code> para guardar los resultados en el directorio de trabajo actual.</p>
<h3 id="por-ejemplo-5"><a class="header" href="#por-ejemplo-5">Por ejemplo</a></h3>
<pre><code class="language-bash">gget pdb 7S7U -o 7S7U.pdb
</code></pre>
<pre><code class="language-python"># Python
gget.pdb(&quot;7S7U&quot;, save=True)
</code></pre>
<p>‚Üí Guarda la estructura de 7S7U en formato PDB como '7S7U.pdb' en el directorio de trabajo actual.</p>
<p><strong>Encuentre estructuras cristalinas de PDB para un an√°lisis comparativo de la estructura de prote√≠nas:</strong></p>
<pre><code class="language-bash"># Encuentre IDs de PDB asociados con un ID de Ensembl
gget info ENSG00000130234

# Alternativamente: como que muchas entradas en el PDB no tienen ID de Ensembl vinculados,
# es probable que encuentre m√°s entradas de PDB BLASTing la secuencia contra el PDB:

# Obtenga la secuencia de amino√°cidos
gget seq --translate ENSG00000130234 -o gget_seq_results.fa

# BLAST la secuencia de amino√°cidos para encontrar estructuras similares en el PDB
gget blast --database pdbaa gget_seq_results.fa

# Obtenga archivos PDB de los IDs de PDB regresados por gget blast para un an√°lisis comparativo
gget pdb 7DQA -o 7DQA.pdb
gget pdb 7CT5 -o 7CT5.pdb
</code></pre>
<pre><code class="language-python"># Encuentre IDs de PDB asociados con un ID de Ensembl
gget.info(&quot;ENSG00000130234&quot;)

# Alternativamente: como que muchas entradas en el PDB no tienen ID de Ensembl vinculados,
# es probable que encuentre m√°s entradas de PDB BLASTing la secuencia contra el PDB:

# Obtenga la secuencia de amino√°cidos
gget.seq(&quot;ENSG00000130234&quot;, translate=True, save=True)

# BLAST la secuencia de amino√°cidos para encontrar estructuras similares en el PDB
gget.blast(&quot;gget_seq_results.fa&quot;, database=&quot;pdbaa&quot;)

# Obtenga archivos PDB de los IDs de PDB regresados por gget blast para un an√°lisis comparativo
gget.pdb(&quot;7DQA&quot;, save=True)
gget.pdb(&quot;7CT5&quot;, save=True)
</code></pre>
<p>‚Üí Este caso de uso ejemplifica c√≥mo encontrar archivos PDB para un an√°lisis comparativo de la estructura de las prote√≠nas asociado con IDs de Ensembl o secuencias de amino√°cidos. Los archivos PDB obtenidos tambi√©n se pueden comparar con las estructuras predichas generadas por <a href="es/alphafold.html"><code>gget alphafold</code></a>. Los archivos PDB se pueden ver de forma interactiva en 3D <a href="https://rcsb.org/3d-view">aqu√≠</a>, o usando programas como <a href="https://pymol.org/">PyMOL</a> o <a href="https://www.blender.org/">Blender</a>. M√∫ltiple archivos PDB se pueden visualizar para comparaci√≥n <a href="https://rcsb.org/alignment">aqu√≠</a>.</p>
<h4 id="m√°s-ejemplos-4"><a class="header" href="#m√°s-ejemplos-4"><a href="https://github.com/pachterlab/gget_examples">M√°s ejemplos</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-ref--1"><a class="header" href="#gget-ref--1">gget ref üìñ</a></h2>
<p>Obtenga enlaces FTP y sus respectivos metadatos (o use la bandera <code>ftp</code> para regresar solo los enlaces) para referenciar genomas y anotaciones de <a href="https://www.ensembl.org/">Ensembl</a>.<br />
Regresa: Resultados en formato JSON.</p>
<p><strong>Par√°metro posicional</strong><br />
<code>species</code><br />
La especie por la cual que se buscar√° los FTP en el formato g√©nero_especies, p. ej. homo_sapiens.<br />
Nota: No se requiere cuando se llama a la bandera <code>--list_species</code>.<br />
Accesos directos: 'human', 'mouse'</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-w</code> <code>--which</code><br />
Define qu√© resultados devolver. Por defecto: 'all' -&gt; Regresa todos los resultados disponibles.<br />
Las entradas posibles son uno solo o una combinaci√≥n de las siguientes (como lista separada por comas):<br />
'gtf'‚ÄÉ- Regresa la anotaci√≥n (GTF).<br />
'cdna'‚ÄÉ- Regresa el transcriptoma (cDNA).<br />
'dna'‚ÄÉ- Regresa el genoma (DNA).<br />
'cds'‚ÄÉ- Regresa las secuencias codificantes correspondientes a los genes Ensembl. (No contiene UTR ni secuencia intr√≥nica).<br />
'cdrna'‚ÄÉ- Regresa secuencias de transcripci√≥n correspondientes a genes de ARN no codificantes (ncRNA).<br />
'pep'‚ÄÉ- Regresa las traducciones de prote√≠nas de los genes Ensembl.</p>
<p><code>-r</code> <code>--release</code><br />
Define el n√∫mero de versi√≥n de Ensembl desde el que se obtienen los archivos, p. ej. 104. Default: latest Ensembl release.</p>
<p><code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados, p. ej. ruta/al/directorio/resultados.json. Por defecto: salida est√°ndar (STDOUT).<br />
Para Python, usa <code>save=True</code> para guardar los resultados en el directorio de trabajo actual.</p>
<p><strong>Banderas</strong><br />
<code>-l</code> <code>--list_species</code><br />
Enumera todas las especies disponibles. (Para Python: combina con <code>species=None</code>.)</p>
<p><code>-ftp</code> <code>--ftp</code><br />
Regresa solo los enlaces FTP solicitados.</p>
<p><code>-d</code> <code>--download</code><br />
Solo para Terminal. Descarga los FTP solicitados al directorio actual (requiere <a href="https://curl.se/docs/">curl</a> para ser instalado).</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para la Terminal. Impide la informacion de progreso de ser exhibida durante la corrida.<br />
Para Python, usa <code>verbose=False</code> para imipidir la informacion de progreso de ser exhibida durante la corrida.</p>
<h3 id="por-ejemplo-6"><a class="header" href="#por-ejemplo-6">Por ejemplo</a></h3>
<p><strong>Use <code>gget ref</code> en combinaci√≥n con <a href="https://www.kallistobus.tools/kb_usage/kb_ref/">kallisto | bustools</a> para construir un √≠ndice de referencia:</strong></p>
<pre><code class="language-bash">kb ref -i INDEX -g T2G -f1 FASTA $(gget ref --ftp -w dna,gtf homo_sapiens)
</code></pre>
<p>‚Üí kb ref crea un √≠ndice de referencia utilizando los √∫ltimos archivos de ADN y GTF de especies <strong>Homo sapiens</strong> que le ha pasado <code>gget ref</code>.</p>
<p><br/><br/></p>
<p><strong>Enumere todos los genomas disponibles de la versi√≥n 103 de Ensembl:</strong></p>
<pre><code class="language-bash">gget ref --list_species -r 103
</code></pre>
<pre><code class="language-python"># Python
gget.ref(species=None, list_species=True, release=103)
</code></pre>
<p>‚Üí Regresa una lista con todos los genomas disponibles (<code>gget ref</code> verifica si GTF y FASTA est√°n disponibles) de la versi√≥n 103 de Ensembl.<br />
(Si no se especifica ninguna versi√≥n, <code>gget ref</code> siempre devolver√° informaci√≥n de la √∫ltima versi√≥n de Ensembl).</p>
<p><br/><br/></p>
<p><strong>Obtenga la referencia del genoma para una especie espec√≠fica:</strong></p>
<pre><code class="language-bash">gget ref -w gtf,dna homo_sapiens
</code></pre>
<pre><code class="language-python"># Python
gget.ref(&quot;homo_sapiens&quot;, which=[&quot;gtf&quot;, &quot;dna&quot;])
</code></pre>
<p>‚Üí Regresa un JSON con los √∫ltimos FTP humanos GTF y FASTA, y sus respectivos metadatos, en el formato:</p>
<pre><code>{
    &quot;homo_sapiens&quot;: {
        &quot;annotation_gtf&quot;: {
            &quot;ftp&quot;: &quot;http://ftp.ensembl.org/pub/release-106/gtf/homo_sapiens/Homo_sapiens.GRCh38.106.gtf.gz&quot;,
            &quot;ensembl_release&quot;: 106,
            &quot;release_date&quot;: &quot;28-Feb-2022&quot;,
            &quot;release_time&quot;: &quot;23:27&quot;,
            &quot;bytes&quot;: &quot;51379459&quot;
        },
        &quot;genome_dna&quot;: {
            &quot;ftp&quot;: &quot;http://ftp.ensembl.org/pub/release-106/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz&quot;,
            &quot;ensembl_release&quot;: 106,
            &quot;release_date&quot;: &quot;21-Feb-2022&quot;,
            &quot;release_time&quot;: &quot;09:35&quot;,
            &quot;bytes&quot;: &quot;881211416&quot;
        }
    }
}
</code></pre>
<h4 id="m√°s-ejemplos-5"><a class="header" href="#m√°s-ejemplos-5"><a href="https://github.com/pachterlab/gget_examples">M√°s ejemplos</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-search--1"><a class="header" href="#gget-search--1">gget search üîé</a></h2>
<p>Obtenga genes y transcripciones de <a href="https://www.ensembl.org/">Ensembl</a> usando t√©rminos de b√∫squeda de forma libre.<br />
Los resultados se comparan seg√∫n las secciones &quot;nombre del gen&quot; y &quot;descripci√≥n&quot; en la base de datos de Ensembl. <code>gget</code> versi√≥n &gt;= 0.27.9 tambi√©n incluye resultados que coinciden con la secci√≥n &quot;sin√≥nimo&quot; de Ensembl.<br />
Regresa: Resultados en formato JSON (Terminal) o Dataframe/CSV (Python).</p>
<p><strong>Par√°metro posicional</strong><br />
<code>searchwords</code><br />
Una o m√°s palabras de b√∫squeda de forma libre, p. ej. gaba nmda. (Nota: la b√∫squeda no distingue entre may√∫sculas y min√∫sculas).</p>
<p><strong>Otros par√°metros requeridos</strong><br />
<code>-s</code> <code>--species</code><br />
Especies o base de datos a buscar.<br />
Una especie se puede pasar en el formato 'g√©nero_especie', p. ej. 'homo_sapiens' o 'arabidopsis_thaliana'.<br />
Para pasar una base de datos espec√≠fica, pase el nombre de la base de datos CORE, p. ej. 'mus_musculus_dba2j_core_105_1'.<br />
Todas las bases de datos disponibles para cada versi√≥n de Ensembl se pueden encontrar aqu√≠:<br />
Vertebrados: <a href="http://ftp.ensembl.org/pub/current/mysql/">http://ftp.ensembl.org/pub/current/mysql/</a><br />
Invertebrados: <a href="http://ftp.ensemblgenomes.org/pub/current/">http://ftp.ensemblgenomes.org/pub/current/</a> + selecciona reino animal + selecciona mysql/<br />
Accesos directos: 'human', 'mouse'. </p>
<p><strong>Par√°metros optionales</strong><br />
<code>-r</code> <code>--release</code><br />
Define el n√∫mero de versi√≥n de Ensembl desde el que se obtienen los archivos, p. ej. 104. Por defecto: None -&gt; se usa la √∫ltima versi√≥n de Ensembl.<br />
Nota: <em>No se aplica a las especies invertebrados</em> (en su lugar, puede pasar una base de datos de una especies espec√≠fica (incluyen un n√∫mero de versi√≥n) al argumento <code>species</code>).<br />
Este argumento se sobrescribe si se pasa una base de datos espec√≠fica (que incluye un n√∫mero de publicaci√≥n) al argumento <code>species</code>.</p>
<p><code>-t</code> <code>--id_type</code><br />
'gene' (esto se use por defecto) o 'transcript'<br />
Regesa genes o transcripciones, respectivamente.</p>
<p><code>-ao</code> <code>--andor</code><br />
'or' (esto se use por defecto) o 'and'<br />
'or' ('o'): Regresa todos los genes que INCLUYEN AL MENOS UNA de las palabras de b√∫squeda en su nombre/descripci√≥n.<br />
'and' ('y'): Regresa solo los genes que INCLUYEN TODAS las palabras de b√∫squeda en su nombre/descripci√≥n.</p>
<p><code>-l</code> <code>--limit</code><br />
Limita el n√∫mero de resultados de b√∫squeda, p. ej. 10. Por defecto: None.</p>
<p><code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados, p. ej. ruta/al/directorio/resultados.csv (o .json). Por defecto: salida est√°ndar (STDOUT).<br />
Para Python, usa <code>save=True</code> para guardar los resultados en el directorio de trabajo actual.</p>
<p><strong>Banderas</strong><br />
<code>-csv</code> <code>--csv</code><br />
Solo para la Terminal. Regresa los resultados en formato CSV.<br />
Para Python, usa <code>json=True</code> para regresar los resultados en formato JSON.</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para la Terminal. Impide la informacion de progreso de ser exhibida durante la corrida.<br />
Para Python, usa <code>verbose=False</code> para imipidir la informacion de progreso de ser exhibida durante la corrida.</p>
<p><code>wrap_text</code><br />
Solo para Python. <code>wrap_text=True</code> muestra los resultados con texto envuelto para facilitar la lectura (por defecto: False). </p>
<h3 id="por-ejemplo-7"><a class="header" href="#por-ejemplo-7">Por ejemplo</a></h3>
<pre><code class="language-bash">gget search -s human gaba gamma-aminobutyric
</code></pre>
<pre><code class="language-python"># Python
gget.search([&quot;gaba&quot;, &quot;gamma-aminobutyric&quot;], &quot;homo_sapiens&quot;)
</code></pre>
<p>‚Üí Regresa todos los genes que contienen al menos una de las palabras de b√∫squeda en su nombre o descripci√≥n de Ensembl/referencia externa:</p>
<div class="table-wrapper"><table><thead><tr><th>ensembl_id</th><th>gene_name</th><th>ensembl_description</th><th>ext_ref_description</th><th>biotype</th><th>url</th></tr></thead><tbody>
<tr><td>ENSG00000034713</td><td>GABARAPL2</td><td>GABA type A receptor associated protein like 2 [Source:HGNC Symbol;Acc:HGNC:13291]</td><td>GABA type A receptor associated protein like 2</td><td>protein_coding</td><td>https://uswest.ensembl.org/homo_sapiens/Gene/Summary?g=ENSG00000034713</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td></tr>
</tbody></table>
</div>
<h4 id="m√°s-ejemplos-6"><a class="header" href="#m√°s-ejemplos-6"><a href="https://github.com/pachterlab/gget_examples">M√°s ejemplos</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-setup--1"><a class="header" href="#gget-setup--1">gget setup üîß</a></h2>
<p>Funci√≥n para instalar/descargar dependencias de terceros para un m√≥dulo de gget.</p>
<p><strong>Par√°metro posicional</strong><br />
<code>module</code><br />
M√≥dulo gget para el que se deben instalar las dependencias.</p>
<h3 id="por-ejemplo-8"><a class="header" href="#por-ejemplo-8">Por ejemplo</a></h3>
<pre><code class="language-bash">gget setup alphafold
</code></pre>
<pre><code class="language-python"># Python
gget.setup(&quot;alphafold&quot;)
</code></pre>
<p>‚Üí Instala todas las dependencias de terceros (modificadas) y descarga los par√°metros del algoritmo (~4 GB) necesarios para ejecutar <a href="es/alphafold.html"><code>gget alphafold</code></a>.</p>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Par√°metros de Python s√≥n iguales a los par√°metros largos (<code>--par√°metro</code>) de Terminal, si no especificado de otra manera. Banderas son par√°metros de verdadero o falso (True/False) en Python. El manu√°l para cualquier modulo de gget se puede llamar desde la Terminal con la bandera <code>-h</code> <code>--help</code>.</p>
</blockquote>
<h2 id="gget-seq--1"><a class="header" href="#gget-seq--1">gget seq üß¨</a></h2>
<p>Obtenga la(s) secuencia(s) nucle√≥tidos o amino√°cidos de un gen (y todas sus isoformas) con su ID de Ensembl.<br />
Regresa: Archivo de tipo FASTA.</p>
<p><strong>Par√°metro posicional</strong><br />
<code>ens_ids</code><br />
One or more Ensembl IDs.</p>
<p><strong>Par√°metros optionales</strong><br />
<code>-o</code> <code>--out</code><br />
Ruta al archivo en el que se guardar√°n los resultados, p. ruta/al/directorio/resultados.fa. Por defecto: salida est√°ndar (STDOUT).<br />
Para Python, usa <code>save=True</code> para guardar los resultados en el directorio de trabajo actual.</p>
<p><strong>Banderas</strong><br />
<code>-t</code> <code>--translate</code><br />
Regresa secuencias de amino√°cidos (en lugar de nucle√≥tidos).<br />
Las secuencias de nucle√≥tidos se obtienen de <a href="https://www.ensembl.org/">Ensembl</a>.<br />
Las secuencias de amino√°cidos se obtienen de <a href="https://www.uniprot.org/">UniProt</a>.</p>
<p><code>-iso</code> <code>--isoforms</code><br />
Regresa las secuencias de todas las transcripciones conocidas.<br />
(Solo para IDs de genes).</p>
<p><code>-q</code> <code>--quiet</code><br />
Solo para la Terminal. Impide la informacion de progreso de ser exhibida durante la corrida.<br />
Para Python, usa <code>verbose=False</code> para imipidir la informacion de progreso de ser exhibida durante la corrida.</p>
<h3 id="por-ejemplo-9"><a class="header" href="#por-ejemplo-9">Por ejemplo</a></h3>
<pre><code class="language-bash">gget seq ENSG00000034713 ENSG00000104853 ENSG00000170296
</code></pre>
<pre><code class="language-python"># Python
gget.seq([&quot;ENSG00000034713&quot;, &quot;ENSG00000104853&quot;, &quot;ENSG00000170296&quot;])
</code></pre>
<p>‚Üí Regresa las secuencias de nucle√≥tidos de ENSG00000034713, ENSG00000104853, y ENSG00000170296 en formato FASTA.</p>
<p><br/><br/></p>
<pre><code class="language-bash">gget seq -t -iso ENSG00000034713
</code></pre>
<pre><code class="language-python"># Python
gget.seq(&quot;ENSG00000034713&quot;, translate=True, isoforms=True)
</code></pre>
<p>‚Üí Regresa las secuencias de amino√°cidos de todas las transcripciones conocidas de ENSG00000034713 en formato FASTA.</p>
<h4 id="m√°s-ejemplos-7"><a class="header" href="#m√°s-ejemplos-7"><a href="https://github.com/pachterlab/gget_examples">M√°s ejemplos</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><h1 id="bienvenido-a-la-gu√≠a-de-contribuci√≥n-de-gget"><a class="header" href="#bienvenido-a-la-gu√≠a-de-contribuci√≥n-de-gget">Bienvenido a la gu√≠a de contribuci√≥n de gget</a></h1>
<p>¬°Gracias por invertir su tiempo en contribuir con nuestro proyecto! Cualquier contribuci√≥n que hagas se ver√° reflejada en el <a href="https://github.com/pachterlab/gget">repositorio de GitHub de gget</a>. ‚ú®</p>
<p>Lea nuestro <a href="es/./code_of_conduct.html">C√≥digo de conducta</a> para mantener nuestra comunidad accesible y respetable.</p>
<p>En esta gu√≠a, obtendr√° una descripci√≥n general del flujo de trabajo de contribuci√≥n desde la creaci√≥n de un GitHub Issue (asunto) o la creaci√≥n de un GitHub Pull Request (PR) hasta la revisi√≥n y fusi√≥n de un PR.</p>
<h2 id="issues-asuntos"><a class="header" href="#issues-asuntos">Issues (asuntos)</a></h2>
<h3 id="crear-un-nuevo-issue"><a class="header" href="#crear-un-nuevo-issue">Crear un nuevo Issue</a></h3>
<p>Si detecta un problema con gget o tiene una idea para una nueva funci√≥n, <a href="https://github.com/pachterlab/gget/issues">comproba si ya existe un Issue para este problema/sugerencia</a>. Si no existe un Issue relacionado, puede abrir un nuevo Issue utilizando el <a href="https://github.com/pachterlab/gget/issues/new/choose">formulario correspondiente</a>.</p>
<h3 id="resolver-un-issue"><a class="header" href="#resolver-un-issue">Resolver un Issue</a></h3>
<p>Explore nuestros <a href="https://github.com/pachterlab/gget/issues">Issues existentes</a> para encontrar uno que le interese. Puede restringir la b√∫squeda utilizando &quot;labels&quot; como filtros. Si encuentra un Issue en el que desea trabajar, puede abrir un PR con una soluci√≥n.</p>
<h2 id="contribuir-a-trav√©s-de-pull-requests-prs"><a class="header" href="#contribuir-a-trav√©s-de-pull-requests-prs">Contribuir a trav√©s de Pull Requests (PRs)</a></h2>
<h3 id="empezar"><a class="header" href="#empezar">Empezar</a></h3>
<ol>
<li>Bifurcar (&quot;fork&quot;) el <a href="https://github.com/pachterlab/gget">repositorio de GitHub de gget</a>.</li>
</ol>
<ul>
<li>
<p>Usando GitHub Desktop:</p>
<ul>
<li><a href="https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/getting-started-with-github-desktop">&quot;Getting started with GitHub Desktop&quot;</a> lo guiar√° a trav√©s de la configuraci√≥n de Desktop.</li>
<li>Una vez que GitHub Desktop est√° configurado, puede usarlo para <a href="https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/cloning-and-forking-repositories-from-github-desktop">bifurcar el repositorio</a>!</li>
</ul>
</li>
<li>
<p>Usando la Terminal:</p>
<ul>
<li><a href="https://docs.github.com/en/github/getting-started-with-github/fork-a-repo#fork-an-example-repository">Bifurca el repositorio</a> para que pueda realizar sus cambios sin afectando el proyecto original hasta que est√© listo para fusionarlos.</li>
</ul>
</li>
</ul>
<ol start="2">
<li>¬°Cree una rama de trabajo y comience con sus cambios!</li>
</ol>
<h3 id="confirma-sus-actualizaciones"><a class="header" href="#confirma-sus-actualizaciones">Confirma sus actualizaciones</a></h3>
<p>Confirme sus cambios una vez que est√© satisfecho con ellos.</p>
<h3 id="-auto-revisa-lo-siguiente-antes-de-crear-un-pr-"><a class="header" href="#-auto-revisa-lo-siguiente-antes-de-crear-un-pr-">‚ÄºÔ∏è Auto-revisa lo siguiente antes de crear un PR ‚ÄºÔ∏è</a></h3>
<ol>
<li>Revise el contenido para mantener precisi√≥n t√©cnica.</li>
<li>Edite los cambios/comentarios de gram√°tica, ortograf√≠a y adherencia al estilo general del c√≥digo de gget existente.</li>
<li>Formatee su c√≥digo usando <a href="https://black.readthedocs.io/en/stable/getting_started.html">&quot;black&quot;</a>.</li>
<li>Aseg√∫rese de que las pruebas unitarias pasen:
<ul>
<li>Las dependencias de desarrollador se pueden instalar con <code>pip install -r dev-requirements.txt</code></li>
<li>Ejecute pruebas unitarias existentes desde la carpeta de gget con <code>coverage run -m pytest -ra -v tests &amp;&amp; coverage report --omit=main.py,tests*</code></li>
</ul>
</li>
<li>Agregue nuevas pruebas unitarias si corresponde:
<ul>
<li>Los par√°metros y los resultados esperados se pueden encontrar en archivos json en ./tests/fixtures/</li>
<li>Las pruebas unitarias se pueden agregar a ./tests/test_*.py y ser√°n detectado autom√°ticamente</li>
</ul>
</li>
<li>Aseg√∫rese de que las ediciones sean compatibles tanto con Python como con la Terminal
<ul>
<li>Los par√°metros para la Terminal se definen en ./gget/main.py</li>
</ul>
</li>
<li>Agregue m√≥dulos/argumentos nuevos a la documentaci√≥n, si corresponde:
<ul>
<li>El manual de cada m√≥dulo se puede editar/a√±adir como ./docs/src/*.md</li>
</ul>
</li>
</ol>
<p>Si tiene alguna pregunta, no dude en iniciar una <a href="https://github.com/pachterlab/gget/discussions">discusi√≥n</a> o crear un Issue como se describe anteriormente.</p>
<h3 id="crear-un-pull-request-pr"><a class="header" href="#crear-un-pull-request-pr">Crear un Pull Request (PR)</a></h3>
<p>Cuando haya terminado con los cambios, <a href="https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request">cree un Pull Request</a>, tambi√©n conocido como &quot;PR&quot;.</p>
<p>‚ÄºÔ∏è Realice todos los PRs contra la rama <code>dev</code> del repositorio gget</p>
<ul>
<li>No olvide de <a href="https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue">vincular su PR con un Issue</a> si est√°s resolviendo uno.</li>
<li>Habilite la casilla de verificaci√≥n para <a href="https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork">permitir ediciones del mantenedor</a> para que la rama se pueda actualizar para una fusi√≥n.</li>
<li>Si se encuentra con problemas durante la fusi√≥n, consulte este <a href="https://github.com/skills/resolve-merge-conflicts">tutorial de git</a> para ayudarlo a resolver conflictos de fusi√≥n y otros problemas.</li>
</ul>
<p>Una vez que env√≠e su PR, un miembro del equipo gget revisar√° su propuesta. Podemos hacer preguntas o solicitar informaci√≥n adicional.</p>
<h3 id="su-pr-est√°-fusionado"><a class="header" href="#su-pr-est√°-fusionado">¬°Su PR est√° fusionado!</a></h3>
<p>¬°Felicidades! üéâ	 El equipo de gget te lo agradece. ‚ú®</p>
<p>Una vez que su PR se fusione, sus contribuciones ser√°n visibles p√∫blicamente en el <a href="https://github.com/pachterlab/gget">repositorio de gget</a>.</p>
<div style="break-before: page; page-break-before: always;"></div><h1 id="c√≥digo-de-conducta-convenido-para-contribuyentes"><a class="header" href="#c√≥digo-de-conducta-convenido-para-contribuyentes">C√≥digo de Conducta convenido para Contribuyentes</a></h1>
<h2 id="nuestro-compromiso"><a class="header" href="#nuestro-compromiso">Nuestro compromiso</a></h2>
<p>Nosotros, como miembros, contribuyentes y administradores nos comprometemos a hacer de la participaci√≥n en nuestra comunidad sea una experiencia libre de acoso para todo el mundo, independientemente de la edad, dimensi√≥n corporal, discapacidad visible o invisible, etnicidad, caracter√≠sticas sexuales, identidad y expresi√≥n de g√©nero, nivel de experiencia, educaci√≥n, nivel socio-econ√≥mico, nacionalidad, apariencia personal, raza, casta, color, religi√≥n, o identidad u orientaci√≥n sexual.</p>
<p>Nos comprometemos a actuar e interactuar de maneras que contribuyan a una comunidad abierta, acogedora, diversa, inclusiva y sana.</p>
<h2 id="nuestros-est√°ndares"><a class="header" href="#nuestros-est√°ndares">Nuestros est√°ndares</a></h2>
<p>Ejemplos de comportamientos que contribuyen a crear un ambiente positivo para nuestra comunidad:</p>
<ul>
<li>Demostrar empat√≠a y amabilidad ante otras personas</li>
<li>Respeto a diferentes opiniones, puntos de vista y experiencias</li>
<li>Dar y aceptar adecuadamente retroalimentaci√≥n constructiva</li>
<li>Aceptar la responsabilidad y disculparse ante quienes se vean afectados por nuestros errores, aprendiendo de la experiencia</li>
<li>Centrarse en lo que sea mejor no s√≥lo para nosotros como individuos, sino para la comunidad en general</li>
</ul>
<p>Ejemplos de comportamiento inaceptable:</p>
<ul>
<li>El uso de lenguaje o im√°genes sexualizadas, y aproximaciones o
atenciones sexuales de cualquier tipo</li>
<li>Comentarios despectivos (<em>trolling</em>), insultantes o derogatorios, y ataques personales o pol√≠ticos</li>
<li>El acoso en p√∫blico o privado</li>
<li>Publicar informaci√≥n privada de otras personas, tales como direcciones f√≠sicas o de correo
electr√≥nico, sin su permiso expl√≠cito</li>
<li>Otras conductas que puedan ser razonablemente consideradas como inapropiadas en un
entorno profesional</li>
</ul>
<h2 id="aplicaci√≥n-de-las-responsabilidades"><a class="header" href="#aplicaci√≥n-de-las-responsabilidades">Aplicaci√≥n de las responsabilidades</a></h2>
<p>Los administradores de la comunidad son responsables de aclarar y hacer cumplir nuestros est√°ndares de comportamiento aceptable y tomar√°n acciones apropiadas y correctivas de forma justa en respuesta a cualquier comportamiento que consideren inapropiado, amenazante, ofensivo o da√±ino.</p>
<p>Los administradores de la comunidad tendr√°n el derecho y la responsabilidad de eliminar, editar o rechazar comentarios, <em>commits</em>, c√≥digo, ediciones de p√°ginas de wiki, <em>issues</em> y otras contribuciones que no se alineen con este C√≥digo de Conducta, y comunicar√°n las razones para sus decisiones de moderaci√≥n cuando sea apropiado.</p>
<h2 id="alcance"><a class="header" href="#alcance">Alcance</a></h2>
<p>Este c√≥digo de conducta aplica tanto a espacios del proyecto como a espacios p√∫blicos donde un individuo est√© en representaci√≥n del proyecto o comunidad. Ejemplos de esto incluyen el uso de la cuenta oficial de correo electr√≥nico, publicaciones a trav√©s de las redes sociales oficiales, o presentaciones con personas designadas en eventos en l√≠nea o no.</p>
<h2 id="aplicaci√≥n"><a class="header" href="#aplicaci√≥n">Aplicaci√≥n</a></h2>
<p>Instancias de comportamiento abusivo, acosador o inaceptable de otro modo podr√°n ser reportadas a los administradores de la comunidad responsables del cumplimiento a trav√©s de lpachter@caltech.edu. Todas las quejas ser√°n evaluadas e investigadas de una manera puntual y justa.</p>
<p>Todos los administradores de la comunidad est√°n obligados a respetar la privacidad y la seguridad de quienes reporten incidentes.</p>
<h2 id="gu√≠as-de-aplicaci√≥n"><a class="header" href="#gu√≠as-de-aplicaci√≥n">Gu√≠as de Aplicaci√≥n</a></h2>
<p>Los administradores de la comunidad seguir√°n estas Gu√≠as de Impacto en la Comunidad para determinar las consecuencias de cualquier acci√≥n que juzguen como un incumplimiento de este C√≥digo de Conducta:</p>
<h3 id="1-correcci√≥n"><a class="header" href="#1-correcci√≥n">1. Correcci√≥n</a></h3>
<p><strong>Impacto en la Comunidad</strong>: El uso de lenguaje inapropiado u otro comportamiento considerado no profesional o no acogedor en la comunidad.</p>
<p><strong>Consecuencia</strong>: Un aviso escrito y privado por parte de los administradores de la comunidad, proporcionando claridad alrededor de la naturaleza de este incumplimiento y una explicaci√≥n de por qu√© el comportamiento es inaceptable. Una disculpa p√∫blica podr√≠a ser solicitada.</p>
<h3 id="2-aviso"><a class="header" href="#2-aviso">2. Aviso</a></h3>
<p><strong>Impacto en la Comunidad</strong>: Un incumplimiento causado por un √∫nico incidente o por una cadena de acciones.</p>
<p><strong>Consecuencia</strong>: Un aviso con consecuencias por comportamiento prolongado. No se interact√∫a con las personas involucradas, incluyendo interacci√≥n no solicitada con quienes se encuentran aplicando el C√≥digo de Conducta, por un periodo especificado de tiempo. Esto incluye evitar las interacciones en espacios de la comunidad, as√≠ como a trav√©s de canales externos como las redes sociales. Incumplir estos t√©rminos puede conducir a una expulsi√≥n temporal o permanente.</p>
<h3 id="3-expulsi√≥n-temporal"><a class="header" href="#3-expulsi√≥n-temporal">3. Expulsi√≥n temporal</a></h3>
<p><strong>Impacto en la Comunidad</strong>: Una serie de incumplimientos de los est√°ndares de la comunidad, incluyendo comportamiento inapropiado continuo.</p>
<p><strong>Consecuencia</strong>: Una expulsi√≥n temporal de cualquier forma de interacci√≥n o comunicaci√≥n p√∫blica con la comunidad durante un intervalo de tiempo especificado. No se permite interactuar de manera p√∫blica o privada con las personas involucradas, incluyendo interacciones no solicitadas con quienes se encuentran aplicando el C√≥digo de Conducta, durante este periodo. Incumplir estos t√©rminos puede conducir a una expulsi√≥n permanente.</p>
<h3 id="4-expulsi√≥n-permanente"><a class="header" href="#4-expulsi√≥n-permanente">4. Expulsi√≥n permanente</a></h3>
<p><strong>Impacto en la Comunidad</strong>: Demostrar un patr√≥n sistem√°tico de incumplimientos de los est√°ndares de la comunidad, incluyendo conductas inapropiadas prolongadas en el tiempo, acoso de individuos, o agresiones o menosprecio a grupos de individuos.</p>
<p><strong>Consecuencia</strong>: Una expulsi√≥n permanente de cualquier tipo de interacci√≥n p√∫blica con la comunidad del proyecto.</p>
<h2 id="atribuci√≥n"><a class="header" href="#atribuci√≥n">Atribuci√≥n</a></h2>
<p>Este C√≥digo de Conducta es una adaptaci√≥n del <a href="https://www.contributor-covenant.org">Contributor Covenant</a>, versi√≥n 2.1,
disponible en https://www.contributor-covenant.org/es/version/2/1/code_of_conduct.html</p>
<p>Las Gu√≠as de Impacto en la Comunidad est√°n inspiradas en la <a href="https://github.com/mozilla/diversity">escalera de aplicaci√≥n del c√≥digo de conducta de Mozilla</a>.</p>
<p>Para respuestas a las preguntas frecuentes de este c√≥digo de conducta, consulta las FAQ en
<a href="https://www.contributor-covenant.org/faq">https://www.contributor-covenant.org/faq</a>. Hay traducciones disponibles en <a href="https://www.contributor-covenant.org/translations">https://www.contributor-covenant.org/translations</a>https://www.contributor-covenant.org/translations</p>
<div style="break-before: page; page-break-before: always;"></div><p><a href="es/LICENSE"><img src="https://img.shields.io/pypi/l/gget" alt="license" /></a></p>
<h1 id="citar"><a class="header" href="#citar">Citar</a></h1>
<p>Si utiliza <code>gget</code> en una publicaci√≥n, favor de citar:<br />
Luebbert, L., &amp; Pachter, L. (2023). Efficient querying of genomic reference databases with gget. Bioinformatics. https://doi.org/10.1093/bioinformatics/btac836</p>
<ul>
<li>
<p>Si utiliza <code>gget alphafold</code>, favor de citar:</p>
<ul>
<li>Jumper, J., Evans, R., Pritzel, A. et al. Highly accurate protein structure prediction with AlphaFold. Nature 596, 583‚Äì589 (2021). https://doi.org/10.1038/s41586-021-03819-2</li>
</ul>
<p>Y si aplica:</p>
<ul>
<li>Evans, R. et al. Protein complex prediction with AlphaFold-Multimer. bioRxiv 2021.10.04.463034; https://doi.org/10.1101/2021.10.04.463034</li>
</ul>
</li>
<li>
<p>Si utiliza <code>gget archs4</code>, favor de citar:</p>
<ul>
<li>
<p>Lachmann A, Torre D, Keenan AB, Jagodnik KM, Lee HJ, Wang L, Silverstein MC, Ma‚Äôayan A. Massive mining of publicly available RNA-seq data from human and mouse. Nature Communications 9. Article number: 1366 (2018), doi:10.1038/s41467-018-03751-6</p>
</li>
<li>
<p>Bray NL, Pimentel H, Melsted P and Pachter L, Near optimal probabilistic RNA-seq quantification, Nature Biotechnology 34, p 525--527 (2016). https://doi.org/10.1038/nbt.3519</p>
</li>
</ul>
</li>
<li>
<p>Si utilizan <code>gget blast</code>, por favor tambi√©n de citar:</p>
<ul>
<li>Altschul SF, Gish W, Miller W, Myers EW, Lipman DJ. Basic local alignment search tool. J Mol Biol. 1990 Oct 5;215(3):403-10. doi: 10.1016/S0022-2836(05)80360-2. PMID: 2231712.</li>
</ul>
</li>
<li>
<p>Si utiliza <code>gget blat</code>, favor de citar:</p>
<ul>
<li>Kent WJ. BLAT--the BLAST-like alignment tool. Genome Res. 2002 Apr;12(4):656-64. doi: 10.1101/gr.229202. PMID: 11932250; PMCID: PMC187518.</li>
</ul>
</li>
<li>
<p>Si utiliza <code>gget cellxgene</code>, favor de citar:</p>
<ul>
<li>Chanzuckerberg Initiative. (n.d.). CZ CELLxGENE Discover. Retrieved [insert date here], from https://cellxgene.cziscience.com/</li>
</ul>
</li>
<li>
<p>Si utiliza <code>gget diamond</code>, favor de citar:</p>
<ul>
<li>Buchfink, B., Xie, C. &amp; Huson, D. Fast and sensitive protein alignment using DIAMOND. Nat Methods 12, 59‚Äì60 (2015). https://doi.org/10.1038/nmeth.3176</li>
</ul>
</li>
<li>
<p>Si utiliza <code>gget elm</code>, favor de citar:</p>
<ul>
<li>Manjeet Kumar, Sushama Michael, Jes√∫s Alvarado-Valverde, B√°lint M√©sz√°ros, Hugo S√°mano‚ÄêS√°nchez, Andr√°s Zeke, Laszlo Dobson, Tamas Lazar, Mihkel √ñrd, Anurag Nagpal, Nazanin Farahi, Melanie K√§ser, Ramya Kraleti, Norman E Davey, Rita Pancsa, Luc√≠a B Chemes, Toby J Gibson, The Eukaryotic Linear Motif resource: 2022 release, Nucleic Acids Research, Volume 50, Issue D1, 7 January 2022, Pages D497‚ÄìD508, https://doi.org/10.1093/nar/gkab975</li>
</ul>
</li>
<li>
<p>Si utiliza <code>gget enrichr</code>, favor de citar:</p>
<ul>
<li>
<p>Chen EY, Tan CM, Kou Y, Duan Q, Wang Z, Meirelles GV, Clark NR, Ma'ayan A.
Enrichr: interactive and collaborative HTML5 gene list enrichment analysis tool. BMC Bioinformatics. 2013; 128(14).</p>
</li>
<li>
<p>Kuleshov MV, Jones MR, Rouillard AD, Fernandez NF, Duan Q, Wang Z, Koplev S, Jenkins SL, Jagodnik KM, Lachmann A, McDermott MG, Monteiro CD, Gundersen GW, Ma'ayan A.
Enrichr: a comprehensive gene set enrichment analysis web server 2016 update. Nucleic Acids Research. 2016; gkw377.</p>
</li>
<li>
<p>Xie Z, Bailey A, Kuleshov MV, Clarke DJB., Evangelista JE, Jenkins SL, Lachmann A, Wojciechowicz ML, Kropiwnicki E, Jagodnik KM, Jeon M, &amp; Ma‚Äôayan A.
Gene set knowledge discovery with Enrichr. Current Protocols, 1, e90. 2021. doi: 10.1002/cpz1.90. </p>
</li>
</ul>
</li>
<li>
<p>Si utiliza <code>gget muscle</code>, favor de citar:</p>
<ul>
<li>Edgar RC (2021), MUSCLE v5 enables improved estimates of phylogenetic tree confidence by ensemble bootstrapping, bioRxiv 2021.06.20.449169. https://doi.org/10.1101/2021.06.20.449169.</li>
</ul>
</li>
</ul>
<hr />
<h1 id="descargo-de-responsabilidad"><a class="header" href="#descargo-de-responsabilidad">Descargo de responsabilidad</a></h1>
<p><code>gget</code> es tan preciso como la base de datos/servidores/APIs que utiliza. La exactitud o fiabilidad de los datos no es garantizada por ning√∫n motivo. Los proveedores por ning√∫n motivo seran responsables de (incluyendo, sin limite alguno) la calidad, ejecuci√≥n, o comerciabilidad para cualquier prop√≥sito particular surgiendo del uso o la incapacidad de usar los datos.</p>

                    </main>

                    <nav class="nav-wrapper" aria-label="Page navigation">
                        <!-- Mobile navigation buttons -->


                        <div style="clear: both"></div>
                    </nav>
                </div>
            </div>

            <nav class="nav-wide-wrapper" aria-label="Page navigation">

            </nav>

        </div>




        <script>
            window.playground_copyable = true;
        </script>


        <script src="elasticlunr.min.js"></script>
        <script src="mark.min.js"></script>
        <script src="searcher.js"></script>

        <script src="clipboard.min.js"></script>
        <script src="highlight.js"></script>
        <script src="book.js"></script>

        <!-- Custom JS scripts -->

        <script>
        window.addEventListener('load', function() {
            window.setTimeout(window.print, 100);
        });
        </script>

    </div>
    </body>
</html>
